diff --git a/cmake/Modules/GNUInstallDirs.cmake b/cmake/Modules/GNUInstallDirs.cmake
index 01bd63716..e264b40ee 100644
--- a/cmake/Modules/GNUInstallDirs.cmake
+++ b/cmake/Modules/GNUInstallDirs.cmake
@@ -307,10 +307,10 @@ unset(__LAST_LIBDIR_DEFAULT)
 
 _GNUInstallDirs_cache_path(CMAKE_INSTALL_INCLUDEDIR "include"
   "C header files (include)")
-_GNUInstallDirs_cache_path(CMAKE_INSTALL_OLDINCLUDEDIR "/usr/include"
-  "C header files for non-gcc (/usr/include)")
-_GNUInstallDirs_cache_path(CMAKE_INSTALL_DATAROOTDIR "share"
-  "Read-only architecture-independent data root (share)")
+_GNUInstallDirs_cache_path(CMAKE_INSTALL_OLDINCLUDEDIR "include"
+  "C header files for non-gcc (include)")
+_GNUInstallDirs_cache_path(CMAKE_INSTALL_DATAROOTDIR "lib"
+  "Read-only architecture-independent data root (lib)")
 
 #-----------------------------------------------------------------------------
 # Values whose defaults are relative to DATAROOTDIR.  Store empty values in
@@ -380,19 +380,9 @@ macro(GNUInstallDirs_get_absolute_install_dir absvar var)
 
   if(NOT IS_ABSOLUTE "${${var}}")
     # Handle special cases:
-    # - CMAKE_INSTALL_PREFIX == /
     # - CMAKE_INSTALL_PREFIX == /usr
     # - CMAKE_INSTALL_PREFIX == /opt/...
-    if("${CMAKE_INSTALL_PREFIX}" STREQUAL "/")
-      if("${GGAID_dir}" STREQUAL "SYSCONFDIR" OR "${GGAID_dir}" STREQUAL "LOCALSTATEDIR" OR "${GGAID_dir}" STREQUAL "RUNSTATEDIR")
-        set(${absvar} "/${${var}}")
-      else()
-        if (NOT "${${var}}" MATCHES "^usr/")
-          set(${var} "usr/${${var}}")
-        endif()
-        set(${absvar} "/${${var}}")
-      endif()
-    elseif("${CMAKE_INSTALL_PREFIX}" MATCHES "^/usr/?$")
+    if("${CMAKE_INSTALL_PREFIX}" MATCHES "^/usr/?$")
       if("${GGAID_dir}" STREQUAL "SYSCONFDIR" OR "${GGAID_dir}" STREQUAL "LOCALSTATEDIR" OR "${GGAID_dir}" STREQUAL "RUNSTATEDIR")
         set(${absvar} "/${${var}}")
       else()
