diff --git a/cmake/Source/cmSystemTools.cxx b/cmake/Source/cmSystemTools.cxx
index 75a5a8d87..6002eaaad 100644
--- a/cmake/Source/cmSystemTools.cxx
+++ b/cmake/Source/cmSystemTools.cxx
@@ -2869,7 +2869,7 @@ bool cmSystemTools::ChangeRPath(std::string const& file,
         file, oldRPath, newRPath, removeEnvironmentRPath, emsg, changed)) {
     return result.value();
   }
-  return false;
+  return newRPath.empty();
 }
 
 bool cmSystemTools::SetRPath(std::string const& file,
@@ -2883,7 +2883,7 @@ bool cmSystemTools::SetRPath(std::string const& file,
         SetRPathXCOFF(file, newRPath, emsg, changed)) {
     return result.value();
   }
-  return false;
+  return newRPath.empty();
 }
 
 namespace {
@@ -3212,7 +3212,7 @@ bool cmSystemTools::RemoveRPath(std::string const& file, std::string* emsg,
   if (cm::optional<bool> result = RemoveRPathXCOFF(file, emsg, removed)) {
     return result.value();
   }
-  return false;
+  return true;
 }
 
 bool cmSystemTools::CheckRPath(std::string const& file,
