diff --git a/lldb/source/Host/common/Editline.cpp b/lldb/source/Host/common/Editline.cpp
index 39b0a649..6942769f 100644
--- a/lldb/source/Host/common/Editline.cpp
+++ b/lldb/source/Host/common/Editline.cpp
@@ -214,14 +214,10 @@ private:
     if (m_path.empty() && m_history && !m_prefix.empty()) {
       FileSpec lldb_dir = HostInfo::GetUserLLDBDir();
 
-      // LLDB stores its history in ~/.lldb/. If for some reason this directory
-      // isn't writable or cannot be created, history won't be available.
-      if (!llvm::sys::fs::create_directory(lldb_dir.GetPath())) {
-#if LLDB_EDITLINE_USE_WCHAR
-        std::string filename = m_prefix + "-widehistory";
-#else
+      // LLDB stores its history in ~/.local/state/lldb/. If this directory
+      // doesn't already exist, assume we shouldn't persist history.
+      if (llvm::sys::fs::is_directory(lldb_dir.GetPath())) {
         std::string filename = m_prefix + "-history";
-#endif
         FileSpec lldb_history_file =
             lldb_dir.CopyByAppendingPathComponent(filename);
         m_path = lldb_history_file.GetPath();
@@ -1273,10 +1269,7 @@ void Editline::ConfigureEditor(bool multiline) {
   ApplyTerminalSizeChange();
 
   if (m_history_sp && m_history_sp->IsValid()) {
-    if (!m_history_sp->Load()) {
-      fputs("Could not load history file\n.",
-            locked_output_stream.GetFile().GetStream());
-    }
+    m_history_sp->Load();
     el_wset(m_editline, EL_HIST, history, m_history_sp->GetHistoryPtr());
   }
   el_set(m_editline, EL_CLIENTDATA, this);
diff --git a/lldb/source/Interpreter/CommandInterpreter.cpp b/lldb/source/Interpreter/CommandInterpreter.cpp
index 7be7ce00..1fe5e5d8 100644
--- a/lldb/source/Interpreter/CommandInterpreter.cpp
+++ b/lldb/source/Interpreter/CommandInterpreter.cpp
@@ -2592,7 +2592,7 @@ int CommandInterpreter::GetOptionArgumentPosition(const char *in_string) {
 }
 
 static void GetHomeInitFile(FileSpec &init_file, llvm::StringRef suffix = {}) {
-  std::string init_file_name = ".lldbinit";
+  std::string init_file_name = ".config/lldb/init";
   if (!suffix.empty()) {
     init_file_name.append("-");
     init_file_name.append(suffix.str());
@@ -2612,7 +2612,7 @@ static void GetHomeREPLInitFile(FileSpec &init_file, LanguageType language) {
   }
 
   std::string init_file_name =
-      (llvm::Twine(".lldbinit-") +
+      (llvm::Twine(".config/lldb/init-") +
        llvm::Twine(Language::GetNameForLanguageType(language)) +
        llvm::Twine("-repl"))
           .str();
