diff --git a/clang/tools/clang-format/CMakeLists.txt b/clang/tools/clang-format/CMakeLists.txt
index 35ecdb11253c..66d4ed60c2b5 100644
--- a/clang/tools/clang-format/CMakeLists.txt
+++ b/clang/tools/clang-format/CMakeLists.txt
@@ -19,22 +19,3 @@ clang_target_link_libraries(clang-format
 if( LLVM_LIB_FUZZING_ENGINE OR LLVM_USE_SANITIZE_COVERAGE )
   add_subdirectory(fuzzer)
 endif()
-
-install(PROGRAMS clang-format-bbedit.applescript
-  DESTINATION share/clang
-  COMPONENT clang-format)
-install(PROGRAMS clang-format-diff.py
-  DESTINATION share/clang
-  COMPONENT clang-format)
-install(PROGRAMS clang-format-sublime.py
-  DESTINATION share/clang
-  COMPONENT clang-format)
-install(PROGRAMS clang-format.el
-  DESTINATION share/clang
-  COMPONENT clang-format)
-install(PROGRAMS clang-format.py
-  DESTINATION share/clang
-  COMPONENT clang-format)
-install(PROGRAMS git-clang-format
-  DESTINATION bin
-  COMPONENT clang-format)
diff --git a/clang/tools/clang-rename/CMakeLists.txt b/clang/tools/clang-rename/CMakeLists.txt
index cda8e29ec5b1..e448cf879fe0 100644
--- a/clang/tools/clang-rename/CMakeLists.txt
+++ b/clang/tools/clang-rename/CMakeLists.txt
@@ -17,10 +17,3 @@ clang_target_link_libraries(clang-rename
   clangToolingCore
   clangToolingRefactoring
   )
-
-install(PROGRAMS clang-rename.py
-  DESTINATION share/clang
-  COMPONENT clang-rename)
-install(PROGRAMS clang-rename.el
-  DESTINATION share/clang
-  COMPONENT clang-rename)
diff --git a/lldb/tools/CMakeLists.txt b/lldb/tools/CMakeLists.txt
index 1585fd4dc..020e3f26e 100644
--- a/lldb/tools/CMakeLists.txt
+++ b/lldb/tools/CMakeLists.txt
@@ -8,7 +8,6 @@ add_subdirectory(intel-features)
 add_subdirectory(lldb-test EXCLUDE_FROM_ALL)
 
 add_lldb_tool_subdirectory(lldb-instr)
-add_lldb_tool_subdirectory(lldb-vscode)
 
 if (CMAKE_SYSTEM_NAME MATCHES "Darwin")
   add_lldb_tool_subdirectory(darwin-debug)
diff --git a/llvm/tools/opt-viewer/CMakeLists.txt b/llvm/tools/opt-viewer/CMakeLists.txt
index ead73ec13a8f..8b137891791f 100644
--- a/llvm/tools/opt-viewer/CMakeLists.txt
+++ b/llvm/tools/opt-viewer/CMakeLists.txt
@@ -1,20 +1 @@
-set (files
-  "opt-diff.py"
-  "opt-stats.py"
-  "opt-viewer.py"
-  "optpmap.py"
-  "optrecord.py"
-  "style.css")
 
-foreach (file ${files})
-  install(PROGRAMS ${file}
-    DESTINATION share/opt-viewer
-    COMPONENT opt-viewer)
-endforeach (file)
-
-add_custom_target(opt-viewer DEPENDS ${files})
-if(NOT LLVM_ENABLE_IDE)
-  add_llvm_install_targets("install-opt-viewer"
-                           DEPENDS opt-viewer
-                           COMPONENT opt-viewer)
-endif()
