diff --git a/clang/tools/CMakeLists.txt b/clang/tools/CMakeLists.txt
index f60db6ef0..65218e27a 100644
--- a/clang/tools/CMakeLists.txt
+++ b/clang/tools/CMakeLists.txt
@@ -34,9 +34,6 @@ endif()
 if(CLANG_ENABLE_STATIC_ANALYZER)
   add_clang_subdirectory(clang-check)
   add_clang_subdirectory(clang-extdef-mapping)
-  add_clang_subdirectory(scan-build)
-  add_clang_subdirectory(scan-build-py)
-  add_clang_subdirectory(scan-view)
 endif()
 
 # We support checking out the clang-tools-extra repository into the 'extra'
diff --git a/llvm/cmake/modules/AddLLVM.cmake b/llvm/cmake/modules/AddLLVM.cmake
index 3bc78b0dc..703b9dc77 100644
--- a/llvm/cmake/modules/AddLLVM.cmake
+++ b/llvm/cmake/modules/AddLLVM.cmake
@@ -1311,6 +1311,7 @@ endfunction()
 if(NOT LLVM_TOOLCHAIN_TOOLS)
   set (LLVM_TOOLCHAIN_TOOLS
     llvm-ar
+    llvm-config
     llvm-cov
     llvm-cxxfilt
     llvm-dlltool
@@ -1324,6 +1325,7 @@ if(NOT LLVM_TOOLCHAIN_TOOLS)
     llvm-objdump
     llvm-pdbutil
     llvm-rc
+    llvm-readelf
     llvm-readobj
     llvm-size
     llvm-strings
