diff --git a/clang/tools/CMakeLists.txt b/clang/tools/CMakeLists.txt
index b071a776b..03c122549 100644
--- a/clang/tools/CMakeLists.txt
+++ b/clang/tools/CMakeLists.txt
@@ -33,9 +33,9 @@ endif()
 if(CLANG_ENABLE_STATIC_ANALYZER)
   add_clang_subdirectory(clang-check)
   add_clang_subdirectory(clang-extdef-mapping)
-  add_clang_subdirectory(scan-build)
-  add_clang_subdirectory(scan-build-py)
-  add_clang_subdirectory(scan-view)
+  # add_clang_subdirectory(scan-build)
+  # add_clang_subdirectory(scan-build-py)
+  # add_clang_subdirectory(scan-view)
 endif()
 
 # We support checking out the clang-tools-extra repository into the 'extra'
diff --git a/llvm/cmake/modules/AddLLVM.cmake b/llvm/cmake/modules/AddLLVM.cmake
index 37bc98f9e..e75429784 100644
--- a/llvm/cmake/modules/AddLLVM.cmake
+++ b/llvm/cmake/modules/AddLLVM.cmake
@@ -1208,6 +1208,7 @@ endfunction()
 if(NOT LLVM_TOOLCHAIN_TOOLS)
   set (LLVM_TOOLCHAIN_TOOLS
     llvm-ar
+    llvm-as
     llvm-cov
     llvm-cxxfilt
     llvm-dwp
@@ -1229,6 +1230,7 @@ if(NOT LLVM_TOOLCHAIN_TOOLS)
     # LLVM_INSTALL_BINUTILS_SYMLINKS.
     addr2line
     ar
+    as
     c++filt
     ranlib
     nm
diff --git a/llvm/tools/llvm-as/CMakeLists.txt b/llvm/tools/llvm-as/CMakeLists.txt
index 3a157a3d4..d9917d9e1 100644
--- a/llvm/tools/llvm-as/CMakeLists.txt
+++ b/llvm/tools/llvm-as/CMakeLists.txt
@@ -11,3 +11,7 @@ add_llvm_tool(llvm-as
   DEPENDS
   intrinsics_gen
   )
+
+if(LLVM_INSTALL_BINUTILS_SYMLINKS)
+  add_llvm_tool_symlink(as llvm-as)
+endif()
