diff --git a/src/radv.c b/src/radv.c
index 5820f4a..16737c4 100644
--- a/src/radv.c
+++ b/src/radv.c
@@ -402,8 +402,8 @@ static void send_ra_alias(time_t now, int iface, char *iface_name, struct in6_ad
 	up = &context->next;
     }
     
-  /* If we're advertising only old prefixes, set router lifetime to zero. */
-  if (old_prefix && !parm.found_context)
+  /* If we're advertising only local or old prefixes, set router lifetime to zero. */
+  if ((old_prefix && !parm.found_context) || parm.glob_pref_time == 0)
     ra->lifetime = htons(0);
 
   /* No prefixes to advertise. */
