diff --git a/src/ntpd.h b/src/ntpd.h
index 39b8e7b..124109b 100644
--- a/src/ntpd.h
+++ b/src/ntpd.h
@@ -36,7 +36,7 @@
 #define MAXIMUM(a, b)	((a) > (b) ? (a) : (b))
 
 #ifndef NTPD_USER
-#define	NTPD_USER	"_ntp"
+#define	NTPD_USER	"ntp"
 #endif
 
 #ifndef	SYSCONFDIR
@@ -47,8 +47,8 @@
 #ifndef	LOCALSTATEDIR
 #define	LOCALSTATEDIR	"/var"
 #endif
-#define	DRIFTFILE	LOCALSTATEDIR "/db/ntpd.drift"
-#define	CTLSOCKET	LOCALSTATEDIR "/run/ntpd.sock"
+#define	DRIFTFILE	LOCALSTATEDIR "/lib/ntpd.drift"
+#define	CTLSOCKET	"/run/ntpd.sock"
 
 #if defined(SO_SETFIB)
 #define	SO_RTABLE	SO_SETFIB
