diff --git a/.bundle/gems/debug-1.4.0/ext/debug/extconf.rb b/.bundle/gems/debug-1.4.0/ext/debug/extconf.rb
index bd26a40..20d2819 100755
--- a/.bundle/gems/debug-1.4.0/ext/debug/extconf.rb
+++ b/.bundle/gems/debug-1.4.0/ext/debug/extconf.rb
@@ -1,4 +1,6 @@
 require 'mkmf'
 require_relative '../../lib/debug/version'
 File.write("debug_version.h", "#define RUBY_DEBUG_VERSION \"#{DEBUGGER__::VERSION}\"\n")
+$DLDFLAGS << " -L$(top_srcdir)"
+$arch_hdrdir = "/src/.ext/include/$(arch)"
 create_makefile 'debug/debug'
diff --git a/.bundle/gems/rbs-2.1.0/ext/rbs_extension/extconf.rb b/.bundle/gems/rbs-2.1.0/ext/rbs_extension/extconf.rb
index ef68154..03557a5 100644
--- a/.bundle/gems/rbs-2.1.0/ext/rbs_extension/extconf.rb
+++ b/.bundle/gems/rbs-2.1.0/ext/rbs_extension/extconf.rb
@@ -1,3 +1,5 @@
 require 'mkmf'
 $INCFLAGS << " -I$(top_srcdir)" if $extmk
+$DLDFLAGS << " -L$(top_srcdir)"
+$arch_hdrdir = "/src/.ext/include/$(arch)"
 create_makefile 'rbs_extension'
