diff --git a/lib/irb/init.rb b/lib/irb/init.rb
index 66e7b61..1758c6f 100644
--- a/lib/irb/init.rb
+++ b/lib/irb/init.rb
@@ -89,7 +89,7 @@ module IRB # :nodoc:
     @CONF[:VERBOSE] = nil
 
     @CONF[:EVAL_HISTORY] = nil
-    @CONF[:SAVE_HISTORY] = 1000
+    @CONF[:SAVE_HISTORY] = nil
 
     @CONF[:BACK_TRACE_LIMIT] = 16
 
@@ -434,8 +434,8 @@ module IRB # :nodoc:
       end
     end
     if home = ENV["HOME"]
-      yield proc{|rc| home+"/.irb#{rc}"}
       yield proc{|rc| home+"/.config/irb/irb#{rc}"}
+      yield proc{|rc| home+"/.irb#{rc}"}
     end
     current_dir = Dir.pwd
     yield proc{|rc| current_dir+"/.irb#{rc}"}
