diff --git a/ruby.c b/ruby.c
index b00fc150..7011475c 100644
--- a/ruby.c
+++ b/ruby.c
@@ -2073,7 +2073,7 @@ show_help(const char *progname, int help)
                 tty = tty_enabled();
                 dup2(fd, 1);
                 dup2(fd, 2);
-                usage(progname, 1, tty, columns);
+                usage(progname, 1, 0, columns);
                 fflush(stdout);
                 dup2(oldout, 1);
                 dup2(olderr, 2);
@@ -2083,7 +2083,7 @@ show_help(const char *progname, int help)
 #endif
         }
     }
-    usage(progname, help, tty, columns);
+    usage(progname, help, 0, columns);
 }
 
 static VALUE
