diff --git a/src/ntpd.c b/src/ntpd.c
index a214b3f..49f8f3a 100644
--- a/src/ntpd.c
+++ b/src/ntpd.c
@@ -528,21 +528,11 @@ ntpd_adjtime(double d)
 		log_debug("adjusting local clock by %fs", d);
 
 #ifdef HAVE_ADJTIMEX
-	int rc;
-
-	long offset = d * 1000000;
 	struct timex tx = { 0 };
-	tx.offset = offset;
-	tx.modes = ADJ_OFFSET_SINGLESHOT | ADJ_STATUS;
+	tx.offset = d * 1000000;
+	tx.modes = ADJ_OFFSET_SINGLESHOT;
 
-	do {
-		rc = adjtimex(&tx);
-	} while (rc == TIME_ERROR && (tx.offset /= 2) > threshold);
-
-	if (rc == TIME_ERROR) {
-		if ((tx.status & ~STA_UNSYNC))
-			log_warn("adjtimex returned TIME_ERROR");
-	} else if (rc < 0) {
+	if (adjtimex(&tx) < 0) {
 		log_warn("adjtimex failed");
 	} else if (!firstadj && tx.offset == 0) {
 		synced = 1;
