diff --git a/src/ntpd.c b/src/ntpd.c
index bb3cc7c..00f7846 100644
--- a/src/ntpd.c
+++ b/src/ntpd.c
@@ -285,7 +285,7 @@ main(int argc, char *argv[])
 	if (!lconf.settime) {
 		log_init(logdest, lconf.verbose, LOG_DAEMON);
 		if (!lconf.debug) {
-			if (daemon(1, 0))
+			if (getsid(0) != getpid() && daemon(1, 0))
 				fatal("daemon");
 			writepid(&lconf);
 		}
@@ -371,7 +371,7 @@ main(int argc, char *argv[])
 			log_warnx("no reply received in time, skipping initial "
 			    "time setting");
 			if (!lconf.debug) {
-				if (daemon(1, 0))
+				if (getsid(0) != getpid() && daemon(1, 0))
 					fatal("daemon");
 				writepid(&lconf);
 			}
@@ -476,7 +476,7 @@ dispatch_imsg(struct ntpd_conf *lconf, int argc, char **argv)
 			ntpd_settime(d);
 			/* daemonize now */
 			if (!lconf->debug) {
-				if (daemon(1, 0))
+				if (getsid(0) != getpid() && daemon(1, 0))
 					fatal("daemon");
 				writepid(lconf);
 			}
