diff --git a/src/ntp_dns.c b/src/ntp_dns.c
index e74a8ab..353105d 100644
--- a/src/ntp_dns.c
+++ b/src/ntp_dns.c
@@ -35,6 +35,7 @@
 #include <syslog.h>
 #include <resolv.h>
 #include <unistd.h>
+#include <time.h>
 
 #include "ntpd.h"
 
@@ -242,14 +243,18 @@ probe_root_ns(void)
 void
 probe_root(void)
 {
-	int		n;
+	time_t start = time(NULL), before = start, after;
+	int n = probe_root_ns();
 
-	n = probe_root_ns();
-	if (n < 0) {
-		/* give programs like unwind a second chance */
-		sleep(1);
-		n = probe_root_ns();
+	while (n < 0) {
+		after = time(NULL);
+		if (after >= start + SETTIME_TIMEOUT)
+			break;
+		if (after <= before + 1)
+			sleep(1);
+		before = after, n = probe_root_ns();
 	}
+
 	if (imsg_compose(ibuf_dns, IMSG_PROBE_ROOT, 0, 0, -1, &n,
 	    sizeof(int)) == -1)
 		fatalx("probe_root");
