diff --git a/src/ntpd.c b/src/ntpd.c
index b28dfd4..fd77895 100644
--- a/src/ntpd.c
+++ b/src/ntpd.c
@@ -325,8 +325,8 @@ main(int argc, char *argv[])
 	 * Constraint processes are forked with certificates in memory,
 	 * then privdrop into chroot before speaking to the outside world.
 	 */
-	if (unveil("/usr/sbin/ntpd", "x") == -1)
-		err(1, "unveil /usr/sbin/ntpd");
+	if (unveil("/bin/ntpd", "x") == -1)
+		err(1, "unveil /bin/ntpd");
 	if (pledge("stdio settime proc exec", NULL) == -1)
 		err(1, "pledge");
 
diff --git a/src/ntpd.h b/src/ntpd.h
index 31025fd..8542076 100644
--- a/src/ntpd.h
+++ b/src/ntpd.h
@@ -37,7 +37,7 @@
 #define MAXIMUM(a, b)	((a) > (b) ? (a) : (b))
 
 #ifndef NTPD_USER
-#define	NTPD_USER	"_ntp"
+#define	NTPD_USER	"ntp"
 #endif
 
 #ifndef	SYSCONFDIR
@@ -48,8 +48,8 @@
 #ifndef	LOCALSTATEDIR
 #define	LOCALSTATEDIR	"/var"
 #endif
-#define	DRIFTFILE	LOCALSTATEDIR "/db/ntpd.drift"
-#define	CTLSOCKET	LOCALSTATEDIR "/run/ntpd.sock"
+#define	DRIFTFILE	LOCALSTATEDIR "/lib/ntpd.drift"
+#define	CTLSOCKET	"/run/ntpd.sock"
 
 #define	INTERVAL_QUERY_NORMAL		30	/* sync to peers every n secs */
 #define	INTERVAL_QUERY_PATHETIC		60
