diff --git a/src/client.c b/src/client.c
index 9522028..8bf768f 100644
--- a/src/client.c
+++ b/src/client.c
@@ -253,9 +253,9 @@ handle_auto(uint8_t trusted, double offset)
 	if (!trusted && conf->constraint_median == 0)
 		return;
 
-	if (offset < AUTO_THRESHOLD) {
+	if (-AUTO_THRESHOLD < offset && offset < AUTO_THRESHOLD) {
 		/* don't bother */
-		priv_settime(0, "offset is negative or close enough");
+		priv_settime(0, "offset is too small");
 		return;
 	}
 	/* collect some more */
diff --git a/src/ntp.c b/src/ntp.c
index a3914e4..2d1d0be 100644
--- a/src/ntp.c
+++ b/src/ntp.c
@@ -849,8 +849,8 @@ offset_compare(const void *aa, const void *bb)
 void
 priv_settime(double offset, char *msg)
 {
-	if (offset == 0)
-		log_info("cancel settime because %s", msg);
+	if (offset == 0 && msg && msg[0])
+		log_warnx("not setting time because %s", msg);
 	imsg_compose(ibuf_main, IMSG_SETTIME, 0, 0, -1,
 	    &offset, sizeof(offset));
 	conf->settime = 0;
diff --git a/src/ntpd.c b/src/ntpd.c
index e7510b6..b75a79f 100644
--- a/src/ntpd.c
+++ b/src/ntpd.c
@@ -371,8 +371,7 @@ main(int argc, char *argv[])
 			lconf.settime = 0;
 			timeout = INFTIM;
 			log_init(logdest, lconf.verbose, LOG_DAEMON);
-			log_warnx("no reply received in time, skipping initial "
-			    "time setting");
+			log_warnx("not setting time because no reply received in time");
 			if (!lconf.debug) {
 				if (daemon(1, 0))
 					fatal("daemon");
diff --git a/src/ntpd.h b/src/ntpd.h
index 8542076..237681d 100644
--- a/src/ntpd.h
+++ b/src/ntpd.h
@@ -69,7 +69,7 @@
 #define	QUERYTIME_MAX		15	/* single query might take n secs max */
 #define	OFFSET_ARRAY_SIZE	8
 #define	SENSOR_OFFSETS		6
-#define	SETTIME_TIMEOUT		15	/* max seconds to wait with -s */
+#define	SETTIME_TIMEOUT		30	/* max seconds to wait with -s */
 #define	LOG_NEGLIGIBLE_ADJTIME	32	/* negligible drift to not log (ms) */
 #define	LOG_NEGLIGIBLE_ADJFREQ	0.05	/* negligible rate to not log (ppm) */
 #define	FREQUENCY_SAMPLES	8	/* samples for est. of permanent drift */
@@ -79,7 +79,7 @@
 
 #define FILTER_ADJFREQ		0x01	/* set after doing adjfreq */
 #define AUTO_REPLIES    	4	/* # of ntp replies we want for auto */
-#define AUTO_THRESHOLD		60	/* dont bother auto setting < this */
+#define AUTO_THRESHOLD		1	/* dont bother auto setting < this */
 #define INTERVAL_AUIO_DNSFAIL	1	/* DNS tmpfail interval for auto */
 #define TRIES_AUTO_DNSFAIL	4	/* DNS tmpfail quick retries */
 
