diff --git a/src/client.c b/src/client.c
index 6fc60af..3e3865b 100644
--- a/src/client.c
+++ b/src/client.c
@@ -254,9 +254,9 @@ handle_auto(uint8_t trusted, double offset)
 	if (!trusted && conf->constraint_median == 0)
 		return;
 
-	if (offset < AUTO_THRESHOLD) {
+	if (-AUTO_THRESHOLD < offset && offset < AUTO_THRESHOLD) {
 		/* don't bother */
-		priv_settime(0, "offset is negative or close enough");
+		priv_settime(0, "offset is close enough");
 		return;
 	}
 	/* collect some more */
diff --git a/src/ntpd.h b/src/ntpd.h
index 3872efc..b9ec58d 100644
--- a/src/ntpd.h
+++ b/src/ntpd.h
@@ -79,7 +79,7 @@
 
 #define FILTER_ADJFREQ		0x01	/* set after doing adjfreq */
 #define AUTO_REPLIES    	4	/* # of ntp replies we want for auto */
-#define AUTO_THRESHOLD		60	/* dont bother auto setting < this */
+#define AUTO_THRESHOLD		0.128	/* dont bother auto setting < this */
 #define INTERVAL_AUIO_DNSFAIL	1	/* DNS tmpfail interval for auto */
 #define TRIES_AUTO_DNSFAIL	4	/* DNS tmpfail quick retries */
 
