diff --git a/update.sh b/update.sh
index 811606a..8bb9ea5 100755
--- a/update.sh
+++ b/update.sh
@@ -1,23 +1,6 @@
 #!/bin/sh
 set -e
 
-openbsd_branch=`cat OPENBSD_BRANCH`
-openntpd_version=`cat VERSION`
-
-# pull in latest upstream code
-echo "pulling upstream openbsd source"
-if [ ! -d openbsd ]; then
-	if [ -z "${OPENNTPD_GIT}" ]; then
-		git clone https://github.com/openntpd-portable/openntpd-openbsd.git openbsd
-	else
-		git clone "${OPENNTPD_GIT}/openbsd"
-	fi
-fi
-(cd openbsd
- git fetch
- git checkout "${openbsd_branch}"
- git pull --rebase)
-
 # setup source paths
 dir=`pwd`
 patches="${dir}/patches"
