diff --git a/crates/typst-cli/src/args.rs b/crates/typst-cli/src/args.rs
index ee3ae7e3..a9b68f96 100644
--- a/crates/typst-cli/src/args.rs
+++ b/crates/typst-cli/src/args.rs
@@ -25,8 +25,6 @@ const ENV_PATH_SEP: char = if cfg!(windows) { ';' } else { ':' };
 /// The overall structure of the help.
 #[rustfmt::skip]
 const HELP_TEMPLATE: &str = "\
-Typst {version}
-
 {usage-heading} {usage}
 
 {all-args}{after-help}\
@@ -34,6 +32,7 @@ Typst {version}
 
 /// Adds a list of useful links after the normal help.
 #[rustfmt::skip]
+#[expect(unused)]
 const AFTER_HELP: &str = color_print::cstr!("\
 <s><u>Resources:</></>
   <s>Tutorial:</>                 https://typst.app/docs/tutorial/
@@ -51,14 +50,13 @@ const STYLES: Styles = Styles::styled()
 #[clap(
     name = "typst",
     version = format!(
-        "{} ({})",
+        "{}",
         typst_utils::version().raw(),
-        typst_utils::display_commit(typst_utils::version().commit()),
     ),
     author,
     help_template = HELP_TEMPLATE,
-    after_help = AFTER_HELP,
     max_term_width = 80,
+    color = ColorChoice::Never,
     styles = STYLES,
 )]
 pub struct CliArguments {
diff --git a/crates/typst-cli/src/main.rs b/crates/typst-cli/src/main.rs
index 6e47b1c8..30781eaa 100644
--- a/crates/typst-cli/src/main.rs
+++ b/crates/typst-cli/src/main.rs
@@ -5,7 +5,6 @@ mod deps;
 mod download;
 mod eval;
 mod fonts;
-mod greet;
 mod info;
 mod init;
 mod packages;
@@ -22,7 +21,6 @@ use std::process::ExitCode;
 use std::sync::LazyLock;
 
 use clap::Parser;
-use clap::error::ErrorKind;
 use codespan_reporting::term;
 use codespan_reporting::term::termcolor::WriteColor;
 use ecow::eco_format;
@@ -39,9 +37,6 @@ thread_local! {
 /// The parsed command line arguments.
 static ARGS: LazyLock<CliArguments> = LazyLock::new(|| {
     CliArguments::try_parse().unwrap_or_else(|error| {
-        if error.kind() == ErrorKind::DisplayHelpOnMissingArgumentOrSubcommand {
-            crate::greet::greet();
-        }
         error.exit();
     })
 });
