diff --git a/configure b/configure
index 57bec69..6a42c8b 100755
--- a/configure
+++ b/configure
@@ -1770,7 +1770,7 @@ Optional Packages:
   --with-dmeventd-pidfile=PATH
                           dmeventd pidfile [PID_DIR/dmeventd.pid]
   --with-dmeventd-path=PATH
-                          dmeventd path [EPREFIX/sbin/dmeventd]
+                          dmeventd path [EPREFIX/bin/dmeventd]
   --with-default-system-dir=DIR
                           default LVM system directory [/etc/lvm]
   --with-default-profile-subdir=SUBDIR
@@ -13863,7 +13863,7 @@ fi
 lvm_exec_prefix=$exec_prefix
 test "$lvm_exec_prefix" = NONE && lvm_exec_prefix=$prefix
 test "$lvm_exec_prefix" = NONE && lvm_exec_prefix=$ac_default_prefix
-LVM_PATH="$lvm_exec_prefix/sbin/lvm"
+LVM_PATH="$lvm_exec_prefix/bin/lvm"
 
 cat >>confdefs.h <<_ACEOF
 #define LVM_PATH "$LVM_PATH"
@@ -13873,7 +13873,7 @@ _ACEOF
 if  test "$CLVMD" != none; then
 	clvmd_prefix=$ac_default_prefix
 	test "$prefix" != NONE && clvmd_prefix=$prefix
-	CLVMD_PATH="$clvmd_prefix/sbin/clvmd"
+	CLVMD_PATH="$clvmd_prefix/bin/clvmd"
 
 cat >>confdefs.h <<_ACEOF
 #define CLVMD_PATH "$CLVMD_PATH"
@@ -13904,7 +13904,7 @@ if test "$BUILD_DMEVENTD" = yes; then
 if test "${with_dmeventd_path+set}" = set; then :
   withval=$with_dmeventd_path; DMEVENTD_PATH=$withval
 else
-  DMEVENTD_PATH="$lvm_exec_prefix/sbin/dmeventd"
+  DMEVENTD_PATH="$lvm_exec_prefix/bin/dmeventd"
 fi
 
 
diff --git a/configure.in b/configure.in
index 2c89d8c..3993b9b 100644
--- a/configure.in
+++ b/configure.in
@@ -1785,13 +1785,13 @@ fi
 lvm_exec_prefix=$exec_prefix
 test "$lvm_exec_prefix" = NONE && lvm_exec_prefix=$prefix
 test "$lvm_exec_prefix" = NONE && lvm_exec_prefix=$ac_default_prefix
-LVM_PATH="$lvm_exec_prefix/sbin/lvm"
+LVM_PATH="$lvm_exec_prefix/bin/lvm"
 AC_DEFINE_UNQUOTED(LVM_PATH, ["$LVM_PATH"], [Path to lvm binary.])
 
 if  test "$CLVMD" != none; then
 	clvmd_prefix=$ac_default_prefix
 	test "$prefix" != NONE && clvmd_prefix=$prefix
-	CLVMD_PATH="$clvmd_prefix/sbin/clvmd"
+	CLVMD_PATH="$clvmd_prefix/bin/clvmd"
 	AC_DEFINE_UNQUOTED(CLVMD_PATH, ["$CLVMD_PATH"], [Path to clvmd binary.])
 fi
 
@@ -1810,9 +1810,9 @@ fi
 if test "$BUILD_DMEVENTD" = yes; then
 	AC_ARG_WITH(dmeventd-path,
 		    AC_HELP_STRING([--with-dmeventd-path=PATH],
-				   [dmeventd path [EPREFIX/sbin/dmeventd]]),
+				   [dmeventd path [EPREFIX/bin/dmeventd]]),
 		    DMEVENTD_PATH=$withval,
-		    DMEVENTD_PATH="$lvm_exec_prefix/sbin/dmeventd")
+		    DMEVENTD_PATH="$lvm_exec_prefix/bin/dmeventd")
 	AC_DEFINE_UNQUOTED(DMEVENTD_PATH, ["$DMEVENTD_PATH"],
 			   [Path to dmeventd binary.])
 fi
diff --git a/scripts/fsadm.sh b/scripts/fsadm.sh
index f4ea796..2e807bc 100755
--- a/scripts/fsadm.sh
+++ b/scripts/fsadm.sh
@@ -32,7 +32,7 @@
 TOOL=fsadm
 
 _SAVEPATH=$PATH
-PATH=/sbin:/usr/sbin:/bin:/usr/sbin:$PATH
+PATH=/bin:$PATH
 
 # utilities
 TUNE_EXT=tune2fs
