diff --git a/library/readproc.c b/library/readproc.c
index 4bc9d96..cefad52 100644
--- a/library/readproc.c
+++ b/library/readproc.c
@@ -33,6 +33,7 @@
 #include <sys/stat.h>
 #include <limits.h>
 #include <stdint.h>
+#include <stddef.h>
 #ifdef WITH_SYSTEMD
 #include <systemd/sd-login.h>
 #endif
diff --git a/src/w.c b/src/w.c
index 8b37179..b0e9e59 100644
--- a/src/w.c
+++ b/src/w.c
@@ -43,11 +43,7 @@
 #include <termios.h>
 #include <time.h>
 #include <unistd.h>
-#ifdef HAVE_UTMPX_H
-#	include <utmpx.h>
-#else
-#	include <utmp.h>
-#endif
+#include <utmp.h>
 #include <arpa/inet.h>
 
 #include "c.h"
diff --git a/src/watch.c b/src/watch.c
index a69a402..389db61 100644
--- a/src/watch.c
+++ b/src/watch.c
@@ -52,10 +52,8 @@
 # define _XOPEN_SOURCE_EXTENDED 1
 # include <wchar.h>
 # include <wctype.h>
-# include <ncursesw/ncurses.h>
-#else
-# include <ncurses.h>
 #endif	/* WITH_WATCH8BIT */
+#include <ncurses.h>
 
 #ifdef FORCE_8BIT
 # undef isprint
