diff --git a/library/readproc.c b/library/readproc.c
index 2dfe4c9..51f049b 100644
--- a/library/readproc.c
+++ b/library/readproc.c
@@ -37,6 +37,7 @@
 #include <sys/stat.h>
 #include <limits.h>
 #include <stdint.h>
+#include <stddef.h>
 #ifdef WITH_SYSTEMD
 #include <systemd/sd-login.h>
 #endif
diff --git a/src/watch.c b/src/watch.c
index 5c159a9..45fc377 100644
--- a/src/watch.c
+++ b/src/watch.c
@@ -53,10 +53,8 @@
 # define _XOPEN_SOURCE_EXTENDED 1
 # include <wchar.h>
 # include <wctype.h>
-# include <ncursesw/ncurses.h>
-#else
-# include <ncurses.h>
 #endif	/* WITH_WATCH8BIT */
+#include <ncurses.h>
 
 #ifdef FORCE_8BIT
 # undef isprint
