diff --git a/library/include/readproc.h b/library/include/readproc.h
index 5fcc2209..eaa3d2be 100644
--- a/library/include/readproc.h
+++ b/library/include/readproc.h
@@ -228,6 +228,7 @@ typedef struct PROCTAB {
     int         nuid;   // cannot really sentinel-terminate unsigned short[]
     int         i;  // generic
     int         hide_kernel;  // getenv LIBPROC_HIDE_KERNEL was set
+    unsigned long filter_pidns;  // restrict to this PID namespace
     unsigned    flags;
 } PROCTAB;
 
diff --git a/library/readproc.c b/library/readproc.c
index a7f79051..0bf73bbc 100644
--- a/library/readproc.c
+++ b/library/readproc.c
@@ -1259,6 +1259,11 @@ static proc_t *simple_readproc(PROCTAB *restrict const PT, proc_t *restrict cons
         goto next_proc;
     }
 
+    if (PT->filter_pidns && p->ns.ns[PROCPS_NS_PID] != PT->filter_pidns) {
+        free_acquired(p);
+        goto next_proc;
+    }
+
     if (flags & PROC_FILLIO) {                  // read /proc/#/io
         if (file2str(PT->pidfd, "io", &ub) != -1)
             io2proc(ub.buf, p);
@@ -1669,12 +1674,15 @@ PROCTAB *openproc(unsigned flags, ...) {
     struct stat sbuf;
     static __thread int did_stat;
     static __thread int hide_kernel = -1;
+    static __thread int hide_pidns = -1;
     PROCTAB *PT = calloc(1, sizeof(PROCTAB));
 
     if (!PT)
         return NULL;
     if (hide_kernel < 0)
-        hide_kernel = (NULL != getenv("LIBPROC_HIDE_KERNEL"));
+        hide_kernel = (*(getenv("LIBPROC_HIDE_KERNEL") ?: "") != 0);
+    if (hide_pidns < 0)
+        hide_pidns = (*(getenv("LIBPROC_HIDE_PIDNS") ?: "") != 0);
     if (!did_stat){
         task_dir_missing = stat("/proc/self/task", &sbuf);
         did_stat = 1;
@@ -1709,6 +1717,14 @@ PROCTAB *openproc(unsigned flags, ...) {
     if (hide_kernel > 0)
         PT->hide_kernel = 1;
 
+    if (hide_pidns > 0) {
+        struct procps_ns self;
+        if (procps_ns_read_pid(getpid(), &self) >= 0) {
+            PT->filter_pidns = self.ns[PROCPS_NS_PID];
+            PT->flags |= PROC_FILLNS;
+        }
+    }
+
     if (!src_buffer
     && !(src_buffer = malloc(MAX_BUFSZ))) {
         if (PT->procfs) closedir(PT->procfs);
