diff --git a/src/uschedulelist.c b/src/uschedulelist.c
index 02398b00..bb290a1e 100644
--- a/src/uschedulelist.c
+++ b/src/uschedulelist.c
@@ -238,7 +238,7 @@ list_schedules(char **filenames, int *flag)
     else
       print_job(&j[i],&now);
     if (o_show_command) {
-      stralloc sa;
+      stralloc sa=STRALLOC_INIT;
       make_name(&sa,&j[i]); 
       show_command(sa.s);
       stralloc_free(&sa);
