diff --git a/util.c b/util.c
index 3d05d07..f396110 100644
--- a/util.c
+++ b/util.c
@@ -2193,6 +2193,7 @@ void manage_fork_fds(int close_all)
  */
 int continue_via_systemd(char *devnm, char *service_name)
 {
+#if 0
 	int pid, status;
 	char pathbuf[1024];
 
@@ -2216,6 +2217,7 @@ int continue_via_systemd(char *devnm, char *service_name)
 		if (pid >= 0 && status == 0)
 			return 1;
 	}
+#endif
 	return 0;
 }
 
