diff --git a/Incremental.c b/Incremental.c
index a57fc32..352c2d3 100644
--- a/Incremental.c
+++ b/Incremental.c
@@ -1615,6 +1615,7 @@ static int Incremental_container(struct supertype *st, char *devname,
 
 static void run_udisks(char *arg1, char *arg2)
 {
+#if 0
 	int pid = fork();
 	int status;
 	if (pid == 0) {
@@ -1625,6 +1626,7 @@ static void run_udisks(char *arg1, char *arg2)
 	}
 	while (pid > 0 && wait(&status) != pid)
 		;
+#endif
 }
 
 static int force_remove(char *devnm, int fd, struct mdinfo *mdi, int verbose)
