diff --git a/Makefile b/Makefile
index 03bf922..179d146 100644
--- a/Makefile
+++ b/Makefile
@@ -50,7 +50,7 @@ ifeq ($(origin CC),default)
 CC := $(CROSS_COMPILE)gcc
 endif
 CXFLAGS ?= -ggdb
-CWFLAGS = -Wall -Werror -Wstrict-prototypes -Wextra -Wno-unused-parameter
+CWFLAGS = -Wall -Werror -Wstrict-prototypes -Wextra -Wno-unused-but-set-variable -Wno-unused-parameter
 ifdef WARN_UNUSED
 CWFLAGS += -Wp,-D_FORTIFY_SOURCE=2 -O3
 endif
diff --git a/Monitor.c b/Monitor.c
index 30c031a..3835dda 100644
--- a/Monitor.c
+++ b/Monitor.c
@@ -378,7 +378,7 @@ static int check_one_sharer(int scan)
 	return 0;
 }
 
-static void write_autorebuild_pid()
+static void write_autorebuild_pid(void)
 {
 	char path[PATH_MAX];
 	int pid;
diff --git a/config.c b/config.c
index 9c72545..e3d8726 100644
--- a/config.c
+++ b/config.c
@@ -970,7 +970,7 @@ static void append_dlist(struct mddev_dev **dlp, struct mddev_dev *list)
 	*dlp = list;
 }
 
-struct mddev_dev *conf_get_devs()
+struct mddev_dev *conf_get_devs(void)
 {
 	glob_t globbuf;
 	struct conf_dev *cd;
diff --git a/dlink.c b/dlink.c
index 69aa7aa..2077eda 100644
--- a/dlink.c
+++ b/dlink.c
@@ -11,7 +11,7 @@ char *strncpy(char *dest, const char *src, size_t n) __THROW;
 void *xcalloc(size_t num, size_t size);
 #include	"dlink.h"
 
-void *dl_head()
+void *dl_head(void)
 {
     void *h;
     h = dl_alloc(0);
diff --git a/util.c b/util.c
index f396110..c16f21a 100644
--- a/util.c
+++ b/util.c
@@ -306,7 +306,7 @@ int md_get_disk_info(int fd, struct mdu_disk_info_s *disk)
 	return ioctl(fd, GET_DISK_INFO, disk);
 }
 
-int get_linux_version()
+int get_linux_version(void)
 {
 	struct utsname name;
 	char *cp;
