diff --git a/ui/curses.c b/ui/curses.c
index a7588ca..1e25629 100644
--- a/ui/curses.c
+++ b/ui/curses.c
@@ -679,7 +679,7 @@ void mtr_curses_redraw(
 
     move(0, 0);
     attron(A_BOLD);
-    snprintf(buf, sizeof(buf), "%s%s%s", "My traceroute  [v",
+    snprintf(buf, sizeof(buf), "%s%s%s", "My Traceroute  [v",
              PACKAGE_VERSION, "]");
     pwcenter(buf);
     attroff(A_BOLD);
@@ -707,7 +707,7 @@ void mtr_curses_redraw(
     attroff(A_BOLD);
     printw("rder of fields   ");
     attron(A_BOLD);
-    printw("q");
+    printw("Q");
     attroff(A_BOLD);
     printw("uit\n");
 
diff --git a/ui/report.c b/ui/report.c
index 7700391..d1fffaa 100644
--- a/ui/report.c
+++ b/ui/report.c
@@ -119,6 +119,8 @@ void report_close(
             if (!ctl->ipinfo_no)
                 len_tmp += 2;   /* align header: AS */
         }
+    } else if (ctl->reportwide) {
+        len_tmp += 2;
     }
     snprintf(fmt, sizeof(fmt), "HOST: %%-%ds", len_tmp);
 #else
