diff --git a/ui/curses.c b/ui/curses.c
index d01e178..233b75d 100644
--- a/ui/curses.c
+++ b/ui/curses.c
@@ -693,8 +693,7 @@ void mtr_curses_redraw(
 
     move(0, 0);
     attron(A_BOLD);
-    snprintf(buf, sizeof(buf), "%s%s%s", "My traceroute  [v",
-             PACKAGE_VERSION, "]");
+    snprintf(buf, sizeof(buf), "My Traceroute %s", PACKAGE_VERSION);
     pwcenter(buf);
     attroff(A_BOLD);
 
@@ -723,7 +722,7 @@ void mtr_curses_redraw(
     attroff(A_BOLD);
     printw("rder of fields   ");
     attron(A_BOLD);
-    printw("q");
+    printw("Q");
     attroff(A_BOLD);
     printw("uit\n");
 
diff --git a/ui/mtr.c b/ui/mtr.c
index 2c6fc68..e3cc182 100644
--- a/ui/mtr.c
+++ b/ui/mtr.c
@@ -67,7 +67,7 @@ char *myname;
 const struct fields data_fields[MAXFLD] = {
     /* key, Remark, Header, Format, Width, CallBackFunc */
     {' ', "<sp>: Space between fields", " ", " ", 1, &net_drop},
-    {'L', "L: Loss Ratio", "Loss%", " %4.1f%%", 6, &net_loss},
+    {'L', "L: Loss Ratio", "Loss%", " %5.1f%%", 7, &net_loss},
     {'D', "D: Dropped Packets", "Drop", " %4d", 5, &net_drop},
     {'R', "R: Received Packets", "Rcv", " %5d", 6, &net_returned},
     {'S', "S: Sent Packets", "Snt", " %5d", 6, &net_xmit},
diff --git a/ui/report.c b/ui/report.c
index a63ed55..38d6a41 100644
--- a/ui/report.c
+++ b/ui/report.c
@@ -127,10 +127,12 @@ void report_close(
             if (!ctl->ipinfo_no)
                 len_tmp += 2;   /* align header: AS */
         }
+    } else if (ctl->reportwide) {
+        len_tmp += 2;
     }
-    snprintf(fmt, sizeof(fmt), "HOST: %%-%ds", len_tmp);
+    snprintf(fmt, sizeof(fmt), "Host: %%-%ds", len_tmp);
 #else
-    snprintf(fmt, sizeof(fmt), "HOST: %%-%zus", len_hosts);
+    snprintf(fmt, sizeof(fmt), "Host: %%-%zus", len_hosts);
 #endif
     snprintf(buf, sizeof(buf), fmt, ctl->LocalHostname);
     len = ctl->reportwide ? strlen(buf) : len_hosts;
