diff --git a/vmm/src/vm_config.rs b/vmm/src/vm_config.rs
index 93304e47..cbb7f7b0 100644
--- a/vmm/src/vm_config.rs
+++ b/vmm/src/vm_config.rs
@@ -164,8 +164,8 @@ impl ApplyLandlock for MemoryZoneConfig {
 
 #[derive(Clone, Copy, Debug, PartialEq, Eq, Deserialize, Serialize, Default)]
 pub enum HotplugMethod {
-    #[default]
     Acpi,
+    #[default]
     VirtioMem,
 }
 
@@ -205,7 +205,7 @@ impl Default for MemoryConfig {
         MemoryConfig {
             size: DEFAULT_MEMORY_MB << 20,
             mergeable: false,
-            hotplug_method: HotplugMethod::Acpi,
+            hotplug_method: HotplugMethod::VirtioMem,
             hotplug_size: None,
             hotplugged_size: None,
             shared: false,
