diff --git a/vmm/src/vm_config.rs b/vmm/src/vm_config.rs
index e74fbc1d..52ace1fd 100644
--- a/vmm/src/vm_config.rs
+++ b/vmm/src/vm_config.rs
@@ -157,8 +157,8 @@ impl ApplyLandlock for MemoryZoneConfig {
 
 #[derive(Clone, Copy, Debug, PartialEq, Eq, Deserialize, Serialize, Default)]
 pub enum HotplugMethod {
-    #[default]
     Acpi,
+    #[default]
     VirtioMem,
 }
 
@@ -198,7 +198,7 @@ impl Default for MemoryConfig {
         MemoryConfig {
             size: DEFAULT_MEMORY_MB << 20,
             mergeable: false,
-            hotplug_method: HotplugMethod::Acpi,
+            hotplug_method: HotplugMethod::VirtioMem,
             hotplug_size: None,
             hotplugged_size: None,
             shared: false,
