diff --git a/vmm/src/vm_config.rs b/vmm/src/vm_config.rs
index 1f323df0..bf257770 100644
--- a/vmm/src/vm_config.rs
+++ b/vmm/src/vm_config.rs
@@ -134,8 +134,8 @@ pub struct MemoryZoneConfig {
 
 #[derive(Clone, Copy, Debug, PartialEq, Eq, Deserialize, Serialize, Default)]
 pub enum HotplugMethod {
-    #[default]
     Acpi,
+    #[default]
     VirtioMem,
 }
 
@@ -175,7 +175,7 @@ impl Default for MemoryConfig {
         MemoryConfig {
             size: DEFAULT_MEMORY_MB << 20,
             mergeable: false,
-            hotplug_method: HotplugMethod::Acpi,
+            hotplug_method: HotplugMethod::VirtioMem,
             hotplug_size: None,
             hotplugged_size: None,
             shared: false,
