diff --git a/vmm/src/vm.rs b/vmm/src/vm.rs
index 1c3baf86..ae58f514 100644
--- a/vmm/src/vm.rs
+++ b/vmm/src/vm.rs
@@ -467,7 +467,8 @@ impl VmOps for VmOpsHandler {
 
     fn mmio_read(&self, gpa: u64, data: &mut [u8]) -> result::Result<(), HypervisorVmError> {
         if let Err(vm_device::BusError::MissingAddressRange) = self.mmio_bus.read(gpa, data) {
-            info!("Guest MMIO read to unregistered address 0x{gpa:x}");
+            info!("Guest MMIO read from unregistered address 0x{gpa:x}");
+            data.fill(0xff);
         }
         Ok(())
     }
@@ -490,7 +491,8 @@ impl VmOps for VmOpsHandler {
     #[cfg(target_arch = "x86_64")]
     fn pio_read(&self, port: u64, data: &mut [u8]) -> result::Result<(), HypervisorVmError> {
         if let Err(vm_device::BusError::MissingAddressRange) = self.io_bus.read(port, data) {
-            info!("Guest PIO read to unregistered address 0x{port:x}");
+            info!("Guest PIO read from unregistered address 0x{port:x}");
+            data.fill(0xff);
         }
         Ok(())
     }
