diff --git a/vmm/src/device_manager.rs b/vmm/src/device_manager.rs
index 416132e6..13ec09d9 100644
--- a/vmm/src/device_manager.rs
+++ b/vmm/src/device_manager.rs
@@ -2750,44 +2750,17 @@ impl DeviceManager {
             .map_err(DeviceManagerError::Disk)?;
 
             let detected_image_type = opened.image_type;
-            let mut disable_sector0_writes = false;
 
             if disk_cfg.image_type == ImageType::Unknown {
-                warn!(
-                    "DEPRECATION: auto-detection of disk image type is deprecated and will be \
-                    removed in a future release. Please specify image type explicitly."
-                );
-
-                warn!(
-                    "No image_type specified - detected as {detected_image_type}. \
-                    Configuration updated to persist type across reboots and migrations."
-                );
-
-                if detected_image_type == ImageType::Raw
-                    || detected_image_type == ImageType::FixedVhd
-                {
-                    warn!(
-                        "Autodetected {detected_image_type} image type. Disabling sector 0 writes."
-                    );
-                    disable_sector0_writes = true;
-                } else {
-                    warn!(
-                        "Non-raw image type detected. In the future it will be necessary \
-                        to specify image_type for non-raw files."
-                    );
-                }
-
-                if detected_image_type == ImageType::Qcow2 && disk_cfg.backing_files {
-                    warn!("QCOW2 image type autodetected. Disabling backing files");
-                    disk_cfg.backing_files = false;
+                // Always access as raw image unless otherwise specified
+                disk_cfg.image_type = ImageType::Raw;
+            } else if disk_cfg.image_type != ImageType::Raw {
+                if disk_cfg.image_type != detected_image_type {
+                    return Err(DeviceManagerError::DiskImageTypeMismatch {
+                        specified: disk_cfg.image_type,
+                        detected: detected_image_type,
+                    });
                 }
-
-                disk_cfg.image_type = detected_image_type;
-            } else if disk_cfg.image_type != detected_image_type {
-                return Err(DeviceManagerError::DiskImageTypeMismatch {
-                    specified: disk_cfg.image_type,
-                    detected: detected_image_type,
-                });
             }
 
             if disk_cfg.image_type != ImageType::Qcow2 && disk_cfg.backing_files {
@@ -2857,7 +2830,7 @@ impl DeviceManager {
                     .map_err(DeviceManagerError::RestoreGetState)?,
                 queue_affinity,
                 disk_cfg.sparse,
-                disable_sector0_writes,
+                false,
                 disk_cfg.lock_granularity,
             )
             .map_err(DeviceManagerError::CreateVirtioBlock)?;
