diff --git a/vmm/src/device_manager.rs b/vmm/src/device_manager.rs
index a7d3254c..d5aebff0 100644
--- a/vmm/src/device_manager.rs
+++ b/vmm/src/device_manager.rs
@@ -2675,37 +2675,21 @@ impl DeviceManager {
                 )
                 .map_err(DeviceManagerError::Disk)?;
 
-            let detected_image_type =
-                detect_image_type(&mut file).map_err(DeviceManagerError::DetectImageType)?;
-            let mut disable_sector0_writes = false;
 
             if disk_cfg.image_type == ImageType::Unknown {
-                warn!(
-                    "No image_type specified - detected as {detected_image_type}. \
-                    Configuration updated to persist type across reboots and migrations."
-                );
-
-                if detected_image_type == ImageType::Raw {
-                    warn!("Autodetected raw image type. Disabling sector 0 writes.");
-                    disable_sector0_writes = true;
-                } else {
-                    warn!(
-                        "Non-raw image type detected. In the future it will be necessary \
-                        to specify image_type for non-raw files."
-                    );
-                }
+                // Always access as raw image unless otherwise specified
+                disk_cfg.image_type = ImageType::Raw;
+            }
 
-                if detected_image_type == ImageType::Qcow2 && disk_cfg.backing_files {
-                    warn!("QCOW2 image type autodetected. Disabling backing files");
-                    disk_cfg.backing_files = false;
+            if disk_cfg.image_type != ImageType::Raw {
+                let detected_image_type =
+                    detect_image_type(&mut file).map_err(DeviceManagerError::DetectImageType)?;
+                if disk_cfg.image_type != detected_image_type {
+                    return Err(DeviceManagerError::DiskImageTypeMismatch {
+                        specified: disk_cfg.image_type,
+                        detected: detected_image_type,
+                    });
                 }
-
-                disk_cfg.image_type = detected_image_type;
-            } else if disk_cfg.image_type != detected_image_type {
-                return Err(DeviceManagerError::DiskImageTypeMismatch {
-                    specified: disk_cfg.image_type,
-                    detected: detected_image_type,
-                });
             }
 
             if disk_cfg.image_type != ImageType::Qcow2 && disk_cfg.backing_files {
@@ -2855,7 +2839,7 @@ impl DeviceManager {
                     .map_err(DeviceManagerError::RestoreGetState)?,
                 queue_affinity,
                 disk_cfg.sparse,
-                disable_sector0_writes,
+                false,
             )
             .map_err(DeviceManagerError::CreateVirtioBlock)?;
 
