diff --git a/net_util/src/open_tap.rs b/net_util/src/open_tap.rs
index 4276012d..1c16e599 100644
--- a/net_util/src/open_tap.rs
+++ b/net_util/src/open_tap.rs
@@ -3,7 +3,6 @@
 // SPDX-License-Identifier: Apache-2.0 AND BSD-3-Clause
 
 use super::{vnet_hdr_len, MacAddr, Tap, TapError};
-use std::net::Ipv4Addr;
 use std::path::Path;
 use std::{fs, io};
 use thiserror::Error;
@@ -62,8 +61,6 @@ fn check_mq_support(if_name: &Option<&str>, queue_pairs: usize) -> Result<()> {
 /// netmask.
 pub fn open_tap(
     if_name: Option<&str>,
-    ip_addr: Option<Ipv4Addr>,
-    netmask: Option<Ipv4Addr>,
     host_mac: &mut Option<MacAddr>,
     mtu: Option<u16>,
     num_rx_q: usize,
@@ -72,10 +69,6 @@ pub fn open_tap(
     let mut taps: Vec<Tap> = Vec::new();
     let mut ifname: String = String::new();
     let vnet_hdr_size = vnet_hdr_len() as i32;
-    // Check if the given interface exists before we create it.
-    let tap_existed = if_name.map_or(false, |n| {
-        Path::new(&format!("/sys/class/net/{n}")).exists()
-    });
 
     // In case the tap interface already exists, check if the number of
     // queues is appropriate. The tap might not support multiqueue while
@@ -91,20 +84,6 @@ pub fn open_tap(
                 Some(name) => Tap::open_named(name, num_rx_q, flags).map_err(Error::TapOpen)?,
                 None => Tap::new(num_rx_q).map_err(Error::TapOpen)?,
             };
-            // Don't overwrite ip configuration of existing interfaces:
-            if !tap_existed {
-                if let Some(ip) = ip_addr {
-                    tap.set_ip_addr(ip).map_err(Error::TapSetIp)?;
-                }
-                if let Some(mask) = netmask {
-                    tap.set_netmask(mask).map_err(Error::TapSetNetmask)?;
-                }
-            } else {
-                warn!(
-                    "Tap {} already exists. IP configuration will not be overwritten.",
-                    if_name.unwrap_or_default()
-                );
-            }
             if let Some(mac) = host_mac {
                 tap.set_mac_addr(*mac).map_err(Error::TapSetMac)?
             } else {
diff --git a/vhost_user_net/src/lib.rs b/vhost_user_net/src/lib.rs
index b9b9d9c4..57d46d09 100644
--- a/vhost_user_net/src/lib.rs
+++ b/vhost_user_net/src/lib.rs
@@ -15,7 +15,6 @@ use option_parser::Toggle;
 use option_parser::{OptionParser, OptionParserError};
 use std::fmt;
 use std::io;
-use std::net::Ipv4Addr;
 use std::ops::Deref;
 use std::os::unix::io::{AsRawFd, RawFd};
 use std::process;
@@ -57,7 +56,7 @@ pub enum Error {
 }
 
 pub const SYNTAX: &str = "vhost-user-net backend parameters \
-\"ip=<ip_addr>,mask=<net_mask>,socket=<socket_path>,client=on|off,\
+\"socket=<socket_path>,client=on|off,\
 num_queues=<number_of_queues>,queue_size=<size_of_each_queue>,tap=<if_name>\"";
 
 impl fmt::Display for Error {
@@ -119,9 +118,7 @@ pub struct VhostUserNetBackend {
 impl VhostUserNetBackend {
     #[allow(clippy::too_many_arguments)]
     fn new(
-        ip_addr: Ipv4Addr,
         host_mac: MacAddr,
-        netmask: Ipv4Addr,
         mtu: Option<u16>,
         num_queues: usize,
         queue_size: u16,
@@ -130,8 +127,6 @@ impl VhostUserNetBackend {
     ) -> Result<Self> {
         let mut taps = open_tap(
             ifname,
-            Some(ip_addr),
-            Some(netmask),
             &mut Some(host_mac),
             mtu,
             num_queues / 2,
@@ -289,9 +284,7 @@ impl VhostUserNetBackendConfig {
 
         parser
             .add("tap")
-            .add("ip")
             .add("host_mac")
-            .add("mask")
             .add("mtu")
             .add("queue_size")
             .add("num_queues")
@@ -301,18 +294,10 @@ impl VhostUserNetBackendConfig {
         parser.parse(backend).map_err(Error::FailedConfigParse)?;
 
         let tap = parser.get("tap");
-        let ip = parser
-            .convert("ip")
-            .map_err(Error::FailedConfigParse)?
-            .unwrap_or_else(|| Ipv4Addr::new(192, 168, 100, 1));
         let host_mac = parser
             .convert("host_mac")
             .map_err(Error::FailedConfigParse)?
             .unwrap_or_else(MacAddr::local_random);
-        let mask = parser
-            .convert("mask")
-            .map_err(Error::FailedConfigParse)?
-            .unwrap_or_else(|| Ipv4Addr::new(255, 255, 255, 0));
         let mtu = parser.convert("mtu").map_err(Error::FailedConfigParse)?;
         let queue_size = parser
             .convert("queue_size")
@@ -330,9 +315,7 @@ impl VhostUserNetBackendConfig {
             .0;
 
         Ok(VhostUserNetBackendConfig {
-            ip,
             host_mac,
-            mask,
             mtu,
             socket,
             num_queues,
@@ -358,9 +341,7 @@ pub fn start_net_backend(backend_command: &str) {
 
     let net_backend = Arc::new(RwLock::new(
         VhostUserNetBackend::new(
-            backend_config.ip,
             backend_config.host_mac,
-            backend_config.mask,
             backend_config.mtu,
             backend_config.num_queues,
             backend_config.queue_size,
diff --git a/virtio-devices/src/net.rs b/virtio-devices/src/net.rs
index ce46b938..dbd46c3e 100644
--- a/virtio-devices/src/net.rs
+++ b/virtio-devices/src/net.rs
@@ -26,7 +26,6 @@ use net_util::{
 use seccompiler::SeccompAction;
 use serde::{Deserialize, Serialize};
 use std::collections::HashMap;
-use std::net::Ipv4Addr;
 use std::num::Wrapping;
 use std::ops::Deref;
 use std::os::unix::io::{AsRawFd, RawFd};
@@ -547,8 +546,6 @@ impl Net {
     pub fn new(
         id: String,
         if_name: Option<&str>,
-        ip_addr: Option<Ipv4Addr>,
-        netmask: Option<Ipv4Addr>,
         guest_mac: Option<MacAddr>,
         host_mac: &mut Option<MacAddr>,
         mtu: Option<u16>,
@@ -565,8 +562,6 @@ impl Net {
     ) -> Result<Self> {
         let taps = open_tap(
             if_name,
-            ip_addr,
-            netmask,
             host_mac,
             mtu,
             num_queues / 2,
diff --git a/vmm/src/config.rs b/vmm/src/config.rs
index 765baf89..31444859 100644
--- a/vmm/src/config.rs
+++ b/vmm/src/config.rs
@@ -1296,7 +1296,7 @@ impl FromStr for VhostMode {
 
 impl NetConfig {
     pub const SYNTAX: &'static str = "Network parameters: \
-    tap=<if_name>,ip=<ip_addr>,mask=<net_mask>,mac=<mac_addr>,fd=<fd1>,<fd2>...,iommu=on|off,\
+    tap=<if_name>,mac=<mac_addr>,fd=<fd1>,<fd2>...,iommu=on|off,\
     num_queues=<number_of_queues>,queue_size=<size_of_each_queue>,id=<device_id>,\
     vhost_user=<vhost_user_enable>,socket=<vhost_user_socket_path>,vhost_mode=client|server,\
     bw_size=<bytes>,bw_one_time_burst=<bytes>,bw_refill_time=<ms>,\
@@ -1308,8 +1308,6 @@ impl NetConfig {
 
         parser
             .add("tap")
-            .add("ip")
-            .add("mask")
             .add("mac")
             .add("host_mac")
             .add("offload_tso")
@@ -1334,14 +1332,6 @@ impl NetConfig {
         parser.parse(net).map_err(Error::ParseNetwork)?;
 
         let tap = parser.get("tap");
-        let ip = parser
-            .convert("ip")
-            .map_err(Error::ParseNetwork)?
-            .unwrap_or_else(default_netconfig_ip);
-        let mask = parser
-            .convert("mask")
-            .map_err(Error::ParseNetwork)?
-            .unwrap_or_else(default_netconfig_mask);
         let mac = parser
             .convert("mac")
             .map_err(Error::ParseNetwork)?
@@ -1448,8 +1438,6 @@ impl NetConfig {
 
         let config = NetConfig {
             tap,
-            ip,
-            mask,
             mac,
             host_mac,
             mtu,
diff --git a/vmm/src/device_manager.rs b/vmm/src/device_manager.rs
index 80eac72c..7cef6a42 100644
--- a/vmm/src/device_manager.rs
+++ b/vmm/src/device_manager.rs
@@ -2679,8 +2679,6 @@ impl DeviceManager {
                     virtio_devices::Net::new(
                         id.clone(),
                         Some(tap_if_name),
-                        Some(net_cfg.ip),
-                        Some(net_cfg.mask),
                         Some(net_cfg.mac),
                         &mut net_cfg.host_mac,
                         net_cfg.mtu,
@@ -2730,8 +2728,6 @@ impl DeviceManager {
                     virtio_devices::Net::new(
                         id.clone(),
                         None,
-                        Some(net_cfg.ip),
-                        Some(net_cfg.mask),
                         Some(net_cfg.mac),
                         &mut net_cfg.host_mac,
                         net_cfg.mtu,
diff --git a/vmm/src/vm_config.rs b/vmm/src/vm_config.rs
index bf257770..80c500df 100644
--- a/vmm/src/vm_config.rs
+++ b/vmm/src/vm_config.rs
@@ -4,7 +4,7 @@
 //
 use net_util::MacAddr;
 use serde::{Deserialize, Serialize};
-use std::{net::Ipv4Addr, path::PathBuf};
+use std::path::PathBuf;
 use virtio_devices::RateLimiterConfig;
 
 #[derive(Clone, Debug, PartialEq, Eq, Deserialize, Serialize)]
@@ -261,10 +261,6 @@ pub fn default_diskconfig_queue_size() -> u16 {
 pub struct NetConfig {
     #[serde(default = "default_netconfig_tap")]
     pub tap: Option<String>,
-    #[serde(default = "default_netconfig_ip")]
-    pub ip: Ipv4Addr,
-    #[serde(default = "default_netconfig_mask")]
-    pub mask: Ipv4Addr,
     #[serde(default = "default_netconfig_mac")]
     pub mac: MacAddr,
     #[serde(default)]
@@ -310,14 +306,6 @@ pub fn default_netconfig_tap() -> Option<String> {
     None
 }
 
-pub fn default_netconfig_ip() -> Ipv4Addr {
-    Ipv4Addr::new(192, 168, 249, 1)
-}
-
-pub fn default_netconfig_mask() -> Ipv4Addr {
-    Ipv4Addr::new(255, 255, 255, 0)
-}
-
 pub fn default_netconfig_mac() -> MacAddr {
     MacAddr::local_random()
 }
