diff --git a/hostapd/main.c b/hostapd/main.c
index c9ec38d..8384c25 100644
--- a/hostapd/main.c
+++ b/hostapd/main.c
@@ -452,9 +452,7 @@ static void show_version(void)
 {
 	fprintf(stderr,
 		"hostapd v%s\n"
-		"User space daemon for IEEE 802.11 AP management,\n"
-		"IEEE 802.1X/WPA/WPA2/EAP/RADIUS Authenticator\n"
-		"Copyright (c) 2002-2022, Jouni Malinen <j@w1.fi> "
+		"Copyright (C) 2002-2022, Jouni Malinen <j@w1.fi> "
 		"and contributors\n",
 		VERSION_STR);
 }
@@ -465,35 +463,34 @@ static void usage(void)
 	show_version();
 	fprintf(stderr,
 		"\n"
-		"usage: hostapd [-hdBKtv] [-P <PID file>] [-e <entropy file>] "
+		"Usage: hostapd [-hdBKtv] [-P <pid-file>] [-e <entropy-file>] "
 		"\\\n"
-		"         [-g <global ctrl_iface>] [-G <group>]\\\n"
-		"         [-i <comma-separated list of interface names>]\\\n"
-		"         <configuration file(s)>\n"
+		"         [-g <global-ctrl>] [-G <group>] \\\n"
+		"         [-i <comma-separated-list-of-interfaces>] \\\n"
+		"         <config-file>...\n"
 		"\n"
-		"options:\n"
-		"   -h   show this usage\n"
-		"   -d   show more debug messages (-dd for even more)\n"
-		"   -B   run daemon in the background\n"
-		"   -e   entropy file\n"
-		"   -g   global control interface path\n"
-		"   -G   group for control interfaces\n"
-		"   -P   PID file\n"
-		"   -K   include key data in debug messages\n"
+		"Options:\n"
+		"  -h   show this usage\n"
+		"  -d   show more debug messages (-dd for even more)\n"
+		"  -B   run daemon in the background\n"
+		"  -e   entropy file\n"
+		"  -g   global control interface path\n"
+		"  -G   group for control interfaces\n"
+		"  -P   PID file\n"
+		"  -K   include key data in debug messages\n"
 #ifdef CONFIG_DEBUG_FILE
-		"   -f   log output to debug file instead of stdout\n"
+		"  -f   log output to debug file instead of stdout\n"
 #endif /* CONFIG_DEBUG_FILE */
 #ifdef CONFIG_DEBUG_LINUX_TRACING
-		"   -T   record to Linux tracing in addition to logging\n"
-		"        (records all messages regardless of debug verbosity)\n"
+		"  -T   record to Linux tracing in addition to logging\n"
 #endif /* CONFIG_DEBUG_LINUX_TRACING */
-		"   -i   list of interface names to use\n"
+		"  -i   comma-separated list of interface names\n"
 #ifdef CONFIG_DEBUG_SYSLOG
-		"   -s   log output to syslog instead of stdout\n"
+		"  -s   log output to syslog instead of stdout\n"
 #endif /* CONFIG_DEBUG_SYSLOG */
-		"   -S   start all the interfaces synchronously\n"
-		"   -t   include timestamps in some debug messages\n"
-		"   -v   show hostapd version\n");
+		"  -S   start all the interfaces synchronously\n"
+		"  -t   include timestamps in some debug messages\n"
+		"  -v   show version\n");
 
 	exit(1);
 }
diff --git a/src/utils/wpa_debug.c b/src/utils/wpa_debug.c
index a338a20..33eb411 100644
--- a/src/utils/wpa_debug.c
+++ b/src/utils/wpa_debug.c
@@ -93,7 +93,7 @@ void wpa_debug_print_timestamp(void)
 
 void wpa_debug_open_syslog(void)
 {
-	openlog("wpa_supplicant", LOG_PID | LOG_NDELAY, LOG_HOSTAPD);
+	openlog("wpad", LOG_PID | LOG_NDELAY, LOG_HOSTAPD);
 	wpa_debug_syslog++;
 }
 
diff --git a/wpa_supplicant/main.c b/wpa_supplicant/main.c
index 51a8a02..5534418 100644
--- a/wpa_supplicant/main.c
+++ b/wpa_supplicant/main.c
@@ -21,9 +21,8 @@
 static void usage(void)
 {
 	int i;
-	printf("%s\n\n%s\n"
-	       "usage:\n"
-	       "  wpa_supplicant [-BddhKLqq"
+	printf("%s\n\n"
+	       "Usage: wpad [-BddhKLqq"
 #ifdef CONFIG_DEBUG_SYSLOG
 	       "s"
 #endif /* CONFIG_DEBUG_SYSLOG */
@@ -31,27 +30,26 @@ static void usage(void)
 #ifdef CONFIG_CTRL_IFACE_DBUS_NEW
 	       "u"
 #endif /* CONFIG_CTRL_IFACE_DBUS_NEW */
-	       "vW] [-P<pid file>] "
-	       "[-g<global ctrl>] \\\n"
-	       "        [-G<group>] \\\n"
-	       "        -i<ifname> -c<config file> [-C<ctrl>] [-D<driver>] "
-	       "[-p<driver_param>] \\\n"
-	       "        [-b<br_ifname>] [-e<entropy file>]"
+	       "vW] [-P <pid-file>] \\\n"
+	       "         [-g <global-ctrl>] [-G <group>] \\\n"
+	       "         -i <interface> -c <config-file> [-C <ctrl-param>] \\\n"
+	       "         [-D <driver>] [-p <driver-param>] \\\n"
+	       "         [-b <bridge>] [-e <entropy-file>]"
 #ifdef CONFIG_DEBUG_FILE
-	       " [-f<debug file>]"
+	       " [-f <debug-file>]"
 #endif /* CONFIG_DEBUG_FILE */
 	       " \\\n"
-	       "        [-o<override driver>] [-O<override ctrl>] \\\n"
-	       "        [-N -i<ifname> -c<conf> [-C<ctrl>] "
-	       "[-D<driver>] \\\n"
+	       "         [-o <override-driver>] [-O <override-ctrl>] \\\n"
+	       "         [-N -i <interface> -c <config-file> [-C <ctrl-param>] \\\n"
+	       "           [-D <driver>]"
 #ifdef CONFIG_P2P
-	       "        [-m<P2P Device config file>] \\\n"
+	       " [-m <p2p-device-config-file>]"
 #endif /* CONFIG_P2P */
-	       "        [-p<driver_param>] [-b<br_ifname>] [-I<config file>] "
-	       "...]\n"
+	       " [-p <driver-param>] \\\n"
+	       "           [-b <bridge>] [-I <config-file>]]...\n"
 	       "\n"
-	       "drivers:\n",
-	       wpa_supplicant_version, wpa_supplicant_license);
+	       "Drivers:\n",
+	       wpa_supplicant_version);
 
 	for (i = 0; wpa_drivers[i]; i++) {
 		printf("  %s = %s\n",
@@ -60,71 +58,55 @@ static void usage(void)
 	}
 
 #ifndef CONFIG_NO_STDOUT_DEBUG
-	printf("options:\n"
-	       "  -b = optional bridge interface name\n"
-	       "  -B = run daemon in the background\n"
-	       "  -c = Configuration file\n"
-	       "  -C = ctrl_interface parameter (only used if -c is not)\n"
-	       "  -d = increase debugging verbosity (-dd even more)\n"
-	       "  -D = driver name (can be multiple drivers: nl80211,wext)\n"
-	       "  -e = entropy file\n"
+	printf("Options:\n"
+	       "  -b   optional bridge interface name\n"
+	       "  -B   run daemon in the background\n"
+	       "  -c   configuration file\n"
+	       "  -C   control interface parameter (only used if -c is not)\n"
+	       "  -d   show more debug messages (-dd for even more)\n"
+	       "  -D   comma-separated driver name(s)\n"
+	       "  -e   entropy file\n"
 #ifdef CONFIG_DEBUG_FILE
-	       "  -f = log output to debug file instead of stdout\n"
+	       "  -f   log output to debug file instead of stdout\n"
 #endif /* CONFIG_DEBUG_FILE */
-	       "  -g = global ctrl_interface\n"
-	       "  -G = global ctrl_interface group\n"
-	       "  -h = show this help text\n"
-	       "  -i = interface name\n"
-	       "  -I = additional configuration file\n"
-	       "  -K = include keys (passwords, etc.) in debug output\n"
-	       "  -L = show license (BSD)\n"
+	       "  -g   global control interface path\n"
+	       "  -G   group for control interfaces\n"
+	       "  -h   show this usage\n"
+	       "  -i   interface name\n"
+	       "  -I   additional configuration file\n"
+	       "  -K   include key data in debug messages\n"
 #ifdef CONFIG_P2P
-	       "  -m = Configuration file for the P2P Device interface\n"
+	       "  -m   configuration file for the P2P device interface\n"
 #endif /* CONFIG_P2P */
 #ifdef CONFIG_MATCH_IFACE
-	       "  -M = start describing new matching interface\n"
+	       "  -M   start describing new matching interface\n"
 #endif /* CONFIG_MATCH_IFACE */
-	       "  -N = start describing new interface\n"
-	       "  -o = override driver parameter for new interfaces\n"
-	       "  -O = override ctrl_interface parameter for new interfaces\n"
-	       "  -p = driver parameters\n"
-	       "  -P = PID file\n"
-	       "  -q = decrease debugging verbosity (-qq even less)\n"
+	       "  -N   start describing new interface\n"
+	       "  -o   override driver parameter for new interfaces\n"
+	       "  -O   override control interface parameter for new interfaces\n"
+	       "  -p   driver parameters\n"
+	       "  -P   PID file\n"
+	       "  -q   show fewer debug messages (-qq for even fewer)\n"
 #ifdef CONFIG_DEBUG_SYSLOG
-	       "  -s = log output to syslog instead of stdout\n"
+	       "  -s   log output to syslog instead of stdout\n"
 #endif /* CONFIG_DEBUG_SYSLOG */
-	       "  -t = include timestamp in debug messages\n"
+	       "  -t   include timestamps in debug messages\n"
 #ifdef CONFIG_DEBUG_LINUX_TRACING
-	       "  -T = record to Linux tracing in addition to logging\n"
-	       "       (records all messages regardless of debug verbosity)\n"
+	       "  -T   record to Linux tracing in addition to logging\n"
 #endif /* CONFIG_DEBUG_LINUX_TRACING */
 #ifdef CONFIG_CTRL_IFACE_DBUS_NEW
-	       "  -u = enable DBus control interface\n"
+	       "  -u   enable dbus control interface\n"
 #endif /* CONFIG_CTRL_IFACE_DBUS_NEW */
-	       "  -v = show version\n"
-	       "  -W = wait for a control interface monitor before starting\n");
+	       "  -v   show version\n"
+	       "  -W   wait for a control interface monitor before starting\n");
 
-	printf("example:\n"
-	       "  wpa_supplicant -D%s -iwlan0 -c/etc/wpa_supplicant.conf\n",
+	printf("Example:\n"
+	       "  wpad -D %s -i wlan0 -c /etc/wpad.conf\n",
 	       wpa_drivers[0] ? wpa_drivers[0]->name : "nl80211");
 #endif /* CONFIG_NO_STDOUT_DEBUG */
 }
 
 
-static void license(void)
-{
-#ifndef CONFIG_NO_STDOUT_DEBUG
-	printf("%s\n\n%s%s%s%s%s\n",
-	       wpa_supplicant_version,
-	       wpa_supplicant_full_license1,
-	       wpa_supplicant_full_license2,
-	       wpa_supplicant_full_license3,
-	       wpa_supplicant_full_license4,
-	       wpa_supplicant_full_license5);
-#endif /* CONFIG_NO_STDOUT_DEBUG */
-}
-
-
 static void wpa_supplicant_fd_workaround(int start)
 {
 #ifdef __linux__
@@ -257,10 +239,6 @@ int main(int argc, char *argv[])
 		case 'K':
 			params.wpa_debug_show_keys++;
 			break;
-		case 'L':
-			license();
-			exitcode = 0;
-			goto out;
 #ifdef CONFIG_P2P
 		case 'm':
 			params.conf_p2p_dev = optarg;
@@ -341,12 +319,9 @@ int main(int argc, char *argv[])
 	exitcode = 0;
 	global = wpa_supplicant_init(&params);
 	if (global == NULL) {
-		wpa_printf(MSG_ERROR, "Failed to initialize wpa_supplicant");
+		wpa_printf(MSG_ERROR, "Failed to initialize wpad");
 		exitcode = -1;
 		goto out;
-	} else {
-		wpa_printf(MSG_INFO, "Successfully initialized "
-			   "wpa_supplicant");
 	}
 
 	if (fst_global_init()) {
diff --git a/wpa_supplicant/wpa_supplicant.c b/wpa_supplicant/wpa_supplicant.c
index d37a994..795bdad 100644
--- a/wpa_supplicant/wpa_supplicant.c
+++ b/wpa_supplicant/wpa_supplicant.c
@@ -70,8 +70,8 @@
 #endif /* CONFIG_MESH */
 
 const char *const wpa_supplicant_version =
-"wpa_supplicant v" VERSION_STR "\n"
-"Copyright (c) 2003-2022, Jouni Malinen <j@w1.fi> and contributors";
+"wpad v" VERSION_STR "\n"
+"Copyright (C) 2003-2022, Jouni Malinen <j@w1.fi> and contributors";
 
 const char *const wpa_supplicant_license =
 "This software may be distributed under the terms of the BSD license.\n"
@@ -4934,8 +4934,7 @@ static int wpa_supplicant_set_driver(struct wpa_supplicant *wpa_s,
 		return -1;
 
 	if (wpa_drivers[0] == NULL) {
-		wpa_msg(wpa_s, MSG_ERROR, "No driver interfaces build into "
-			"wpa_supplicant");
+		wpa_msg(wpa_s, MSG_ERROR, "No driver interfaces build into wpad");
 		return -1;
 	}
 
@@ -6815,12 +6814,12 @@ static int wpa_supplicant_init_iface(struct wpa_supplicant *wpa_s,
 	if (wpa_s->ctrl_iface == NULL) {
 		wpa_printf(MSG_ERROR,
 			   "Failed to initialize control interface '%s'.\n"
-			   "You may have another wpa_supplicant process "
+			   "You may have another wpad process "
 			   "already running or the file was\n"
-			   "left by an unclean termination of wpa_supplicant "
+			   "left by an unclean termination of wpad "
 			   "in which case you will need\n"
 			   "to manually remove this file before starting "
-			   "wpa_supplicant again.\n",
+			   "wpad again.\n",
 			   wpa_s->conf->ctrl_interface);
 		return -1;
 	}
@@ -7392,7 +7391,7 @@ struct wpa_global * wpa_supplicant_init(struct wpa_params *params)
 	wpa_debug_timestamp = global->params.wpa_debug_timestamp =
 		params->wpa_debug_timestamp;
 
-	wpa_printf(MSG_DEBUG, "wpa_supplicant v%s", VERSION_STR);
+	wpa_printf(MSG_DEBUG, "wpad v%s", VERSION_STR);
 
 	if (eloop_init()) {
 		wpa_printf(MSG_ERROR, "Failed to initialize event loop");
