diff --git a/Makefile.am b/Makefile.am
index ce3267f..4437693 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -41,7 +41,7 @@ MAINTAINERCLEANFILES = Makefile.in aclocal.m4 configure configure-stamp \
 	$(distdir).tar.gz $(distdir).tar.bz2
 
 SUBDIRS = lib src include doc contrib control
-IANA_PEN = http://www.iana.org/assignments/enterprise-numbers
+IANA_PEN = https://www.iana.org/assignments/enterprise-numbers.txt
 
 dist-hook:
 	cp control/ipmitool.spec $(distdir)
