diff --git a/lib/ipmi_main.c b/lib/ipmi_main.c
index a673a30..3bfa0dd 100644
--- a/lib/ipmi_main.c
+++ b/lib/ipmi_main.c
@@ -821,12 +821,12 @@ ipmi_main(int argc, char ** argv,
 	}
 
 	/* if no interface was specified but a
-	 * hostname was then use LAN by default
+	 * hostname was then use lanplus by default
 	 * otherwise the default is hardcoded
 	 * to use the first entry in the list
 	 */
 	if (!intfname && hostname) {
-		intfname = strdup("lan");
+		intfname = strdup("lanplus");
 		if (!intfname) {
 			lprintf(LOG_ERR, "%s: malloc failure", progname);
 			goto out_free;
