diff --git a/vendor/getrandom/src/util_libc.rs b/vendor/getrandom/src/util_libc.rs
index d057071a7..8e9f1763a 100644
--- a/vendor/getrandom/src/util_libc.rs
+++ b/vendor/getrandom/src/util_libc.rs
@@ -136,7 +136,7 @@ impl Weak {
 }
 
 cfg_if! {
-    if #[cfg(any(target_os = "linux", target_os = "emscripten"))] {
+    if #[cfg(any(all(target_os = "linux", target_env = "gnu"), target_os = "emscripten"))] {
         use libc::open64 as open;
     } else {
         use libc::open;
