diff --git a/src/config/all-settings.c b/src/config/all-settings.c
index bd9d5316..16483f79 100644
--- a/src/config/all-settings.c
+++ b/src/config/all-settings.c
@@ -8085,9 +8085,9 @@ static const struct master_settings master_default_settings = {
 	.libexec_dir = PKG_LIBEXECDIR,
 	.instance_name = PACKAGE,
 	.ssl = "yes:no:required",
-	.default_internal_user = "dovecot",
-	.default_internal_group = "dovecot",
-	.default_login_user = "dovenull",
+	.default_internal_user = "mail",
+	.default_internal_group = "mail",
+	.default_login_user = "nobody",
 	.default_process_limit = 100,
 	.default_client_limit = 1000,
 	.default_idle_kill_interval = 60,
diff --git a/src/master/master-settings.c b/src/master/master-settings.c
index f3d54b6e..a581b950 100644
--- a/src/master/master-settings.c
+++ b/src/master/master-settings.c
@@ -221,9 +221,9 @@ static const struct master_settings master_default_settings = {
 	.libexec_dir = PKG_LIBEXECDIR,
 	.instance_name = PACKAGE,
 	.ssl = "yes:no:required",
-	.default_internal_user = "dovecot",
-	.default_internal_group = "dovecot",
-	.default_login_user = "dovenull",
+	.default_internal_user = "mail",
+	.default_internal_group = "mail",
+	.default_login_user = "nobody",
 	.default_process_limit = 100,
 	.default_client_limit = 1000,
 	.default_idle_kill_interval = 60,
