diff --git a/Makefile b/Makefile
index 50545ef6..510d0f21 100644
--- a/Makefile
+++ b/Makefile
@@ -66,7 +66,7 @@ CCOPTS = -O2 -pipe
 WFLAGS := -Wall -Wstrict-prototypes  -Wmissing-prototypes
 WFLAGS += -Wmissing-declarations -Wold-style-definition -Wformat=2
 
-CFLAGS := $(WFLAGS) $(CCOPTS) -I../include -I../include/uapi $(DEFINES) $(CFLAGS)
+CFLAGS := $(WFLAGS) $(CCOPTS) -I../include $(DEFINES) $(CFLAGS)
 YACCFLAGS = -d -t -v
 
 SUBDIRS=lib ip tc bridge misc netem genl man
diff --git a/lib/utils_math.c b/lib/utils_math.c
index a7e74744..92e739ab 100644
--- a/lib/utils_math.c
+++ b/lib/utils_math.c
@@ -1,5 +1,6 @@
 // SPDX-License-Identifier: GPL-2.0+
 
+#include <limits.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
diff --git a/tc/tc_core.c b/tc/tc_core.c
index a422e02c..5b18717c 100644
--- a/tc/tc_core.c
+++ b/tc/tc_core.c
@@ -5,6 +5,7 @@
  * Authors:	Alexey Kuznetsov, <kuznet@ms2.inr.ac.ru>
  */
 
+#include <limits.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <stdint.h>
