diff --git a/Makefile b/Makefile
index 20992362..357cf5a8 100644
--- a/Makefile
+++ b/Makefile
@@ -66,7 +66,7 @@ CCOPTS = -O2 -pipe
 WFLAGS := -Wall -Wstrict-prototypes  -Wmissing-prototypes
 WFLAGS += -Wmissing-declarations -Wold-style-definition -Wformat=2
 
-CFLAGS := $(WFLAGS) $(CCOPTS) -I../include -I../include/uapi $(DEFINES) $(CFLAGS)
+CFLAGS := $(WFLAGS) $(CCOPTS) -I../include $(DEFINES) $(CFLAGS)
 YACCFLAGS = -d -t -v
 
 SUBDIRS=lib ip tc bridge misc netem genl man
diff --git a/misc/nstat.c b/misc/nstat.c
index 4a9f3326..d96b7583 100644
--- a/misc/nstat.c
+++ b/misc/nstat.c
@@ -23,6 +23,7 @@
 #include <signal.h>
 #include <math.h>
 #include <getopt.h>
+#include <linux/types.h>
 
 #include "json_print.h"
 #include "version.h"
