diff --git a/include/json_print.h b/include/json_print.h
index 0b1d84f..403de29 100644
--- a/include/json_print.h
+++ b/include/json_print.h
@@ -8,6 +8,7 @@
 #ifndef _JSON_PRINT_H_
 #define _JSON_PRINT_H_
 
+#include <sys/time.h>
 #include "json_writer.h"
 #include "color.h"
 
diff --git a/lib/bpf_legacy.c b/lib/bpf_legacy.c
index 844974e..741eec8 100644
--- a/lib/bpf_legacy.c
+++ b/lib/bpf_legacy.c
@@ -18,6 +18,7 @@
 #include <stdarg.h>
 #include <limits.h>
 #include <assert.h>
+#include <libgen.h>
 
 #ifdef HAVE_ELF
 #include <libelf.h>
diff --git a/misc/arpd.c b/misc/arpd.c
index 1ef837c..cf36809 100644
--- a/misc/arpd.c
+++ b/misc/arpd.c
@@ -14,7 +14,7 @@
 #include <netdb.h>
 #include <db_185.h>
 #include <sys/ioctl.h>
-#include <sys/poll.h>
+#include <poll.h>
 #include <errno.h>
 #include <fcntl.h>
 #include <sys/uio.h>
diff --git a/misc/ifstat.c b/misc/ifstat.c
index f6f9ba5..441a337 100644
--- a/misc/ifstat.c
+++ b/misc/ifstat.c
@@ -17,7 +17,7 @@
 #include <sys/file.h>
 #include <sys/socket.h>
 #include <sys/un.h>
-#include <sys/poll.h>
+#include <poll.h>
 #include <sys/wait.h>
 #include <sys/stat.h>
 #include <signal.h>
diff --git a/misc/nstat.c b/misc/nstat.c
index 2c10fea..9270b72 100644
--- a/misc/nstat.c
+++ b/misc/nstat.c
@@ -17,7 +17,7 @@
 #include <sys/file.h>
 #include <sys/socket.h>
 #include <sys/un.h>
-#include <sys/poll.h>
+#include <poll.h>
 #include <sys/wait.h>
 #include <sys/stat.h>
 #include <signal.h>
diff --git a/misc/rtacct.c b/misc/rtacct.c
index 08363bf..cd84b7f 100644
--- a/misc/rtacct.c
+++ b/misc/rtacct.c
@@ -16,7 +16,7 @@
 #include <sys/file.h>
 #include <sys/socket.h>
 #include <sys/un.h>
-#include <sys/poll.h>
+#include <poll.h>
 #include <sys/wait.h>
 #include <sys/stat.h>
 #include <sys/mman.h>
