diff --git a/include/json_print.h b/include/json_print.h
index 91b3457..b3e8413 100644
--- a/include/json_print.h
+++ b/include/json_print.h
@@ -12,6 +12,7 @@
 #ifndef _JSON_PRINT_H_
 #define _JSON_PRINT_H_
 
+#include <sys/time.h>
 #include "json_writer.h"
 #include "color.h"
 
diff --git a/misc/arpd.c b/misc/arpd.c
index 504961c..a9c0fc2 100644
--- a/misc/arpd.c
+++ b/misc/arpd.c
@@ -18,7 +18,7 @@
 #include <netdb.h>
 #include <db_185.h>
 #include <sys/ioctl.h>
-#include <sys/poll.h>
+#include <poll.h>
 #include <errno.h>
 #include <fcntl.h>
 #include <sys/uio.h>
diff --git a/misc/ifstat.c b/misc/ifstat.c
index 291f288..df82bd7 100644
--- a/misc/ifstat.c
+++ b/misc/ifstat.c
@@ -21,7 +21,7 @@
 #include <sys/file.h>
 #include <sys/socket.h>
 #include <sys/un.h>
-#include <sys/poll.h>
+#include <poll.h>
 #include <sys/wait.h>
 #include <sys/stat.h>
 #include <signal.h>
diff --git a/misc/nstat.c b/misc/nstat.c
index 7160c59..0e69027 100644
--- a/misc/nstat.c
+++ b/misc/nstat.c
@@ -21,7 +21,7 @@
 #include <sys/file.h>
 #include <sys/socket.h>
 #include <sys/un.h>
-#include <sys/poll.h>
+#include <poll.h>
 #include <sys/wait.h>
 #include <sys/stat.h>
 #include <signal.h>
diff --git a/misc/rtacct.c b/misc/rtacct.c
index 47b27e3..690b673 100644
--- a/misc/rtacct.c
+++ b/misc/rtacct.c
@@ -21,7 +21,7 @@
 #include <sys/file.h>
 #include <sys/socket.h>
 #include <sys/un.h>
-#include <sys/poll.h>
+#include <poll.h>
 #include <sys/wait.h>
 #include <sys/stat.h>
 #include <sys/mman.h>
