diff --git a/ip/ipaddress.c b/ip/ipaddress.c
index 7accbf7..b5347e5 100644
--- a/ip/ipaddress.c
+++ b/ip/ipaddress.c
@@ -2577,16 +2577,8 @@ static int ipaddr_modify(int cmd, int flags, int argc, char **argv)
 	}
 
 	if (peer_len == 0 && local_len) {
-		if (cmd == RTM_DELADDR && lcl.family == AF_INET && !(lcl.flags & PREFIXLEN_SPECIFIED)) {
-			fprintf(stderr,
-			    "Warning: Executing wildcard deletion to stay compatible with old scripts.\n"
-			    "         Explicitly specify the prefix length (%s/%d) to avoid this warning.\n"
-			    "         This special behaviour is likely to disappear in further releases,\n"
-			    "         fix your scripts!\n", lcl_arg, local_len*8);
-		} else {
-			peer = lcl;
-			addattr_l(&req.n, sizeof(req), IFA_ADDRESS, &lcl.data, lcl.bytelen);
-		}
+		peer = lcl;
+		addattr_l(&req.n, sizeof(req), IFA_ADDRESS, &lcl.data, lcl.bytelen);
 	}
 	if (req.ifa.ifa_prefixlen == 0)
 		req.ifa.ifa_prefixlen = lcl.bitlen;
