diff --git a/Makefile b/Makefile
index f621453..faf3ead 100644
--- a/Makefile
+++ b/Makefile
@@ -17,7 +17,7 @@ endif
 PREFIX?=/usr
 SBINDIR?=/sbin
 CONFDIR?=/etc/iproute2
-NETNS_RUN_DIR?=/var/run/netns
+NETNS_RUN_DIR?=/run/netns
 NETNS_ETC_DIR?=/etc/netns
 DATADIR?=$(PREFIX)/share
 HDRDIR?=$(PREFIX)/include/iproute2
diff --git a/include/namespace.h b/include/namespace.h
index e47f9b5..d495e6b 100644
--- a/include/namespace.h
+++ b/include/namespace.h
@@ -9,7 +9,7 @@
 #include <errno.h>
 
 #ifndef NETNS_RUN_DIR
-#define NETNS_RUN_DIR "/var/run/netns"
+#define NETNS_RUN_DIR "/run/netns"
 #endif
 
 #ifndef NETNS_ETC_DIR
diff --git a/lib/fs.c b/lib/fs.c
index 3752931..7d1cf87 100644
--- a/lib/fs.c
+++ b/lib/fs.c
@@ -32,7 +32,7 @@
 #define CGROUP2_FS_NAME "cgroup2"
 
 /* if not already mounted cgroup2 is mounted here for iproute2's use */
-#define MNT_CGRP2_PATH  "/var/run/cgroup2"
+#define MNT_CGRP2_PATH  "/sys/fs/cgroup"
 
 
 #ifndef HAVE_HANDLE_AT
