diff --git a/misc-utils/logger.c b/misc-utils/logger.c
index 4c2bfb0..f4bb83a 100644
--- a/misc-utils/logger.c
+++ b/misc-utils/logger.c
@@ -420,7 +420,7 @@ static char const *rfc3164_current_time(void)
 	};
 
 	logger_gettimeofday(&tv, NULL);
-	tm = localtime(&tv.tv_sec);
+	tm = gmtime(&tv.tv_sec);
 	snprintf(time, sizeof(time),"%s %2d %2.2d:%2.2d:%2.2d",
 		monthnames[tm->tm_mon], tm->tm_mday,
 		tm->tm_hour, tm->tm_min, tm->tm_sec);
