diff --git a/Lib/aifc.py b/Lib/aifc.py
index 1916e7e..a8234c0 100644
--- a/Lib/aifc.py
+++ b/Lib/aifc.py
@@ -928,7 +928,7 @@ def openfp(f, mode=None):
 if __name__ == '__main__':
     import sys
     if not sys.argv[1:]:
-        sys.argv.append('/usr/demos/data/audio/bach.aiff')
+        sys.argv.append('/lib/demos/data/audio/bach.aiff')
     fn = sys.argv[1]
     with open(fn, 'r') as f:
         print("Reading", fn)
diff --git a/Lib/cgi.py b/Lib/cgi.py
index c22c71b..cdc6415 100755
--- a/Lib/cgi.py
+++ b/Lib/cgi.py
@@ -1,13 +1,4 @@
-#! /usr/local/bin/python
-
-# NOTE: the above "/usr/local/bin/python" is NOT a mistake.  It is
-# intentionally NOT "/usr/bin/env python".  On many systems
-# (e.g. Solaris), /usr/local/bin is not in $PATH as passed to CGI
-# scripts, and /usr/local/bin is the default directory where Python is
-# installed, so /usr/bin/env would be unable to find python.  Granted,
-# binary installations by Linux vendors often install Python in
-# /usr/bin.  So let those vendors patch cgi.py to match their choice
-# of installation.
+#!/usr/bin/python3
 
 """Support module for CGI (Common Gateway Interface) scripts.
 
diff --git a/Lib/ctypes/util.py b/Lib/ctypes/util.py
index 33aae30..90dea29 100644
--- a/Lib/ctypes/util.py
+++ b/Lib/ctypes/util.py
@@ -90,7 +90,7 @@ elif sys.platform.startswith("aix"):
     from ctypes._aix import find_library
 
 elif os.name == "posix":
-    # Andreas Degert's find functions, using gcc, /sbin/ldconfig, objdump
+    # Andreas Degert's find functions, using gcc, /bin/ldconfig, objdump
     import re, tempfile
 
     def _findLib_gcc(name):
@@ -196,7 +196,7 @@ elif os.name == "posix":
             expr = os.fsencode(expr)
 
             try:
-                proc = subprocess.Popen(('/sbin/ldconfig', '-r'),
+                proc = subprocess.Popen(('/bin/ldconfig', '-r'),
                                         stdout=subprocess.PIPE,
                                         stderr=subprocess.DEVNULL)
             except OSError:  # E.g. command not found
diff --git a/Lib/distutils/command/install.py b/Lib/distutils/command/install.py
index c625c95..6f9ee2c 100644
--- a/Lib/distutils/command/install.py
+++ b/Lib/distutils/command/install.py
@@ -31,14 +31,14 @@ INSTALL_SCHEMES = {
     'unix_prefix': {
         'purelib': '$base/lib/python$py_version_short/site-packages',
         'platlib': '$platbase/lib/python$py_version_short/site-packages',
-        'headers': '$base/include/python$py_version_short$abiflags/$dist_name',
+        'headers': '$base/lib/include/python$py_version_short$abiflags/$dist_name',
         'scripts': '$base/bin',
         'data'   : '$base',
         },
     'unix_home': {
         'purelib': '$base/lib/python',
         'platlib': '$base/lib/python',
-        'headers': '$base/include/python/$dist_name',
+        'headers': '$base/lib/include/python/$dist_name',
         'scripts': '$base/bin',
         'data'   : '$base',
         },
diff --git a/Lib/distutils/sysconfig.py b/Lib/distutils/sysconfig.py
index b51629e..24a72d5 100644
--- a/Lib/distutils/sysconfig.py
+++ b/Lib/distutils/sysconfig.py
@@ -111,7 +111,7 @@ def get_python_inc(plat_specific=0, prefix=None):
                 incdir = os.path.join(get_config_var('srcdir'), 'Include')
                 return os.path.normpath(incdir)
         python_dir = 'python' + get_python_version() + build_flags
-        return os.path.join(prefix, "include", python_dir)
+        return os.path.join(prefix, "lib", "include", python_dir)
     elif os.name == "nt":
         if python_build:
             # Include both the include and PC dir to ensure we can find
diff --git a/Lib/encodings/rot_13.py b/Lib/encodings/rot_13.py
index 5627bfb..334624f 100755
--- a/Lib/encodings/rot_13.py
+++ b/Lib/encodings/rot_13.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 """ Python Character Mapping Codec for ROT13.
 
 This codec de/encodes from str to str.
diff --git a/Lib/mailcap.py b/Lib/mailcap.py
index bd0fc09..ee69031 100644
--- a/Lib/mailcap.py
+++ b/Lib/mailcap.py
@@ -54,8 +54,7 @@ def listmailcapfiles():
         else:
             # Don't bother with getpwuid()
             home = '.' # Last resort
-        mailcaps = [home + '/.mailcap', '/etc/mailcap',
-                '/usr/etc/mailcap', '/usr/local/etc/mailcap']
+        mailcaps = [home + '/.mailcap', '/etc/mailcap']
     return mailcaps
 
 
diff --git a/Lib/mimetypes.py b/Lib/mimetypes.py
index f33b658..e7bdc7c 100644
--- a/Lib/mimetypes.py
+++ b/Lib/mimetypes.py
@@ -41,14 +41,6 @@ __all__ = [
 
 knownfiles = [
     "/etc/mime.types",
-    "/etc/httpd/mime.types",                    # Mac OS X
-    "/etc/httpd/conf/mime.types",               # Apache
-    "/etc/apache/mime.types",                   # Apache 1
-    "/etc/apache2/mime.types",                  # Apache 2
-    "/usr/local/etc/httpd/conf/mime.types",
-    "/usr/local/lib/netscape/mime.types",
-    "/usr/local/etc/httpd/conf/mime.types",     # Apache 1.2
-    "/usr/local/etc/mime.types",                # Apache 1.3
     ]
 
 inited = False
diff --git a/Lib/posixpath.py b/Lib/posixpath.py
index ecb4e5a..7a01e6b 100644
--- a/Lib/posixpath.py
+++ b/Lib/posixpath.py
@@ -18,7 +18,7 @@ pardir = '..'
 extsep = '.'
 sep = '/'
 pathsep = ':'
-defpath = '/bin:/usr/bin'
+defpath = '/bin'
 altsep = None
 devnull = '/dev/null'
 
diff --git a/Lib/tempfile.py b/Lib/tempfile.py
index 6287554..64614b1 100644
--- a/Lib/tempfile.py
+++ b/Lib/tempfile.py
@@ -164,7 +164,7 @@ def _candidate_tempdir_list():
                          _os.path.expandvars(r'%SYSTEMROOT%\Temp'),
                          r'c:\temp', r'c:\tmp', r'\temp', r'\tmp' ])
     else:
-        dirlist.extend([ '/tmp', '/var/tmp', '/usr/tmp' ])
+        dirlist.extend([ '/tmp', '/var/tmp' ])
 
     # As a last resort, the current directory.
     try:
diff --git a/Lib/uuid.py b/Lib/uuid.py
index 188e16b..7b16747 100644
--- a/Lib/uuid.py
+++ b/Lib/uuid.py
@@ -351,7 +351,7 @@ def _popen(command, *args):
     import os, shutil, subprocess
     executable = shutil.which(command)
     if executable is None:
-        path = os.pathsep.join(('/sbin', '/usr/sbin'))
+        path = os.pathsep.join(('/bin',))
         executable = shutil.which(command, path=path)
         if executable is None:
             return None
diff --git a/Makefile.pre.in b/Makefile.pre.in
index a914a9c..d0dba79 100644
--- a/Makefile.pre.in
+++ b/Makefile.pre.in
@@ -142,7 +142,7 @@ BINDIR=		@bindir@
 LIBDIR=		@libdir@
 MANDIR=		@mandir@
 INCLUDEDIR=	@includedir@
-CONFINCLUDEDIR=	$(exec_prefix)/include
+CONFINCLUDEDIR=	$(exec_prefix)/lib/include
 SCRIPTDIR=	$(prefix)/lib
 ABIFLAGS=	@ABIFLAGS@
 
diff --git a/setup.py b/setup.py
index 20d7f35..ffc9a3b 100644
--- a/setup.py
+++ b/setup.py
@@ -564,9 +564,11 @@ class PyBuildExt(build_ext):
 
         if multiarch_path_component != '':
             add_dir_to_list(self.compiler.library_dirs,
-                            '/usr/lib/' + multiarch_path_component)
+                            '/lib/shared/' + multiarch_path_component)
+            add_dir_to_list(self.compiler.library_dirs,
+                            '/lib/static/' + multiarch_path_component)
             add_dir_to_list(self.compiler.include_dirs,
-                            '/usr/include/' + multiarch_path_component)
+                            '/lib/include/' + multiarch_path_component)
             return
 
         if not find_executable('dpkg-architecture'):
@@ -585,9 +587,11 @@ class PyBuildExt(build_ext):
                 with open(tmpfile) as fp:
                     multiarch_path_component = fp.readline().strip()
                 add_dir_to_list(self.compiler.library_dirs,
-                                '/usr/lib/' + multiarch_path_component)
+                                '/lib/shared/' + multiarch_path_component)
+                add_dir_to_list(self.compiler.library_dirs,
+                                '/lib/static/' + multiarch_path_component)
                 add_dir_to_list(self.compiler.include_dirs,
-                                '/usr/include/' + multiarch_path_component)
+                                '/lib/include/' + multiarch_path_component)
         finally:
             os.unlink(tmpfile)
 
@@ -670,8 +674,8 @@ class PyBuildExt(build_ext):
             add_dir_to_list(self.compiler.include_dirs,
                             sysconfig.get_config_var("INCLUDEDIR"))
 
-        system_lib_dirs = ['/lib64', '/usr/lib64', '/lib', '/usr/lib']
-        system_include_dirs = ['/usr/include']
+        system_lib_dirs = ['/lib/shared', '/lib/static', '/lib']
+        system_include_dirs = ['/lib/include']
         # lib_dirs and inc_dirs are used to search for files;
         # if a file is found in one of those directories, it can
         # be assumed that no additional -I,-L directives are needed.
@@ -953,11 +957,11 @@ class PyBuildExt(build_ext):
             elif curses_library:
                 readline_libs.append(curses_library)
             elif self.compiler.find_library_file(self.lib_dirs +
-                                                     ['/usr/lib/termcap'],
+                                                     ['/lib/termcap'],
                                                      'termcap'):
                 readline_libs.append('termcap')
             self.add(Extension('readline', ['readline.c'],
-                               library_dirs=['/usr/lib/termcap'],
+                               library_dirs=['/lib/termcap'],
                                extra_link_args=readline_extra_link_args,
                                libraries=readline_libs))
         else:
@@ -971,7 +975,7 @@ class PyBuildExt(build_ext):
         if curses_library == 'ncursesw':
             curses_defines.append(('HAVE_NCURSESW', '1'))
             if not CROSS_COMPILING:
-                curses_includes.append('/usr/include/ncursesw')
+                curses_includes.append('/lib/include/ncursesw')
             # Bug 1464056: If _curses.so links with ncursesw,
             # _curses_panel.so must link with panelw.
             panel_library = 'panelw'
@@ -1096,33 +1100,17 @@ class PyBuildExt(build_ext):
         # construct a list of paths to look for the header file in on
         # top of the normal inc_dirs.
         db_inc_paths = [
-            '/usr/include/db4',
-            '/usr/local/include/db4',
-            '/opt/sfw/include/db4',
-            '/usr/include/db3',
-            '/usr/local/include/db3',
-            '/opt/sfw/include/db3',
-            # Fink defaults (http://fink.sourceforge.net/)
-            '/sw/include/db4',
-            '/sw/include/db3',
+            '/lib/include/db4',
+            '/lib/include/db3',
         ]
         # 4.x minor number specific paths
         for x in gen_db_minor_ver_nums(4):
-            db_inc_paths.append('/usr/include/db4%d' % x)
-            db_inc_paths.append('/usr/include/db4.%d' % x)
-            db_inc_paths.append('/usr/local/BerkeleyDB.4.%d/include' % x)
-            db_inc_paths.append('/usr/local/include/db4%d' % x)
-            db_inc_paths.append('/pkg/db-4.%d/include' % x)
-            db_inc_paths.append('/opt/db-4.%d/include' % x)
-            # MacPorts default (http://www.macports.org/)
-            db_inc_paths.append('/opt/local/include/db4%d' % x)
+            db_inc_paths.append('/lib/include/db4%d' % x)
+            db_inc_paths.append('/lib/include/db4.%d' % x)
         # 3.x minor number specific paths
         for x in gen_db_minor_ver_nums(3):
-            db_inc_paths.append('/usr/include/db3%d' % x)
-            db_inc_paths.append('/usr/local/BerkeleyDB.3.%d/include' % x)
-            db_inc_paths.append('/usr/local/include/db3%d' % x)
-            db_inc_paths.append('/pkg/db-3.%d/include' % x)
-            db_inc_paths.append('/opt/db-3.%d/include' % x)
+            db_inc_paths.append('/lib/include/db3%d' % x)
+            db_inc_paths.append('/lib/include/db3.%d' % x)
 
         if CROSS_COMPILING:
             db_inc_paths = []
@@ -1350,12 +1338,9 @@ class PyBuildExt(build_ext):
         # We hunt for #define SQLITE_VERSION "n.n.n"
         # We need to find >= sqlite version 3.3.9, for sqlite3_prepare_v2
         sqlite_incdir = sqlite_libdir = None
-        sqlite_inc_paths = [ '/usr/include',
-                             '/usr/include/sqlite',
-                             '/usr/include/sqlite3',
-                             '/usr/local/include',
-                             '/usr/local/include/sqlite',
-                             '/usr/local/include/sqlite3',
+        sqlite_inc_paths = [ '/lib/include',
+                             '/lib/include/sqlite',
+                             '/lib/include/sqlite3',
                              ]
         if CROSS_COMPILING:
             sqlite_inc_paths = []
@@ -1673,7 +1658,7 @@ class PyBuildExt(build_ext):
 
     def detect_uuid(self):
         # Build the _uuid module if possible
-        uuid_incs = find_file("uuid.h", self.inc_dirs, ["/usr/include/uuid"])
+        uuid_incs = find_file("uuid.h", self.inc_dirs, ["/lib/include/uuid"])
         if uuid_incs is not None:
             if self.compiler.find_library_file(self.lib_dirs, 'uuid'):
                 uuid_libs = ['uuid']
@@ -2045,7 +2030,7 @@ class PyBuildExt(build_ext):
                 return
             # OS X 10.5 comes with libffi.dylib; the include files are
             # in /usr/include/ffi
-            ffi_inc_dirs.append('/usr/include/ffi')
+            ffi_inc_dirs.append('/lib/include/ffi')
 
         ffi_inc = [sysconfig.get_config_var("LIBFFI_INCLUDEDIR")]
         if not ffi_inc or ffi_inc[0] == '':
