diff --git a/misc-utils/logger.c b/misc-utils/logger.c
index bcdff061..59884482 100644
--- a/misc-utils/logger.c
+++ b/misc-utils/logger.c
@@ -426,7 +426,7 @@ static char const *rfc3164_current_time(void)
 	ret = logger_gettimeofday(&tv, NULL);
 	if (ret == -1)
 		err(EXIT_FAILURE, _("gettimeofday() failed"));
-	localtime_r(&tv.tv_sec, &tm);
+	gmtime_r(&tv.tv_sec, &tm);
 	snprintf(time, sizeof(time),"%s %2d %2.2d:%2.2d:%2.2d",
 		monthnames[tm.tm_mon], tm.tm_mday,
 		tm.tm_hour, tm.tm_min, tm.tm_sec);
