diff --git a/login-utils/sulogin.c b/login-utils/sulogin.c
index aeaa3f11..1b878b30 100644
--- a/login-utils/sulogin.c
+++ b/login-utils/sulogin.c
@@ -68,6 +68,8 @@
 #include "sulogin-consoles.h"
 #define CONMAX		16
 
+#define EXIT_TIMEOUT	142
+
 static unsigned int timeout;
 static int profile;
 static volatile uint32_t openfd;		/* Remember higher file descriptors */
@@ -665,18 +667,10 @@ static void doprompt(const char *crypted, struct console *con, int deny)
 				     "See sulogin(8) man page for more details.\n\n"
 				     "Press Enter to continue.\n"));
 	else {
-#if defined(USE_ONELINE)
 		if (crypted[0] && !locked_account_password(crypted))
-			fprintf(con->file, _("Enter root password for login: "));
+			fprintf(con->file, _("Enter root password for maintenance: "));
 		else
-			fprintf(con->file, _("Press Enter for login: "));
-#else
-		if (crypted[0] && !locked_account_password(crypted))
-			fprintf(con->file, _("Enter root password for system maintenance\n"));
-		else
-			fprintf(con->file, _("Press Enter for system maintenance\n"));
-		fprintf(con->file, _("(or press Control-D to continue): "));
-#endif
+			fprintf(con->file, _("Press Enter for maintenance: "));
 	}
 	fflush(con->file);
 err:
@@ -1000,11 +994,11 @@ int main(int argc, char **argv)
 	struct console *con;
 	char *tty = NULL;
 	struct passwd *pwd;
-	const struct timespec sigwait = { .tv_sec = 0, .tv_nsec = 50000000 };
 	siginfo_t status = { 0 };
 	sigset_t set;
 	int c, reconnect = 0;
 	int opt_e = 0;
+	int result = EXIT_TIMEOUT;
 	int wait = 0;
 	pid_t pid;
 
@@ -1233,12 +1227,11 @@ int main(int argc, char **argv)
 			}
 			if (alarm_raised) {
 				tcfinal(con);
-				warnx(_("Timed out\n\n"));
 			}
 			/*
 			 * User pressed Control-D.
 			 */
-			exit(0);
+			exit(EXIT_TIMEOUT);
 		case -1:
 			warn(_("fork failed"));
 			/* fallthrough */
@@ -1276,9 +1269,11 @@ int main(int argc, char **argv)
 			continue;
 		if (con->pid < 0)
 			continue;
-		if (con->pid == status.si_pid)
+		if (con->pid == status.si_pid) {
+			if (WIFEXITED(status.si_status))
+				result = WEXITSTATUS(status.si_status);
 			con->pid = -1;
-		else {
+		} else {
 			kill(con->pid, SIGTERM);
 			wait++;
 		}
@@ -1288,22 +1283,18 @@ int main(int argc, char **argv)
 	sigaddset(&set, SIGCHLD);
 
 	do {
-		int signum, ret;
+		int ret;
 
 		if (!wait)
 			break;
 
 		status.si_pid = 0;
-		ret = waitid(P_ALL, 0, &status, WEXITED|WNOHANG);
+		ret = waitid(P_ALL, 0, &status, WEXITED);
 
 		if (ret < 0) {
 			if (errno == ECHILD)
 				break;
-			if (errno == EINTR)
-				continue;
-		}
-
-		if (!ret && status.si_pid > 0) {
+		} else if (!ret && status.si_pid > 0) {
 			list_for_each(ptr, &consoles) {
 				con = list_entry(ptr, struct console, entry);
 
@@ -1312,17 +1303,14 @@ int main(int argc, char **argv)
 				if (con->pid < 0)
 					continue;
 				if (con->pid == status.si_pid) {
+					if (WIFEXITED(status.si_status) &&
+					    WEXITSTATUS(status.si_status) != EXIT_TIMEOUT)
+						result = WEXITSTATUS(status.si_status);
 					con->pid = -1;
 					wait--;
 				}
 			}
-			continue;
 		}
-
-		signum = sigtimedwait(&set, NULL, &sigwait);
-		if (signum != SIGCHLD && signum < 0 && errno == EAGAIN)
-			break;
-
 	} while (1);
 
 	mask_signal(SIGCHLD, SIG_DFL, NULL);
@@ -1331,5 +1319,5 @@ int main(int argc, char **argv)
 	tcreset_selinux(&consoles);
 #endif
 
-	return EXIT_SUCCESS;
+	return result;
 }
