diff --git a/src/netdev.c b/src/netdev.c
index 73fbf0c1..b0a8d9b5 100644
--- a/src/netdev.c
+++ b/src/netdev.c
@@ -190,6 +190,7 @@ struct netdev {
 	bool retry_auth : 1;
 	bool in_reassoc : 1;
 	bool privacy : 1;
+	bool bridged : 1;
 };
 
 struct netdev_preauth_state {
@@ -5722,6 +5723,7 @@ static void netdev_bridge_port_event(const struct ifinfomsg *ifi, int bytes,
 	netdev = netdev_find(ifi->ifi_index);
 	if (!netdev)
 		return;
+	netdev->bridged = added;
 
 	for (attr = IFLA_RTA(ifi); RTA_OK(attr, bytes);
 			attr = RTA_NEXT(attr, bytes)) {
@@ -5813,6 +5815,11 @@ bool netdev_get_4addr(struct netdev *netdev)
 	return netdev->use_4addr;
 }
 
+bool netdev_get_bridged(struct netdev *netdev)
+{
+	return netdev->bridged;
+}
+
 static void netdev_newlink_notify(const struct ifinfomsg *ifi, int bytes)
 {
 	struct netdev *netdev;
diff --git a/src/netdev.h b/src/netdev.h
index 6299934e..2042b37f 100644
--- a/src/netdev.h
+++ b/src/netdev.h
@@ -154,6 +154,7 @@ int netdev_set_4addr(struct netdev *netdev, bool use_4addr,
 			netdev_command_cb_t cb, void *user_data,
 			netdev_destroy_func_t destroy);
 bool netdev_get_4addr(struct netdev *netdev);
+bool netdev_get_bridged(struct netdev *netdev);
 const char *netdev_get_name(struct netdev *netdev);
 bool netdev_get_is_up(struct netdev *netdev);
 const char *netdev_get_path(struct netdev *netdev);
diff --git a/src/station.c b/src/station.c
index 30a1232a..819fe88e 100644
--- a/src/station.c
+++ b/src/station.c
@@ -4732,7 +4732,7 @@ static struct station *station_create(struct netdev *netdev)
 	l_dbus_object_add_interface(dbus, netdev_get_path(netdev),
 					IWD_STATION_INTERFACE, station);
 
-	if (netconfig_enabled())
+	if (netconfig_enabled() && !netdev_get_bridged(netdev))
 		station->netconfig = netconfig_new(netdev_get_ifindex(netdev));
 
 	station->anqp_pending = l_queue_new();
