diff --git a/ell/dbus-service.c b/ell/dbus-service.c
index 0300bfe3..50ef3c04 100644
--- a/ell/dbus-service.c
+++ b/ell/dbus-service.c
@@ -1098,7 +1098,7 @@ void _dbus_object_tree_signals_flush(struct l_dbus *dbus, const char *path)
 	struct emit_signals_data data;
 	bool all_done = true;
 
-	if (!tree->emit_signals_work || tree->flushing)
+	if (!tree || !tree->emit_signals_work || tree->flushing)
 		return;
 
 	tree->flushing = true;
@@ -1148,7 +1148,7 @@ static void schedule_emit_signals(struct l_dbus *dbus)
 {
 	struct _dbus_object_tree *tree = _dbus_get_tree(dbus);
 
-	if (tree->emit_signals_work)
+	if (!tree || tree->emit_signals_work)
 		return;
 
 	tree->emit_signals_work = l_idle_create(emit_signals, dbus, NULL);
@@ -1177,6 +1177,9 @@ bool _dbus_object_tree_property_changed(struct l_dbus *dbus,
 	struct _dbus_property *property;
 	struct _dbus_object_tree *tree = _dbus_get_tree(dbus);
 
+	if (!tree)
+		return false;
+
 	object = l_hashmap_lookup(tree->objects, path);
 	if (!object)
 		return false;
@@ -1280,6 +1283,9 @@ static struct l_dbus_message *old_set_property(struct l_dbus *dbus,
 	struct l_dbus_message *reply;
 	l_dbus_property_complete_cb_t complete_cb;
 
+	if (!tree)
+		return NULL;
+
 	interface = l_hashmap_lookup(tree->interfaces,
 					l_dbus_message_get_interface(message));
 	/* If we got here the interface must exist */
@@ -1329,6 +1335,9 @@ static struct l_dbus_message *old_get_properties(struct l_dbus *dbus,
 	struct l_dbus_message_builder *builder;
 	struct _dbus_object_tree *tree = _dbus_get_tree(dbus);
 
+	if (!tree)
+		return NULL;
+
 	interface = l_hashmap_lookup(tree->interfaces,
 					l_dbus_message_get_interface(message));
 	/* If we got here the interface must exist */
@@ -1849,6 +1858,9 @@ static struct l_dbus_message *properties_get(struct l_dbus *dbus,
 	struct l_dbus_message *reply;
 	struct l_dbus_message_builder *builder;
 
+	if (!tree)
+		return NULL;
+
 	if (!l_dbus_message_get_arguments(message, "ss", &interface_name,
 						&property_name))
 		return l_dbus_message_new_error(message,
@@ -1919,6 +1931,9 @@ static struct l_dbus_message *properties_set(struct l_dbus *dbus,
 	struct l_dbus_message *reply;
 	l_dbus_property_complete_cb_t complete_cb;
 
+	if (!tree)
+		return NULL;
+
 	if (!l_dbus_message_get_arguments(message, "ssv", &interface_name,
 						&property_name, &variant))
 		return l_dbus_message_new_error(message,
@@ -1987,6 +2002,9 @@ static struct l_dbus_message *properties_get_all(struct l_dbus *dbus,
 	struct l_dbus_message *reply;
 	struct l_dbus_message_builder *builder;
 
+	if (!tree)
+		return NULL;
+
 	if (!l_dbus_message_get_arguments(message, "s", &interface_name))
 		return l_dbus_message_new_error(message,
 						"org.freedesktop.DBus.Error."
@@ -2143,7 +2161,7 @@ static struct l_dbus_message *get_managed_objects(struct l_dbus *dbus,
 	struct _dbus_object_tree *tree = _dbus_get_tree(dbus);
 	const char *path = l_dbus_message_get_path(message);
 
-	return _dbus_object_tree_get_objects(tree, dbus, path, message);
+	return tree ? _dbus_object_tree_get_objects(tree, dbus, path, message) : NULL;
 }
 
 static void object_manager_setup_func(struct l_dbus_interface *interface)
diff --git a/ell/dbus.c b/ell/dbus.c
index bd6e1b8c..67b3fece 100644
--- a/ell/dbus.c
+++ b/ell/dbus.c
@@ -1528,7 +1528,7 @@ int _dbus_get_fd(struct l_dbus *dbus)
 
 struct _dbus_object_tree *_dbus_get_tree(struct l_dbus *dbus)
 {
-	return dbus->tree;
+	return dbus ? dbus->tree : NULL;
 }
 
 /**
@@ -1555,7 +1555,7 @@ LIB_EXPORT bool l_dbus_register_interface(struct l_dbus *dbus,
 				bool handle_old_style_properties)
 {
 	if (unlikely(!dbus))
-		return false;
+		return true;
 
 	if (unlikely(!dbus->tree))
 		return false;
@@ -1569,7 +1569,7 @@ LIB_EXPORT bool l_dbus_unregister_interface(struct l_dbus *dbus,
 						const char *interface)
 {
 	if (unlikely(!dbus))
-		return false;
+		return true;
 
 	if (unlikely(!dbus->tree))
 		return false;
@@ -1607,7 +1607,7 @@ LIB_EXPORT bool l_dbus_register_object(struct l_dbus *dbus, const char *path,
 	bool r = true;
 
 	if (unlikely(!dbus))
-		return false;
+		return true;
 
 	if (unlikely(!dbus->tree))
 		return false;
@@ -1637,7 +1637,7 @@ LIB_EXPORT bool l_dbus_unregister_object(struct l_dbus *dbus,
 						const char *object)
 {
 	if (unlikely(!dbus))
-		return false;
+		return true;
 
 	if (unlikely(!dbus->tree))
 		return false;
@@ -1670,7 +1670,7 @@ LIB_EXPORT bool l_dbus_object_add_interface(struct l_dbus *dbus,
 						void *user_data)
 {
 	if (unlikely(!dbus))
-		return false;
+		return true;
 
 	if (unlikely(!dbus->tree))
 		return false;
@@ -1684,7 +1684,7 @@ LIB_EXPORT bool l_dbus_object_remove_interface(struct l_dbus *dbus,
 						const char *interface)
 {
 	if (unlikely(!dbus))
-		return false;
+		return true;
 
 	if (unlikely(!dbus->tree))
 		return false;
@@ -1723,7 +1723,7 @@ LIB_EXPORT bool l_dbus_object_manager_enable(struct l_dbus *dbus,
 						const char *root)
 {
 	if (unlikely(!dbus))
-		return false;
+		return true;
 
 	if (unlikely(!dbus->tree))
 		return false;
@@ -1750,7 +1750,7 @@ LIB_EXPORT unsigned int l_dbus_add_service_watch(struct l_dbus *dbus,
 					void *user_data,
 					l_dbus_destroy_func_t destroy)
 {
-	if (!name)
+	if (!name || !dbus)
 		return 0;
 
 	if (!dbus->name_cache)
@@ -1764,7 +1764,7 @@ LIB_EXPORT unsigned int l_dbus_add_service_watch(struct l_dbus *dbus,
 
 LIB_EXPORT bool l_dbus_remove_watch(struct l_dbus *dbus, unsigned int id)
 {
-	if (!dbus->name_cache)
+	if (!dbus || !dbus->name_cache)
 		return false;
 
 	return _dbus_name_cache_remove_watch(dbus->name_cache, id);
@@ -1822,6 +1822,9 @@ LIB_EXPORT unsigned int l_dbus_add_signal_watch(struct l_dbus *dbus,
 	void *user_data;
 	unsigned int id;
 
+	if (!dbus)
+		return 0;
+
 	va_start(args, member);
 
 	rule_len = 0;
@@ -1895,7 +1898,7 @@ LIB_EXPORT unsigned int l_dbus_add_signal_watch(struct l_dbus *dbus,
 
 LIB_EXPORT bool l_dbus_remove_signal_watch(struct l_dbus *dbus, unsigned int id)
 {
-	if (!dbus->filter)
+	if (!dbus || !dbus->filter)
 		return false;
 
 	return _dbus_filter_remove_rule(dbus->filter, id);
@@ -1926,6 +1929,9 @@ LIB_EXPORT uint32_t l_dbus_name_acquire(struct l_dbus *dbus, const char *name,
 				bool queue, l_dbus_name_acquire_func_t callback,
 				void *user_data)
 {
+	if (!dbus)
+		return 0;
+
 	return dbus->driver->name_acquire(dbus, name, allow_replacement,
 						replace_existing, queue,
 						callback, user_data);
diff --git a/src/main.c b/src/main.c
index beaf228f..fbb4033a 100644
--- a/src/main.c
+++ b/src/main.c
@@ -599,14 +599,16 @@ int main(int argc, char *argv[])
 		l_netlink_set_debug(rtnl, do_debug, "[RTNL] ", NULL);
 
 	dbus = l_dbus_new_default(L_DBUS_SYSTEM_BUS);
-	if (!dbus) {
-		l_error("Failed to initialize D-Bus");
-		goto failed_dbus;
-	}
 
-	l_dbus_set_ready_handler(dbus, dbus_ready, dbus, NULL);
-	l_dbus_set_disconnect_handler(dbus, dbus_disconnected, NULL, NULL);
-	dbus_init(dbus);
+	if (dbus) {
+		l_dbus_set_ready_handler(dbus, dbus_ready, dbus, NULL);
+		l_dbus_set_disconnect_handler(dbus, dbus_disconnected, NULL, NULL);
+		dbus_init(dbus);
+	} else {
+		l_info("Running in embedded mode without D-Bus");
+		l_genl_request_family(genl, NL80211_GENL_NAME, nl80211_appeared,
+					NULL, NULL);
+	}
 
 	if (!setup_system_key())
 		goto failed_storage;
@@ -620,7 +622,6 @@ failed_storage:
 	dbus_exit();
 	l_dbus_destroy(dbus);
 
-failed_dbus:
 	l_netlink_destroy(rtnl);
 
 failed_rtnl:
