diff --git a/src/netconfig.c b/src/netconfig.c
index 5230702..4f9d9e1 100644
--- a/src/netconfig.c
+++ b/src/netconfig.c
@@ -752,11 +752,12 @@ void netconfig_destroy(struct netconfig *netconfig)
 
 bool netconfig_enabled(void)
 {
-	bool enabled;
+	bool enabled = true;
 
-	return l_settings_get_bool(iwd_get_config(), "General",
-					"EnableNetworkConfiguration",
-					&enabled) && enabled;
+	l_settings_get_bool(iwd_get_config(), "General",
+				"EnableNetworkConfiguration",
+				&enabled);
+	return enabled;
 }
 
 static int netconfig_init(void)
diff --git a/src/resolve.c b/src/resolve.c
index d674409..df6cb63 100644
--- a/src/resolve.c
+++ b/src/resolve.c
@@ -621,7 +621,7 @@ static int resolve_init(void)
 	method_name = l_settings_get_value(iwd_get_config(), "Network",
 						"NameResolvingService");
 	if (!method_name)
-		method_name = "systemd";
+		method_name = "resolvconf";
 
 	for (i = 0; resolve_method_ops_list[i].name; i++) {
 		if (strcmp(resolve_method_ops_list[i].name, method_name))
