diff --git a/src/netconfig.c b/src/netconfig.c
index 316431e..bd3f78e 100644
--- a/src/netconfig.c
+++ b/src/netconfig.c
@@ -1167,7 +1167,6 @@ struct netconfig *netconfig_new(uint32_t ifindex)
 
 	l_queue_push_tail(netconfig_list, netconfig);
 
-	sysfs_write_ipv6_setting(netdev_get_name(netdev), "accept_ra", "0");
 	sysfs_write_ipv6_setting(netdev_get_name(netdev), "disable_ipv6", "1");
 
 	return netconfig;
@@ -1233,7 +1232,7 @@ static int netconfig_init(void)
 	if (!l_settings_get_bool(iwd_get_config(), "Network",
 					"EnableIPv6",
 					&ipv6_enabled))
-		ipv6_enabled = false;
+		ipv6_enabled = true;
 
 	netconfig_list = l_queue_new();
 
