diff --git a/src/netconfig.c b/src/netconfig.c
index df9abd0..cff8dcc 100644
--- a/src/netconfig.c
+++ b/src/netconfig.c
@@ -1098,6 +1098,9 @@ static void netconfig_dhcp6_event_handler(struct l_dhcp6_client *client,
 	{
 		const struct l_dhcp6_lease *lease =
 			l_dhcp6_client_get_lease(netconfig->dhcp6_client);
+		if (!lease)
+			break;
+
 		_auto_(l_free) char *addr_str =
 			l_dhcp6_lease_get_address(lease);
 		struct l_rtnl_address *address;
@@ -1297,6 +1300,7 @@ static bool netconfig_ipv6_select_and_install(struct netconfig *netconfig)
 	}
 
 	sysfs_write_ipv6_setting(netdev_get_name(netdev), "disable_ipv6", "0");
+	sysfs_write_ipv6_setting(netdev_get_name(netdev), "accept_ra", "1");
 
 	if (netconfig->rtm_v6_protocol == RTPROT_DHCP &&
 			netconfig->fils_override &&
@@ -1738,7 +1742,7 @@ static int netconfig_init(void)
 	if (!l_settings_get_bool(iwd_get_config(), "Network",
 					"EnableIPv6",
 					&ipv6_enabled))
-		ipv6_enabled = false;
+		ipv6_enabled = true;
 
 	netconfig_list = l_queue_new();
 
