diff --git a/configure b/configure
index 68ae03ea..d4e89e40 100755
--- a/configure
+++ b/configure
@@ -13905,7 +13905,7 @@ wired_storagedir=${wired_storagedir}
 printf "%s\n" "#define DAEMON_STORAGEDIR \"${daemon_storagedir}\"" >>confdefs.h
 
 
-printf "%s\n" "#define DAEMON_CONFIGDIR \"/etc/iwd\"" >>confdefs.h
+printf "%s\n" "#define DAEMON_CONFIGDIR \"/etc\"" >>confdefs.h
 
 
 
diff --git a/configure.ac b/configure.ac
index 2d9f29cc..7fa18312 100644
--- a/configure.ac
+++ b/configure.ac
@@ -149,7 +149,7 @@ AC_SUBST([wired_storagedir], [${wired_storagedir}])
 
 AC_DEFINE_UNQUOTED(DAEMON_STORAGEDIR, "${daemon_storagedir}",
 			[Directory for Wireless daemon storage files])
-AC_DEFINE_UNQUOTED(DAEMON_CONFIGDIR, "/etc/iwd",
+AC_DEFINE_UNQUOTED(DAEMON_CONFIGDIR, "/etc",
 			[Directory for Wireless daemon configuration])
 
 AC_DEFINE_UNQUOTED(WIRED_STORAGEDIR, "${wired_storagedir}",
diff --git a/ell/dbus.c b/ell/dbus.c
index 67b3fece..12f8868f 100644
--- a/ell/dbus.c
+++ b/ell/dbus.c
@@ -31,7 +31,7 @@
 #include "useful.h"
 #include "dbus-private.h"
 
-#define DEFAULT_SYSTEM_BUS_ADDRESS "unix:path=/var/run/dbus/system_bus_socket"
+#define DEFAULT_SYSTEM_BUS_ADDRESS "unix:path=/run/dbus/system_bus_socket"
 
 #define DBUS_SERVICE_DBUS	"org.freedesktop.DBus"
 
diff --git a/ell/hwdb.c b/ell/hwdb.c
index 8c18f079..4260fb0c 100644
--- a/ell/hwdb.c
+++ b/ell/hwdb.c
@@ -147,9 +147,7 @@ LIB_EXPORT struct l_hwdb *l_hwdb_new_default(void)
 {
 	struct l_hwdb *db = NULL;
 	size_t i;
-	const char * const paths[] = {"/etc/udev/hwdb.bin",
-					"/usr/lib/udev/hwdb.bin",
-					"/lib/udev/hwdb.bin"};
+	const char * const paths[] = {"/etc/hwdb.bin", "/lib/hwdb.bin"};
 
 	for (i = 0; !db && i < L_ARRAY_SIZE(paths); i++)
 		db = l_hwdb_new(paths[i]);
diff --git a/src/iwd.config.rst b/src/iwd.config.rst
index 9dc0fe0c..3f5a3ab1 100644
--- a/src/iwd.config.rst
+++ b/src/iwd.config.rst
@@ -20,17 +20,17 @@ Configuration file for wireless daemon
 SYNOPSIS
 ========
 
-Configuration file **main.conf**
+Configuration file **iwd.conf**
 
 DESCRIPTION
 ===========
 
-The *main.conf* configuration file configures the system-wide settings for
+The *iwd.conf* configuration file configures the system-wide settings for
 **iwd**.  This file lives in the configuration directory specified by the
 environment variable *$CONFIGURATION_DIRECTORY*, which is normally provided
 by **systemd**.  In the absence of such an environment variable it defaults
-to */etc/iwd*.  If no *main.conf* is present, then default values are
-chosen.  The presence of *main.conf* is not required.
+to */etc*.  If no *iwd.conf* is present, then default values are
+chosen.  The presence of *iwd.conf* is not required.
 
 FILE FORMAT
 ===========
diff --git a/src/iwd.rst b/src/iwd.rst
index c719fbc2..ef8d2b45 100644
--- a/src/iwd.rst
+++ b/src/iwd.rst
@@ -48,7 +48,7 @@ ENVIRONMENT
 for details.
 
 *$CONFIGURATION_DIRECTORY* is used for daemon configuration settings.  Only
-the file **main.conf** is currently read from this directory.  See *iwd.config*
+the file **iwd.conf** is currently read from this directory.  See *iwd.config*
 for details.
 
 See *iwd.debug* for further environment variables for debugging.
diff --git a/src/main.c b/src/main.c
index fbb4033a..d03c181e 100644
--- a/src/main.c
+++ b/src/main.c
@@ -561,7 +561,7 @@ int main(int argc, char *argv[])
 
 	for (i = 0; config_dirs[i]; i++) {
 		L_AUTO_FREE_VAR(char *, path) =
-			l_strdup_printf("%s/%s", config_dirs[i], "main.conf");
+			l_strdup_printf("%s/%s", config_dirs[i], "iwd.conf");
 
 		if (!l_settings_load_from_file(iwd_config, path))
 			continue;
diff --git a/src/resolve.c b/src/resolve.c
index 4506d4b5..df6cb636 100644
--- a/src/resolve.c
+++ b/src/resolve.c
@@ -510,7 +510,7 @@ static struct resolve_ops resolvconf_ops = {
 
 static int resolve_resolvconf_init(void)
 {
-	static const char *default_path = "/sbin:/usr/sbin";
+	static const char *default_path = "/bin";
 	const char *path;
 
 	l_debug("Trying to find resolvconf in $PATH");
