diff --git a/src/resolve.c b/src/resolve.c
index f6e829d..4387c7f 100644
--- a/src/resolve.c
+++ b/src/resolve.c
@@ -553,8 +553,8 @@ static int resolve_init(void)
 		if (method_name)
 			l_warn("[General].dns_resolve_method is deprecated, "
 				"use [Network].NameResolvingService");
-		else /* Default to systemd-resolved service. */
-			method_name = "systemd";
+		else /* Default to resolvconf service. */
+			method_name = "resolvconf";
 	}
 
 	for (i = 0; resolve_method_ops_list[i].name; i++) {
