diff --git a/src/resolve.c b/src/resolve.c
index a01252e..fd7beed 100644
--- a/src/resolve.c
+++ b/src/resolve.c
@@ -596,7 +596,7 @@ static int resolve_init(void)
 	method_name = l_settings_get_value(iwd_get_config(), "Network",
 						"NameResolvingService");
 	if (!method_name)
-		method_name = "systemd";
+		method_name = "resolvconf";
 
 	for (i = 0; resolve_method_ops_list[i].name; i++) {
 		if (strcmp(resolve_method_ops_list[i].name, method_name))
