diff --git a/ell/log.c b/ell/log.c
index 05af3e5..3c621e5 100644
--- a/ell/log.c
+++ b/ell/log.c
@@ -31,6 +31,7 @@
 #include <stdarg.h>
 #include <string.h>
 #include <fnmatch.h>
+#include <syslog.h>
 #include <sys/socket.h>
 #include <sys/un.h>
 
@@ -171,6 +172,9 @@ static void log_syslog(int priority, const char *file, const char *line,
 	if (str_len < 0)
 		return;
 
+	if (LOG_FAC(priority) == LOG_KERN)
+		priority = LOG_DAEMON + LOG_PRI(priority);
+
 	hdr_len = snprintf(hdr, sizeof(hdr), "<%i>%s[%lu]: ", priority,
 					log_ident, (unsigned long) log_pid);
 
diff --git a/src/main.c b/src/main.c
index 8e530a1..6ad2cac 100644
--- a/src/main.c
+++ b/src/main.c
@@ -463,7 +463,8 @@ int main(int argc, char *argv[])
 		return EXIT_FAILURE;
 	}
 
-	l_log_set_stderr();
+	l_log_set_ident("iwd");
+	l_log_set_syslog();
 
 	if (check_crypto() < 0)
 		return EXIT_FAILURE;
diff --git a/src/wiphy.c b/src/wiphy.c
index 1051457..c9d5da3 100644
--- a/src/wiphy.c
+++ b/src/wiphy.c
@@ -1178,9 +1178,7 @@ static void wiphy_print_basic_info(struct wiphy *wiphy)
 			if (!str)
 				continue;
 
-			len += snprintf(buf + len, sizeof(buf) - len, "%s%s",
-					!n || (n % 4) ? " " : "\n\t\t ",
-					str);
+			len += snprintf(buf + len, sizeof(buf) - len, " %s", str);
 
 			n += 1;
 		}
