diff --git a/ell/log.c b/ell/log.c
index a8c26176..8834d68d 100644
--- a/ell/log.c
+++ b/ell/log.c
@@ -16,6 +16,7 @@
 #include <stdarg.h>
 #include <string.h>
 #include <fnmatch.h>
+#include <syslog.h>
 #include <sys/socket.h>
 #include <sys/un.h>
 
@@ -156,6 +157,9 @@ static void log_syslog(int priority, const char *file, const char *line,
 	if (str_len < 0)
 		return;
 
+	if (LOG_FAC(priority) == LOG_KERN)
+		priority = LOG_DAEMON + LOG_PRI(priority);
+
 	hdr_len = snprintf(hdr, sizeof(hdr), "<%i>%s[%lu]: ", priority,
 					log_ident, (unsigned long) log_pid);
 
diff --git a/src/main.c b/src/main.c
index d03c181e..b36d5e5e 100644
--- a/src/main.c
+++ b/src/main.c
@@ -525,12 +525,13 @@ int main(int argc, char *argv[])
 		return EXIT_FAILURE;
 	}
 
-	if (logger && !strcmp(logger, "syslog"))
-		l_log_set_syslog();
-	else if (logger && !strcmp(logger, "journal"))
-		l_log_set_journal();
-	else
+	if (logger && !strcmp(logger, "stderr")) {
 		l_log_set_stderr();
+	} else if (logger && !strcmp(logger, "journal")) {
+		l_log_set_journal();
+	} else {
+		l_log_set_syslog();
+	}
 
 	l_log_set_ident("iwd");
 
diff --git a/src/wiphy.c b/src/wiphy.c
index 06f72ef2..b1399600 100644
--- a/src/wiphy.c
+++ b/src/wiphy.c
@@ -1315,9 +1315,7 @@ static void wiphy_print_basic_info(struct wiphy *wiphy)
 			if (!str)
 				continue;
 
-			len += snprintf(buf + len, sizeof(buf) - len, "%s%s",
-					!n || (n % 4) ? " " : "\n\t\t ",
-					str);
+			len += snprintf(buf + len, sizeof(buf) - len, " %s", str);
 
 			n += 1;
 		}
