diff --git a/src/app/config/config.c b/src/app/config/config.c
index e40a583..851623a 100644
--- a/src/app/config/config.c
+++ b/src/app/config/config.c
@@ -6563,7 +6563,7 @@ parse_ports(or_options_t *options, int validate_only,
   if (port_parse_config(ports,
              options->SocksPort_lines,
              "Socks", CONN_TYPE_AP_LISTENER,
-             "127.0.0.1", 9050,
+             "127.0.0.1", 0,
              ((validate_only ? 0 : CL_PORT_WARN_NONLOCAL)
               | CL_PORT_TAKES_HOSTNAMES | gw_flag)) < 0) {
     *msg = tor_strdup("Invalid SocksPort configuration");
diff --git a/src/lib/thread/numcpus.c b/src/lib/thread/numcpus.c
index 40fac7d..d073344 100644
--- a/src/lib/thread/numcpus.c
+++ b/src/lib/thread/numcpus.c
@@ -70,12 +70,6 @@ compute_num_cpus_impl(void)
 #endif /* defined(_WIN32) || ... */
 }
 
-/** This is an arbitrary number but at this point in time, it is not that
- * uncommon to see servers up to that amount of CPUs. Most servers will likely
- * be around 16 to 32 cores now. Lets take advantage of large machines! The
- * "NumCPUs" torrc option overrides this maximum. */
-#define MAX_DETECTABLE_CPUS 128
-
 /** Return how many CPUs we are running with.  We assume that nobody is
  * using hot-swappable CPUs, so we don't recompute this after the first
  * time.  Return -1 if we don't know how to tell the number of CPUs on this
@@ -88,15 +82,6 @@ compute_num_cpus(void)
   if (num_cpus == -2) {
     num_cpus = compute_num_cpus_impl();
     tor_assert(num_cpus != -2);
-    if (num_cpus > MAX_DETECTABLE_CPUS) {
-      /* LCOV_EXCL_START */
-      log_notice(LD_GENERAL, "Wow!  I detected that you have %d CPUs. I "
-                 "will not autodetect any more than %d, though.  If you "
-                 "want to configure more, set NumCPUs in your torrc",
-                 num_cpus, MAX_DETECTABLE_CPUS);
-      num_cpus = MAX_DETECTABLE_CPUS;
-      /* LCOV_EXCL_STOP */
-    }
   }
   return num_cpus;
 }
