diff --git a/Makefile.in b/Makefile.in
index 0c97f163..e75805b1 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -169,9 +169,9 @@ install-lib: install-lib-$(ENABLE_LIB_STATIC)
 install: install-lib
 
 
-sqlite3$(T.exe):	shell.c sqlite3.c
+sqlite3$(T.exe):	shell.c $(libsqlite3.SO)
 	$(CC) -o $@ \
-		shell.c sqlite3.c \
+		shell.c $(libsqlite3.SO) \
 		-I. $(OPT_FEATURE_FLAGS) $(SHELL_OPT) \
 		$(CFLAGS) $(CFLAGS.readline) $(CFLAGS.icu) \
 		$(LDFLAGS) $(LDFLAGS.libsqlite3) $(LDFLAGS.readline)
