diff --git a/Makefile.def b/Makefile.def
index 36fd26b03..d61a0c481 100644
--- a/Makefile.def
+++ b/Makefile.def
@@ -66,7 +66,7 @@ host_modules= { module= mpc; lib_path=src/.libs; bootstrap=true;
 		extra_configure_flags='--disable-shared @extra_mpc_gmp_configure_flags@ @extra_mpc_mpfr_configure_flags@ --disable-maintainer-mode';
 		no_install= true; };
 host_modules= { module= isl; lib_path=.libs; bootstrap=true;
-		extra_configure_flags='--disable-shared @extra_isl_gmp_configure_flags@';
+		extra_configure_flags='--disable-shared @extra_isl_gmp_configure_flags@ ac_cv_have_decl___builtin_ffs=yes';
 		extra_make_flags='V=1';
 		no_install= true; };
 host_modules= { module= libelf; lib_path=.libs; bootstrap=true;
diff --git a/Makefile.in b/Makefile.in
index 36e369df6..cff3004e4 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -15975,7 +15975,7 @@ configure-isl:
 	  $$s/$$module_srcdir/configure \
 	  --srcdir=$${topdir}/$$module_srcdir \
 	  $(HOST_CONFIGARGS) --build=${build_alias} --host=${host_alias} \
-	  --target=${target_alias} --disable-shared @extra_isl_gmp_configure_flags@ \
+	  --target=${target_alias} --disable-shared @extra_isl_gmp_configure_flags@ ac_cv_have_decl___builtin_ffs=yes \
 	  || exit 1
 @endif isl
 
