diff --git a/Makefile.def b/Makefile.def
index 35e994eb7..1dc7303d7 100644
--- a/Makefile.def
+++ b/Makefile.def
@@ -67,7 +67,7 @@ host_modules= { module= mpc; lib_path=src/.libs; bootstrap=true;
 		extra_configure_flags='--disable-shared @extra_mpc_gmp_configure_flags@ @extra_mpc_mpfr_configure_flags@  @host_libs_picflag@ --disable-maintainer-mode';
 		no_install= true; };
 host_modules= { module= isl; lib_path=.libs; bootstrap=true;
-		extra_configure_flags='--disable-shared @extra_isl_gmp_configure_flags@  @host_libs_picflag@';
+		extra_configure_flags='--disable-shared @extra_isl_gmp_configure_flags@  @host_libs_picflag@ ac_cv_have_decl___builtin_ffs=yes';
 		extra_make_flags='V=1';
 		no_install= true; };
 host_modules= { module= gold; bootstrap=true; };
diff --git a/Makefile.in b/Makefile.in
index 06a9398e1..824520bc5 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -16567,7 +16567,7 @@ configure-isl:
 	  $$s/$$module_srcdir/configure \
 	  --srcdir=$${topdir}/$$module_srcdir \
 	  $(HOST_CONFIGARGS) --build=${build_alias} --host=${host_alias} \
-	  --target=${target_alias} --disable-shared @extra_isl_gmp_configure_flags@  @host_libs_picflag@ \
+	  --target=${target_alias} --disable-shared @extra_isl_gmp_configure_flags@  @host_libs_picflag@ ac_cv_have_decl___builtin_ffs=yes \
 	  || exit 1
 @endif isl
 
