diff --git a/gcc/system.h b/gcc/system.h
index cf45db3f9..39c83edde 100644
--- a/gcc/system.h
+++ b/gcc/system.h
@@ -43,6 +43,7 @@ along with GCC; see the file COPYING3.  If not see
 # include <stddef.h>
 #endif
 
+#include <pthread.h>
 #include <stdio.h>
 
 /* Define a generic NULL if one hasn't already been defined.  */
diff --git a/libvtv/vtv_rts.cc b/libvtv/vtv_rts.cc
index 4c0d2fc16..d37e6c30c 100644
--- a/libvtv/vtv_rts.cc
+++ b/libvtv/vtv_rts.cc
@@ -125,7 +125,7 @@
 #include <windows.h>
 #include <winternl.h>
 #include <psapi.h>
-#else
+#elif defined (__GLIBC__)
 #include <execinfo.h>
 #endif
 
diff --git a/libvtv/vtv_utils.cc b/libvtv/vtv_utils.cc
index d40b9c6a6..a58eea82b 100644
--- a/libvtv/vtv_utils.cc
+++ b/libvtv/vtv_utils.cc
@@ -35,7 +35,7 @@
 #if defined (__CYGWIN__) || defined (__MINGW32__)
 #define WIN32_LEAN_AND_MEAN
 #include <windows.h>
-#else
+#elif defined (__GLIBC__)
 #include <execinfo.h>
 #endif
 
@@ -179,7 +179,7 @@ __vtv_log_verification_failure (const char *log_msg, bool generate_backtrace)
 
   __vtv_add_to_log (vtv_failures_log_fd, "%s", log_msg);
 
-#if !defined (__CYGWIN__) && !defined (__MINGW32__)
+#if !defined (__CYGWIN__) && !defined (__MINGW32__) && defined(__GLIBC__)
   if (generate_backtrace)
     {
 #define STACK_DEPTH 20
