diff --git a/bu/ld/elf-hints-local.h b/bu/ld/elf-hints-local.h
index 9cdfa2f2..fa6eb443 100644
--- a/bu/ld/elf-hints-local.h
+++ b/bu/ld/elf-hints-local.h
@@ -41,6 +41,6 @@ struct elfhints_hdr
 
 #define ELFHINTS_MAGIC	0x746e6845
 
-#define _PATH_ELF_HINTS	"/var/run/ld-elf.so.hints"
+#define _PATH_ELF_HINTS	"/run/ld-elf.so.hints"
 
 #endif /* !_ELF_HINTS_H_ */
diff --git a/gcc/gcc.cc b/gcc/gcc.cc
index 4fd87f2c..e5042f66 100644
--- a/gcc/gcc.cc
+++ b/gcc/gcc.cc
@@ -1611,10 +1611,10 @@ static const char *gcc_libexec_prefix;
 /* Default prefixes to attach to command names.  */
 
 #ifndef STANDARD_STARTFILE_PREFIX_1
-#define STANDARD_STARTFILE_PREFIX_1 "/lib/"
+#define STANDARD_STARTFILE_PREFIX_1 "/lib/static/"
 #endif
 #ifndef STANDARD_STARTFILE_PREFIX_2
-#define STANDARD_STARTFILE_PREFIX_2 "/usr/lib/"
+#define STANDARD_STARTFILE_PREFIX_2 ""
 #endif
 
 #ifdef CROSS_DIRECTORY_STRUCTURE  /* Don't use these prefixes for a cross compiler.  */
diff --git a/libffi/src/closures.c b/libffi/src/closures.c
index f7bead67..352b7e33 100644
--- a/libffi/src/closures.c
+++ b/libffi/src/closures.c
@@ -702,7 +702,6 @@ static struct
   { open_temp_exec_file_env, "LIBFFI_TMPDIR", 0 },
   { open_temp_exec_file_env, "TMPDIR", 0 },
   { open_temp_exec_file_dir, "/tmp", 0 },
-  { open_temp_exec_file_dir, "/var/tmp", 0 },
   { open_temp_exec_file_dir, "/dev/shm", 0 },
   { open_temp_exec_file_env, "HOME", 0 },
 #ifdef HAVE_MNTENT
diff --git a/libiberty/make-temp-file.c b/libiberty/make-temp-file.c
index 01aa8cc7..d09f66af 100644
--- a/libiberty/make-temp-file.c
+++ b/libiberty/make-temp-file.c
@@ -96,8 +96,6 @@ try_dir (const char *dir, const char *base)
 }
 
 static const char tmp[] = { DIR_SEPARATOR, 't', 'm', 'p', 0 };
-static const char vartmp[] =
-{ DIR_SEPARATOR, 'v', 'a', 'r', DIR_SEPARATOR, 't', 'm', 'p', 0 };
 
 #endif
 
@@ -142,8 +140,7 @@ choose_tmpdir (void)
 	base = try_dir (P_tmpdir, base);
 #endif
 
-      /* Try /var/tmp, then /tmp.  */
-      base = try_dir (vartmp, base);
+      /* Try /tmp.  */
       base = try_dir (tmp, base);
       
       /* If all else fails, use the current directory!  */
diff --git a/libiberty/tmpnam.c b/libiberty/tmpnam.c
index cc343336..a1e65abe 100644
--- a/libiberty/tmpnam.c
+++ b/libiberty/tmpnam.c
@@ -18,7 +18,7 @@ not be used in new projects.  Use @code{mkstemp} instead.
 #define L_tmpnam 100
 #endif
 #ifndef P_tmpdir
-#define P_tmpdir "/usr/tmp"
+#define P_tmpdir "/tmp"
 #endif
 
 static char tmpnam_buffer[L_tmpnam];
diff --git a/libtool.m4 b/libtool.m4
index add2d4a1..bb80d650 100644
--- a/libtool.m4
+++ b/libtool.m4
@@ -2529,7 +2529,7 @@ linux* | k*bsd*-gnu | kopensolaris*-gnu | gnu* | uclinuxfdpiceabi)
   need_version=no
   library_names_spec='${libname}${release}${shared_ext}$versuffix ${libname}${release}${shared_ext}$major $libname${shared_ext}'
   soname_spec='${libname}${release}${shared_ext}$major'
-  finish_cmds='PATH="\$PATH:/sbin" ldconfig -n $libdir'
+  finish_cmds='ldconfig -n $libdir'
   shlibpath_var=LD_LIBRARY_PATH
   shlibpath_overrides_runpath=no
 
