diff --git a/gcc/gcc.cc b/gcc/gcc.cc
index e5042f66..af3ec361 100644
--- a/gcc/gcc.cc
+++ b/gcc/gcc.cc
@@ -1164,6 +1164,7 @@ proper position among the other output files.  */
    "%{flto|flto=*:%<fcompare-debug*} \
     %{flto} %{fno-lto} %{flto=*} %l " LINK_PIE_SPEC \
    "%{fuse-ld=*:-fuse-ld=%*} " LINK_COMPRESS_DEBUG_SPEC \
+   "-z relro -z now" \
    "%X %{o*} %{e*} %{N} %{n} %{r}\
     %{s} %{t} %{u*} %{z} %{Z} %{!nostdlib:%{!r:%{!nostartfiles:%S}}} \
     %{static|no-pie|static-pie:} %@{L*} %(link_libgcc) " \
