diff --git a/readpass.c b/readpass.c
index 974d67f..7be25a6 100644
--- a/readpass.c
+++ b/readpass.c
@@ -159,8 +159,6 @@ read_passphrase(const char *prompt, int flags)
 	if (use_askpass && getenv("DISPLAY")) {
 		if (getenv(SSH_ASKPASS_ENV))
 			askpass = getenv(SSH_ASKPASS_ENV);
-		else
-			askpass = _PATH_SSH_ASKPASS_DEFAULT;
 		if ((flags & RP_ASK_PERMISSION) != 0)
 			askpass_hint = "confirm";
 		if ((ret = ssh_askpass(askpass, prompt, askpass_hint)) == NULL)
