diff --git a/magic/Magdir/clojure b/magic/Magdir/clojure
index 1f1cddf9..0a1e62bb 100644
--- a/magic/Magdir/clojure
+++ b/magic/Magdir/clojure
@@ -3,18 +3,30 @@
 # URL:  https://clojure.org/
 # From: Jason Felice <jason.m.felice@gmail.com>
 
+0	string/w	#!\ /bin/clj	Clojure script text executable
+!:mime	text/x-clojure
 0	string/w	#!\ /usr/bin/clj	Clojure script text executable
 !:mime	text/x-clojure
 0	string/w	#!\ /usr/local/bin/clj	Clojure script text executable
 !:mime	text/x-clojure
+0	string/w	#!\ /bin/clojure	Clojure script text executable
+!:mime	text/x-clojure
 0	string/w	#!\ /usr/bin/clojure	Clojure script text executable
 !:mime	text/x-clojure
 0	string/w	#!\ /usr/local/bin/clojure	Clojure script text executable
 !:mime	text/x-clojure
+0	string/W	#!/bin/env\ clj	Clojure script text executable
+!:mime	text/x-clojure
+0	string/W	#!/bin/env\ clojure	Clojure script text executable
+!:mime	text/x-clojure
 0	string/W	#!/usr/bin/env\ clj	Clojure script text executable
 !:mime	text/x-clojure
 0	string/W	#!/usr/bin/env\ clojure	Clojure script text executable
 !:mime	text/x-clojure
+0	string/W	#!\ /bin/env\ clj	Clojure script text executable
+!:mime	text/x-clojure
+0	string/W	#!\ /bin/env\ clojure	Clojure script text executable
+!:mime	text/x-clojure
 0	string/W	#!\ /usr/bin/env\ clj	Clojure script text executable
 !:mime	text/x-clojure
 0	string/W	#!\ /usr/bin/env\ clojure	Clojure script text executable
diff --git a/magic/Magdir/commands b/magic/Magdir/commands
index 4ce97766..3465caf1 100644
--- a/magic/Magdir/commands
+++ b/magic/Magdir/commands
@@ -37,6 +37,8 @@
 !:mime	text/x-shellscript
 0	string/fwt	#!\ /usr/local/bin/zsh	Paul Falstad's zsh script text executable
 !:mime	text/x-shellscript
+0	string/fwt	#!\ /bin/env\ zsh	Paul Falstad's zsh script text executable
+!:mime	text/x-shellscript
 0	string/fwt	#!\ /usr/bin/env\ zsh	Paul Falstad's zsh script text executable
 !:mime	text/x-shellscript
 
@@ -114,6 +116,8 @@
 !:mime	text/x-shellscript
 0	string/fwt	#!\ /usr/bin/env\ fish		fish shell script text executable
 !:mime	text/x-shellscript
+0	string/fwt	#!\ /bin/env\ fish		fish shell script text executable
+!:mime	text/x-shellscript
 
 0	search/1/fwt	#!\ /usr/bin/tclsh	Tcl/Tk script text executable
 !:mime	text/x-tcl
@@ -149,6 +153,8 @@
 !:mime	text/x-execline
 0	string/fwt	#!\ /usr/bin/execlineb		execline script text executable
 !:mime	text/x-execline
+0	string/fwt	#!\ /bin/env\ execlineb		execline script text executable
+!:mime	text/x-execline
 0	string/fwt	#!\ /usr/bin/env\ execlineb		execline script text executable
 !:mime	text/x-execline
 
@@ -174,6 +180,9 @@
 0	search/1/w	#!\ /usr/bin/php	PHP script text executable
 !:strength + 55
 !:mime	text/x-php
+0	search/1/w	#!\ /bin/php	PHP script text executable
+!:strength + 10
+!:mime	text/x-php
 # Smarty compiled template, https://www.smarty.net/
 # Elan Ruusamae <glen@delfi.ee>
 0	string	=<?php
@@ -215,6 +224,7 @@
 # Type: Pdmenu
 # URL:	https://packages.debian.org/pdmenu
 # From: Edward Betts <edward@debian.org>
+0	string		#!/bin/pdmenu	Pdmenu configuration file text
 0	string		#!/usr/bin/pdmenu	Pdmenu configuration file text
 
 # From Danny Weldon
@@ -228,6 +238,9 @@
 >0x10  byte	   x	   version %d
 >>0x11 byte	   x	  \b.%d
 
+0	string/fwt	#!\ /bin/env\ runghc	GHC script executable
 0	string/fwt	#!\ /usr/bin/env\ runghc	GHC script executable
+0	string/fwt	#!\ /bin/env\ runhaskell	Haskell script executable
 0	string/fwt	#!\ /usr/bin/env\ runhaskell	Haskell script executable
+0	string/fwt	#!\ /bin/env\ julia Julia script executable
 0	string/fwt	#!\ /usr/bin/env\ julia Julia script executable
diff --git a/magic/Magdir/javascript b/magic/Magdir/javascript
index 85d4a70b..0d1a6b2e 100644
--- a/magic/Magdir/javascript
+++ b/magic/Magdir/javascript
@@ -11,8 +11,12 @@
 !:mime application/javascript
 0	string/tw	#!/usr/bin/nodejs	Node.js script executable
 !:mime application/javascript
+0	string/t		#!/bin/env\ node	Node.js script executable
+!:mime application/javascript
 0	string/t		#!/usr/bin/env\ node	Node.js script executable
 !:mime application/javascript
+0	string/t		#!/bin/env\ nodejs	Node.js script executable
+!:mime application/javascript
 0	string/t		#!/usr/bin/env\ nodejs	Node.js script executable
 !:mime application/javascript
 
diff --git a/magic/Magdir/lua b/magic/Magdir/lua
index 859862a3..4ab70789 100644
--- a/magic/Magdir/lua
+++ b/magic/Magdir/lua
@@ -6,10 +6,16 @@
 # From: Reuben Thomas <rrt@sc3d.org>, Seo Sanghyeon <tinuviel@sparcs.kaist.ac.kr>
 
 # Lua scripts
+0	search/1/w	#!\ /bin/lua	Lua script text executable
+!:mime	text/x-lua
 0	search/1/w	#!\ /usr/bin/lua	Lua script text executable
 !:mime	text/x-lua
 0	search/1/w	#!\ /usr/local/bin/lua	Lua script text executable
 !:mime	text/x-lua
+0	search/1	#!/bin/env\ lua	Lua script text executable
+!:mime	text/x-lua
+0	search/1	#!\ /bin/env\ lua	Lua script text executable
+!:mime	text/x-lua
 0	search/1	#!/usr/bin/env\ lua	Lua script text executable
 !:mime	text/x-lua
 0	search/1	#!\ /usr/bin/env\ lua	Lua script text executable
diff --git a/magic/Magdir/perl b/magic/Magdir/perl
index 4a3756a4..837e1e29 100644
--- a/magic/Magdir/perl
+++ b/magic/Magdir/perl
@@ -23,6 +23,10 @@
 !:mime	text/x-perl
 0	search/1024	eval\ '(exit\ $?0)'\ &&\ eval\ 'exec	Perl script text
 !:mime	text/x-perl
+0	string	#!/bin/env\ perl	Perl script text executable
+!:mime	text/x-perl
+0	string	#!\ /bin/env\ perl	Perl script text executable
+!:mime	text/x-perl
 0	string	#!/usr/bin/env\ perl	Perl script text executable
 !:mime	text/x-perl
 0	string	#!\ /usr/bin/env\ perl	Perl script text executable
diff --git a/magic/Magdir/python b/magic/Magdir/python
index ede1b5ba..679b88d5 100644
--- a/magic/Magdir/python
+++ b/magic/Magdir/python
@@ -247,12 +247,18 @@
 !:mime application/x-bytecode.python
 >>>0		use		pyc-pep552
 
+0	search/1/w	#!\040/bin/python	Python script text executable
+!:strength + 30
+!:mime text/x-script.python
 0	search/1/w	#!\040/usr/bin/python	Python script text executable
 !:strength + 30
 !:mime text/x-script.python
 0	search/1/w	#!\040/usr/local/bin/python	Python script text executable
 !:strength + 30
 !:mime text/x-script.python
+0	search/10/w	#!\040/bin/env\040python	Python script text executable
+!:strength + 30
+!:mime text/x-script.python
 0	search/10/w	#!\040/usr/bin/env\040python	Python script text executable
 !:strength + 30
 !:mime text/x-script.python
diff --git a/magic/Magdir/ruby b/magic/Magdir/ruby
index 9e67a3e2..2c5b32db 100644
--- a/magic/Magdir/ruby
+++ b/magic/Magdir/ruby
@@ -6,12 +6,21 @@
 # From: Reuben Thomas <rrt@sc3d.org>
 
 # Ruby scripts
+0	search/1/w	#!\ /bin/ruby				Ruby script text executable
+!:strength + 15
+!:mime text/x-ruby
 0	search/1/w	#!\ /usr/bin/ruby				Ruby script text executable
 !:strength + 15
 !:mime text/x-ruby
 0	search/1/w	#!\ /usr/local/bin/ruby	Ruby script text executable
 !:strength + 15
 !:mime text/x-ruby
+0	search/1	#!/bin/env\ ruby				Ruby script text executable
+!:strength + 15
+!:mime text/x-ruby
+0	search/1	#!\ /bin/env\ ruby			Ruby script text executable
+!:strength + 15
+!:mime text/x-ruby
 0	search/1	#!/usr/bin/env\ ruby				Ruby script text executable
 !:strength + 15
 !:mime text/x-ruby
diff --git a/magic/Magdir/tcl b/magic/Magdir/tcl
index edc3ec42..ca614e37 100644
--- a/magic/Magdir/tcl
+++ b/magic/Magdir/tcl
@@ -4,18 +4,30 @@
 # From: gustaf neumann
 
 # Tcl scripts
+0	search/1/w	#!\ /bin/tcl	Tcl script text executable
+!:mime	text/x-tcl
 0	search/1/w	#!\ /usr/bin/tcl	Tcl script text executable
 !:mime	text/x-tcl
 0	search/1/w	#!\ /usr/local/bin/tcl	Tcl script text executable
 !:mime	text/x-tcl
+0	search/1	#!/bin/env\ tcl	Tcl script text executable
+!:mime	text/x-tcl
+0	search/1	#!\ /bin/env\ tcl	Tcl script text executable
+!:mime	text/x-tcl
 0	search/1	#!/usr/bin/env\ tcl	Tcl script text executable
 !:mime	text/x-tcl
 0	search/1	#!\ /usr/bin/env\ tcl	Tcl script text executable
 !:mime	text/x-tcl
+0	search/1/w	#!\ /bin/wish	Tcl/Tk script text executable
+!:mime	text/x-tcl
 0	search/1/w	#!\ /usr/bin/wish	Tcl/Tk script text executable
 !:mime	text/x-tcl
 0	search/1/w	#!\ /usr/local/bin/wish	Tcl/Tk script text executable
 !:mime	text/x-tcl
+0	search/1	#!/bin/env\ wish	Tcl/Tk script text executable
+!:mime	text/x-tcl
+0	search/1	#!\ /bin/env\ wish	Tcl/Tk script text executable
+!:mime	text/x-tcl
 0	search/1	#!/usr/bin/env\ wish	Tcl/Tk script text executable
 !:mime	text/x-tcl
 0	search/1	#!\ /usr/bin/env\ wish	Tcl/Tk script text executable
diff --git a/magic/Magdir/varied.script b/magic/Magdir/varied.script
index a6e2d95e..e52fadea 100644
--- a/magic/Magdir/varied.script
+++ b/magic/Magdir/varied.script
@@ -15,19 +15,35 @@
 
 
 # using env
+0	regex/t		#![[:space:]]*/bin/env[[:space:]]*	a
+>&0	string/T	>\0			%s script text executable
+!:strength / 2
+
 0	regex/t		#![[:space:]]*/usr/bin/env[[:space:]]*	a
 >&0	string/T	>\0			%s script text executable
 !:strength / 2
 
+0	regex/b		#![[:space:]]*/bin/env[[:space:]]*	a
+>&0	string/T	>\0			%s script executable (binary data)
+!:strength / 2
+
 0	regex/b		#![[:space:]]*/usr/bin/env[[:space:]]*	a
 >&0	string/T	>\0			%s script executable (binary data)
 !:strength / 2
 
 # using env
+0	regex/t		#![[:space:]]*/bin/env[[:space:]]*-[Si0]	a
+>&0	string/T	>\0			%s script text executable
+!:strength / 2
+
 0	regex/t		#![[:space:]]*/usr/bin/env[[:space:]]*-[Si0]	a
 >&0	string/T	>\0			%s script text executable
 !:strength / 2
 
+0	regex/b		#![[:space:]]*/bin/env[[:space:]]*-[Si0]	a
+>&0	string/T	>\0			%s script executable (binary data)
+!:strength / 2
+
 0	regex/b		#![[:space:]]*/usr/bin/env[[:space:]]*-[Si0]	a
 >&0	string/T	>\0			%s script executable (binary data)
 !:strength / 2
