diff --git a/configure b/configure
index 5147c791..0661123e 100755
--- a/configure
+++ b/configure
@@ -11273,7 +11273,7 @@ else
 fi
 
 fi
-if test -n "$PKG_CONFIG"; then
+if test -n "$PKG_CONFIG" && test "$PKG_CONFIG" != "false"; then
 	_pkg_min_version=0.9.0
 	{ printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking pkg-config is at least version $_pkg_min_version" >&5
 printf %s "checking pkg-config is at least version $_pkg_min_version... " >&6; }
