diff --git a/libdm/dm-tools/Makefile.in b/libdm/dm-tools/Makefile.in
index 9ddb3c2..d0983c3 100644
--- a/libdm/dm-tools/Makefile.in
+++ b/libdm/dm-tools/Makefile.in
@@ -56,10 +56,10 @@ LIBDM_LIBS = -L$(interfacebuilddir) -ldevmapper
 LIBDM_SHARED = $(interfacebuilddir)/libdevmapper.so
 LIBDM_STATIC = $(interfacebuilddir)/libdevmapper.a
 
-dmsetup: dmsetup.o $(LIBDM_SHARED)
+dmsetup: dmsetup.o $(LIBDM_STATIC)
 	@echo "    [CC] $@"
 	$(Q) $(CC) $(CFLAGS) $(LDFLAGS) $(EXTRA_EXEC_LDFLAGS) $(ELDFLAGS) \
-	      -o $@ $< $(LIBDM_LIBS) $(LIBS)
+	      -o $@ $< $(LIBDM_STATIC) $(LIBS)
 
 dmsetup.static: dmsetup.o $(LIBDM_STATIC)
 	@echo "    [CC] $@"
@@ -77,10 +77,10 @@ install_dmsetup_static: dmsetup.static
 	$(Q) $(LN_S) -f $(<F) $(staticdir)/dmstats.static
 
 
-dmfilemapd: dmfilemapd.o $(LIBDM_SHARED)
+dmfilemapd: dmfilemapd.o $(LIBDM_STATIC)
 	@echo "    [CC] $@"
 	$(Q) $(CC) $(CFLAGS) $(LDFLAGS) $(EXTRA_EXEC_LDFLAGS) $(ELDFLAGS) \
-	      -o $@ $< $(LIBDM_LIBS) $(LIBS)
+	      -o $@ $< $(LIBDM_STATIC) $(LIBS)
 
 dmfilemapd.static: dmfilemapd.o $(LIBDM_STATIC)
 	@echo "    [CC] $@"
