diff --git a/usr.sbin/smtpd/mta_session.c b/usr.sbin/smtpd/mta_session.c
index 327502b..d70ca9d 100644
--- a/usr.sbin/smtpd/mta_session.c
+++ b/usr.sbin/smtpd/mta_session.c
@@ -654,7 +654,7 @@ again:
 		else if (s->relay->secret) {
 			log_debug("debug: mta: %p: not using AUTH on non-TLS "
 			    "session", s);
-			mta_error(s, "Refuse to AUTH over unsecure channel");
+			mta_error(s, "Refuse to AUTH over insecure channel");
 			mta_connect(s);
 		} else {
 			mta_enter_state(s, MTA_READY);
diff --git a/usr.sbin/smtpd/smtpd.c b/usr.sbin/smtpd/smtpd.c
index 190f754..f91e5db 100644
--- a/usr.sbin/smtpd/smtpd.c
+++ b/usr.sbin/smtpd/smtpd.c
@@ -1882,7 +1882,7 @@ parent_forward_open(char *username, char *directory, uid_t uid, gid_t gid)
 	}
 
 	if (!secure_file(fd, pathname, directory, uid, 1)) {
-		log_warnx("warn: smtpd: %s: unsecure file", pathname);
+		log_warnx("warn: smtpd: %s: insecure file", pathname);
 		close(fd);
 		return -1;
 	}
