diff --git a/include/sched.h b/include/sched.h
index 204c34f5..9acb8d6a 100644
--- a/include/sched.h
+++ b/include/sched.h
@@ -90,7 +90,7 @@ int sched_getcpu(void);
 int sched_getaffinity(pid_t, size_t, cpu_set_t *);
 int sched_setaffinity(pid_t, size_t, const cpu_set_t *);
 
-#define __CPU_op_S(i, size, set, op) ( (i)/8U >= (size) ? 0 : \
+#define __CPU_op_S(i, size, set, op) ( (i)/8UL >= (size) ? 0 : \
 	(((unsigned long *)(set))[(i)/8/sizeof(long)] op (1UL<<((i)%(8*sizeof(long))))) )
 
 #define CPU_SET_S(i, size, set) __CPU_op_S(i, size, set, |=)
diff --git a/include/sys/socket.h b/include/sys/socket.h
index 6dc1e40a..a6117321 100644
--- a/include/sys/socket.h
+++ b/include/sys/socket.h
@@ -355,7 +355,7 @@ struct linger {
 
 #define CMSG_DATA(cmsg) ((unsigned char *) (((struct cmsghdr *)(cmsg)) + 1))
 #define CMSG_NXTHDR(mhdr, cmsg) ((cmsg)->cmsg_len < sizeof (struct cmsghdr) || \
-	__CMSG_LEN(cmsg) + sizeof(struct cmsghdr) >= __MHDR_END(mhdr) - (unsigned char *)(cmsg) \
+	__CMSG_LEN(cmsg) + sizeof(struct cmsghdr) >= (size_t) (__MHDR_END(mhdr) - (unsigned char *)(cmsg)) \
 	? 0 : (struct cmsghdr *)__CMSG_NEXT(cmsg))
 #define CMSG_FIRSTHDR(mhdr) ((size_t) (mhdr)->msg_controllen >= sizeof (struct cmsghdr) ? (struct cmsghdr *) (mhdr)->msg_control : (struct cmsghdr *) 0)
 
