diff --git a/e_os.h b/e_os.h
index 79c1392..3ba8350 100644
--- a/e_os.h
+++ b/e_os.h
@@ -85,7 +85,7 @@ extern "C" {
 /* set this to a comma-seperated list of 'egd' sockets to try out. These
  * sockets will be tried in the order listed in case accessing the device files
  * listed in DEVRANDOM did not return enough entropy. */
-#define DEVRANDOM_EGD "/var/run/egd-pool","/dev/egd-pool","/etc/egd-pool","/etc/entropy"
+#define DEVRANDOM_EGD "/run/egd-pool","/dev/egd-pool","/etc/egd-pool","/etc/entropy"
 #endif
 
 #if defined(OPENSSL_SYS_VXWORKS)
