diff --git a/init.c b/init.c
index c84815c2..256bb0a0 100644
--- a/init.c
+++ b/init.c
@@ -3692,15 +3692,12 @@ static char* mutt_find_cfg (const char *home, const char *xdg_cfg_home)
 {
   const char* names[] =
     {
-      "muttrc-" MUTT_VERSION,
       "muttrc",
       NULL,
     };
 
   const char* locations[][2] =
     {
-      { home, ".", },
-      { home, ".mutt/" },
       { xdg_cfg_home, "mutt/", },
       { NULL, NULL },
     };
@@ -3854,7 +3851,7 @@ void mutt_init (int skip_sys_rc, LIST *commands)
   else
   {
     /* Default search path from RFC1524 */
-    MailcapPath = safe_strdup ("~/.mailcap:" PKGDATADIR "/mailcap:" SYSCONFDIR "/mailcap:/etc/mailcap:/usr/etc/mailcap:/usr/local/etc/mailcap");
+    MailcapPath = safe_strdup ("~/.config/mutt/mailcap:" PKGDATADIR "/mailcap:" SYSCONFDIR "/mailcap");
   }
 
   Tempdir = safe_strdup ((p = getenv ("TMPDIR")) ? p : "/tmp");
@@ -3963,13 +3960,9 @@ void mutt_init (int skip_sys_rc, LIST *commands)
      requested not to via "-n".  */
   if (!skip_sys_rc)
   {
-    mutt_buffer_printf (buffer, "%s/Muttrc-%s", SYSCONFDIR, MUTT_VERSION);
+    mutt_buffer_printf (buffer, "%s/muttrc", SYSCONFDIR);
     if (access (mutt_b2s (buffer), F_OK) == -1)
-      mutt_buffer_printf (buffer, "%s/Muttrc", SYSCONFDIR);
-    if (access (mutt_b2s (buffer), F_OK) == -1)
-      mutt_buffer_printf (buffer, "%s/Muttrc-%s", PKGDATADIR, MUTT_VERSION);
-    if (access (mutt_b2s (buffer), F_OK) == -1)
-      mutt_buffer_printf (buffer, "%s/Muttrc", PKGDATADIR);
+      mutt_buffer_printf (buffer, "%s/muttrc", PKGDATADIR);
     if (access (mutt_b2s (buffer), F_OK) != -1)
     {
       if (source_rc (mutt_b2s (buffer), &err) != 0)
diff --git a/init.h b/init.h
index 5e15a3e3..1b8306ef 100644
--- a/init.h
+++ b/init.h
@@ -1546,7 +1546,7 @@ struct option_t MuttVars[] = {
   ** the string history buffer per category. The buffer is cleared each time the
   ** variable is set.
   */
-  { "history_file",     DT_PATH, R_NONE, {.p=&HistFile}, {.p="~/.mutthistory"} },
+  { "history_file",     DT_PATH, R_NONE, {.p=&HistFile}, {.p=0} },
   /*
   ** .pp
   ** The file in which Mutt will save its history.
