diff --git a/src/bootstrap/config.rs b/src/bootstrap/config.rs
index e192cda9a..5721d8cf5 100644
--- a/src/bootstrap/config.rs
+++ b/src/bootstrap/config.rs
@@ -1803,9 +1803,9 @@ impl Config {
         let source_version =
             Version::parse(&fs::read_to_string(self.src.join("src/version")).unwrap().trim())
                 .unwrap();
-        if !(source_version == rustc_version
-            || (source_version.major == rustc_version.major
-                && source_version.minor == rustc_version.minor + 1))
+        if !(source_version.major == rustc_version.major
+            && (source_version.minor == rustc_version.minor
+                || source_version.minor == rustc_version.minor + 1))
         {
             let prev_version = format!("{}.{}.x", source_version.major, source_version.minor - 1);
             eprintln!(
