diff --git a/vendor/curl-sys-0.4.79+curl-8.12.0/build.rs b/vendor/curl-sys-0.4.79+curl-8.12.0/build.rs
index 8315a20f..34c6d3e2 100644
--- a/vendor/curl-sys-0.4.79+curl-8.12.0/build.rs
+++ b/vendor/curl-sys-0.4.79+curl-8.12.0/build.rs
@@ -35,6 +35,11 @@ fn main() {
             return println!("cargo:rustc-flags=-l curl");
         }
 
+        // Force the system libcurl on Linux
+        if target.contains("linux") {
+            return println!("cargo:rustc-flags=-l curl");
+        }
+
         // Next, fall back and try to use pkg-config if its available.
         if windows {
             if try_vcpkg() {
diff --git a/vendor/curl-sys-0.4.83+curl-8.15.0/build.rs b/vendor/curl-sys-0.4.83+curl-8.15.0/build.rs
index 433a6c51..1d602c5b 100644
--- a/vendor/curl-sys-0.4.83+curl-8.15.0/build.rs
+++ b/vendor/curl-sys-0.4.83+curl-8.15.0/build.rs
@@ -35,6 +35,11 @@ fn main() {
             return println!("cargo:rustc-flags=-l curl");
         }
 
+        // Force the system libcurl on Linux
+        if target.contains("linux") {
+            return println!("cargo:rustc-flags=-l curl");
+        }
+
         // Next, fall back and try to use pkg-config if its available.
         if windows {
             if try_vcpkg() {
diff --git a/vendor/curl-sys-0.4.84+curl-8.17.0/build.rs b/vendor/curl-sys-0.4.84+curl-8.17.0/build.rs
index b74c0859..a599d296 100644
--- a/vendor/curl-sys-0.4.84+curl-8.17.0/build.rs
+++ b/vendor/curl-sys-0.4.84+curl-8.17.0/build.rs
@@ -35,6 +35,11 @@ fn main() {
             return println!("cargo:rustc-flags=-l curl");
         }
 
+        // Force the system libcurl on Linux
+        if target.contains("linux") {
+            return println!("cargo:rustc-flags=-l curl");
+        }
+
         // Next, fall back and try to use pkg-config if its available.
         if windows {
             if try_vcpkg() {
