diff --git a/vendor/curl-sys-0.4.52+curl-7.81.0/build.rs b/vendor/curl-sys-0.4.52+curl-7.81.0/build.rs
index e9a2a42d..789372f3 100644
--- a/vendor/curl-sys-0.4.52+curl-7.81.0/build.rs
+++ b/vendor/curl-sys-0.4.52+curl-7.81.0/build.rs
@@ -31,6 +31,11 @@ fn main() {
             return println!("cargo:rustc-flags=-l curl");
         }
 
+        // Force the system libcurl on Linux
+        if target.contains("linux") {
+            return println!("cargo:rustc-flags=-l curl");
+        }
+
         // Next, fall back and try to use pkg-config if its available.
         if windows {
             if try_vcpkg() {
diff --git a/vendor/curl-sys-0.4.74+curl-8.9.0/build.rs b/vendor/curl-sys-0.4.74+curl-8.9.0/build.rs
index 7d9a661f..24bc69d6 100644
--- a/vendor/curl-sys-0.4.74+curl-8.9.0/build.rs
+++ b/vendor/curl-sys-0.4.74+curl-8.9.0/build.rs
@@ -27,6 +27,11 @@ fn main() {
             return println!("cargo:rustc-flags=-l curl");
         }
 
+        // Force the system libcurl on Linux
+        if target.contains("linux") {
+            return println!("cargo:rustc-flags=-l curl");
+        }
+
         // Next, fall back and try to use pkg-config if its available.
         if windows {
             if try_vcpkg() {
