diff --git a/vendor/curl-sys/build.rs b/vendor/curl-sys/build.rs
index 332fe9ead..72394155a 100644
--- a/vendor/curl-sys/build.rs
+++ b/vendor/curl-sys/build.rs
@@ -35,6 +35,11 @@ fn main() {
             return println!("cargo:rustc-flags=-l curl");
         }
 
+        // Force the system libcurl on Linux
+        if target.contains("linux") {
+            return println!("cargo:rustc-flags=-l curl");
+        }
+
         // Next, fall back and try to use pkg-config if its available.
         if windows {
             if try_vcpkg() {
