diff --git a/src/bootstrap/dist.rs b/src/bootstrap/dist.rs
index 12585e80e..b986f6c13 100644
--- a/src/bootstrap/dist.rs
+++ b/src/bootstrap/dist.rs
@@ -394,7 +394,7 @@ impl Step for Rustc {
                     target: compiler.host,
                 })
                 .expect("rust-analyzer-proc-macro-server always builds");
-            builder.install(&ra_proc_macro_srv, &image.join("libexec"), 0o755);
+            builder.install(&ra_proc_macro_srv, &image.join("bin"), 0o755);
 
             let libdir_relative = builder.libdir_relative(compiler);
 
diff --git a/src/tools/rust-analyzer/crates/rust-analyzer/src/reload.rs b/src/tools/rust-analyzer/crates/rust-analyzer/src/reload.rs
index e1f651786..77d2719a9 100644
--- a/src/tools/rust-analyzer/crates/rust-analyzer/src/reload.rs
+++ b/src/tools/rust-analyzer/crates/rust-analyzer/src/reload.rs
@@ -324,7 +324,7 @@ impl GlobalState {
                                 if let Some(sysroot) = sysroot.as_ref() {
                                     let server_path = sysroot
                                         .root()
-                                        .join("libexec")
+                                        .join("bin")
                                         .join(&standalone_server_name);
                                     if std::fs::metadata(&server_path).is_ok() {
                                         tracing::debug!(
