diff --git a/src/bootstrap/dist.rs b/src/bootstrap/dist.rs
index 3cb0eccd3..b9cfea670 100644
--- a/src/bootstrap/dist.rs
+++ b/src/bootstrap/dist.rs
@@ -403,7 +403,7 @@ impl Step for Rustc {
                     target: compiler.host,
                 })
                 .expect("rust-analyzer-proc-macro-server always builds");
-            builder.install(&ra_proc_macro_srv, &image.join("libexec"), 0o755);
+            builder.install(&ra_proc_macro_srv, &image.join("bin"), 0o755);
 
             let libdir_relative = builder.libdir_relative(compiler);
 
@@ -1046,7 +1046,7 @@ impl Step for Cargo {
         for dirent in fs::read_dir(cargo.parent().unwrap()).expect("read_dir") {
             let dirent = dirent.expect("read dir entry");
             if dirent.file_name().to_str().expect("utf8").starts_with("cargo-credential-") {
-                tarball.add_file(&dirent.path(), "libexec", 0o755);
+                tarball.add_file(&dirent.path(), "bin", 0o755);
             }
         }
 
diff --git a/src/bootstrap/tool.rs b/src/bootstrap/tool.rs
index e0be4c432..112003cf2 100644
--- a/src/bootstrap/tool.rs
+++ b/src/bootstrap/tool.rs
@@ -783,7 +783,7 @@ impl Step for RustAnalyzerProcMacroSrv {
 
         // Copy `rust-analyzer-proc-macro-srv` to `<sysroot>/libexec/`
         // so that r-a can use it.
-        let libexec_path = builder.sysroot(self.compiler).join("libexec");
+        let libexec_path = builder.sysroot(self.compiler).join("bin");
         t!(fs::create_dir_all(&libexec_path));
         builder.copy(&path, &libexec_path.join("rust-analyzer-proc-macro-srv"));
 
diff --git a/src/tools/cargo/src/cargo/util/auth.rs b/src/tools/cargo/src/cargo/util/auth.rs
index d67f874f1..b4c1dd8e8 100644
--- a/src/tools/cargo/src/cargo/util/auth.rs
+++ b/src/tools/cargo/src/cargo/util/auth.rs
@@ -476,7 +476,7 @@ fn sysroot_credential(
         .parent()
         .and_then(|p| p.parent())
         .ok_or_else(|| format_err!("expected cargo path {}", cargo.display()))?;
-    let exe = root.join("libexec").join(format!(
+    let exe = root.join("bin").join(format!(
         "cargo-credential-{}{}",
         cred_name,
         std::env::consts::EXE_SUFFIX
diff --git a/src/tools/rust-analyzer/crates/project-model/src/workspace.rs b/src/tools/rust-analyzer/crates/project-model/src/workspace.rs
index 3d199ed24..bd61d5965 100644
--- a/src/tools/rust-analyzer/crates/project-model/src/workspace.rs
+++ b/src/tools/rust-analyzer/crates/project-model/src/workspace.rs
@@ -383,7 +383,7 @@ impl ProjectWorkspace {
             | ProjectWorkspace::Json { sysroot: Some(sysroot), .. } => {
                 let standalone_server_name =
                     format!("rust-analyzer-proc-macro-srv{}", std::env::consts::EXE_SUFFIX);
-                ["libexec", "lib"]
+                ["bin", "lib"]
                     .into_iter()
                     .map(|segment| sysroot.root().join(segment).join(&standalone_server_name))
                     .find(|server_path| std::fs::metadata(&server_path).is_ok())
