diff --git a/vendor/openssl-sys/build/cfgs.rs b/vendor/openssl-sys/build/cfgs.rs
index 18b3e3deb..c1d9ddd5a 100644
--- a/vendor/openssl-sys/build/cfgs.rs
+++ b/vendor/openssl-sys/build/cfgs.rs
@@ -34,6 +34,12 @@ pub fn get(openssl_version: Option<u64>, libressl_version: Option<u64>) -> Vec<&
         if libressl_version >= 0x3_03_02_00_0 {
             cfgs.push("libressl332");
         }
+        if libressl_version >= 0x3_04_00_00_0 {
+            cfgs.push("libressl340");
+        }
+        if libressl_version >= 0x3_05_00_00_0 {
+            cfgs.push("libressl350");
+        }
     } else {
         let openssl_version = openssl_version.unwrap();
 
diff --git a/vendor/openssl-sys/build/main.rs b/vendor/openssl-sys/build/main.rs
index 88752d995..83b5119f4 100644
--- a/vendor/openssl-sys/build/main.rs
+++ b/vendor/openssl-sys/build/main.rs
@@ -237,6 +237,11 @@ See rust-openssl README for more information:
             (3, 3, 0) => ('3', '3', '0'),
             (3, 3, 1) => ('3', '3', '1'),
             (3, 3, _) => ('3', '3', 'x'),
+            (3, 4, 0) => ('3', '4', '0'),
+            (3, 4, 1) => ('3', '4', '1'),
+            (3, 4, _) => ('3', '4', 'x'),
+            (3, 5, 0) => ('3', '5', '0'),
+            (3, 5, _) => ('3', '5', 'x'),
             _ => version_error(),
         };
 
diff --git a/vendor/openssl/build.rs b/vendor/openssl/build.rs
index 688c96240..b1dbfda1b 100644
--- a/vendor/openssl/build.rs
+++ b/vendor/openssl/build.rs
@@ -74,5 +74,13 @@ fn main() {
         if version >= 0x3_03_02_00_0 {
             println!("cargo:rustc-cfg=libressl332");
         }
+
+        if version >= 0x3_04_00_00_0 {
+            println!("cargo:rustc-cfg=libressl340");
+        }
+
+        if version >= 0x3_05_00_00_0 {
+            println!("cargo:rustc-cfg=libressl350");
+        }
     }
 }
