diff --git a/vendor/libssh2-sys/libssh2/src/openssl.c b/vendor/libssh2-sys/libssh2/src/openssl.c
index 7a6810f13..4ce73b9d0 100644
--- a/vendor/libssh2-sys/libssh2/src/openssl.c
+++ b/vendor/libssh2-sys/libssh2/src/openssl.c
@@ -653,8 +653,7 @@ _libssh2_EVP_aes_256_ctr(void)
 
 void _libssh2_openssl_crypto_init(void)
 {
-#if OPENSSL_VERSION_NUMBER >= 0x10100000L && \
-    !defined(LIBRESSL_VERSION_NUMBER)
+#if OPENSSL_VERSION_NUMBER >= 0x10100000L
 #ifndef OPENSSL_NO_ENGINE
     ENGINE_load_builtin_engines();
     ENGINE_register_all_complete();
diff --git a/vendor/libssh2-sys/libssh2/src/openssl.h b/vendor/libssh2-sys/libssh2/src/openssl.h
index 658b040d6..2f1879f85 100644
--- a/vendor/libssh2-sys/libssh2/src/openssl.h
+++ b/vendor/libssh2-sys/libssh2/src/openssl.h
@@ -57,8 +57,7 @@
 #include <openssl/pem.h>
 #include <openssl/rand.h>
 
-#if OPENSSL_VERSION_NUMBER >= 0x10100000L && \
-    !defined(LIBRESSL_VERSION_NUMBER)
+#if OPENSSL_VERSION_NUMBER >= 0x10100000L
 # define HAVE_OPAQUE_STRUCTS 1
 #endif
 
diff --git a/vendor/openssl-sys/build/cfgs.rs b/vendor/openssl-sys/build/cfgs.rs
index 18b3e3deb..c1ab5f485 100644
--- a/vendor/openssl-sys/build/cfgs.rs
+++ b/vendor/openssl-sys/build/cfgs.rs
@@ -34,6 +34,18 @@ pub fn get(openssl_version: Option<u64>, libressl_version: Option<u64>) -> Vec<&
         if libressl_version >= 0x3_03_02_00_0 {
             cfgs.push("libressl332");
         }
+        if libressl_version >= 0x3_04_00_00_0 {
+            cfgs.push("libressl340");
+        }
+        if libressl_version >= 0x3_05_00_00_0 {
+            cfgs.push("libressl350");
+        }
+        if libressl_version >= 0x3_06_00_00_0 {
+            cfgs.push("libressl360");
+        }
+        if libressl_version >= 0x3_07_00_00_0 {
+            cfgs.push("libressl370");
+        }
     } else {
         let openssl_version = openssl_version.unwrap();
 
diff --git a/vendor/openssl-sys/build/main.rs b/vendor/openssl-sys/build/main.rs
index 243f146bf..cfed44073 100644
--- a/vendor/openssl-sys/build/main.rs
+++ b/vendor/openssl-sys/build/main.rs
@@ -254,6 +254,12 @@ See rust-openssl README for more information:
             (3, 3, _) => ('3', '3', 'x'),
             (3, 4, 0) => ('3', '4', '0'),
             (3, 4, _) => ('3', '4', 'x'),
+            (3, 5, 0) => ('3', '5', '0'),
+            (3, 5, _) => ('3', '5', 'x'),
+            (3, 6, 0) => ('3', '6', '0'),
+            (3, 6, _) => ('3', '6', 'x'),
+            (3, 7, 0) => ('3', '7', '0'),
+            (3, 7, _) => ('3', '7', 'x'),
             _ => version_error(),
         };
 
@@ -296,7 +302,7 @@ fn version_error() -> ! {
         "
 
 This crate is only compatible with OpenSSL (version 1.0.1 through 1.1.1, or 3.0.0), or LibreSSL 2.5
-through 3.4.1, but a different version of OpenSSL was found. The build is now aborting
+through 3.7.x, but a different version of OpenSSL was found. The build is now aborting
 due to this version mismatch.
 
 "
diff --git a/vendor/openssl/build.rs b/vendor/openssl/build.rs
index d90ebc17d..6073c290c 100644
--- a/vendor/openssl/build.rs
+++ b/vendor/openssl/build.rs
@@ -78,5 +78,15 @@ fn main() {
         if version >= 0x3_04_00_00_0 {
             println!("cargo:rustc-cfg=libressl340");
         }
+
+        if version >= 0x3_05_00_00_0 {
+            println!("cargo:rustc-cfg=libressl350");
+        }
+        if version >= 0x3_06_00_00_0 {
+            println!("cargo:rustc-cfg=libressl360");
+        }
+        if version >= 0x3_07_00_00_0 {
+            println!("cargo:rustc-cfg=libressl370");
+        }
     }
 }
