diff --git a/vendor/openssl-0.10.38/build.rs b/vendor/openssl-0.10.38/build.rs
index d90ebc17..3e323cea 100644
--- a/vendor/openssl-0.10.38/build.rs
+++ b/vendor/openssl-0.10.38/build.rs
@@ -78,5 +78,41 @@ fn main() {
         if version >= 0x3_04_00_00_0 {
             println!("cargo:rustc-cfg=libressl340");
         }
+
+        if version >= 0x3_05_00_00_0 {
+            println!("cargo:rustc-cfg=libressl350");
+        }
+
+        if version >= 0x3_06_00_00_0 {
+            println!("cargo:rustc-cfg=libressl360");
+        }
+
+        if version >= 0x3_06_01_00_0 {
+            println!("cargo:rustc-cfg=libressl361");
+        }
+
+        if version >= 0x3_07_00_00_0 {
+            println!("cargo:rustc-cfg=libressl370");
+        }
+
+        if version >= 0x3_08_00_00_0 {
+            println!("cargo:rustc-cfg=libressl380");
+        }
+
+        if version >= 0x3_08_02_00_0 {
+            println!("cargo:rustc-cfg=libressl382");
+        }
+
+        if version >= 0x3_09_00_00_0 {
+            println!("cargo:rustc-cfg=libressl390");
+        }
+
+        if version >= 0x4_00_00_00_0 {
+            println!("cargo:rustc-cfg=libressl400");
+        }
+
+        if version >= 0x4_01_00_00_0 {
+            println!("cargo:rustc-cfg=libressl410");
+        }
     }
 }
diff --git a/vendor/openssl-0.10.64/build.rs b/vendor/openssl-0.10.64/build.rs
index 7677abc0..4f5ca600 100644
--- a/vendor/openssl-0.10.64/build.rs
+++ b/vendor/openssl-0.10.64/build.rs
@@ -75,6 +75,12 @@ fn main() {
         if version >= 0x3_09_00_00_0 {
             println!("cargo:rustc-cfg=libressl390");
         }
+        if version >= 0x4_00_00_00_0 {
+            println!("cargo:rustc-cfg=libressl400");
+        }
+        if version >= 0x4_01_00_00_0 {
+            println!("cargo:rustc-cfg=libressl410");
+        }
     }
 
     if let Ok(vars) = env::var("DEP_OPENSSL_CONF") {
diff --git a/vendor/openssl-0.10.64/src/conf.rs b/vendor/openssl-0.10.64/src/conf.rs
index 715519c5..a4261cbc 100644
--- a/vendor/openssl-0.10.64/src/conf.rs
+++ b/vendor/openssl-0.10.64/src/conf.rs
@@ -8,7 +8,7 @@ foreign_type_and_impl_send_sync! {
     pub struct ConfRef;
 }
 
-#[cfg(not(boringssl))]
+#[cfg(not(any(boringssl, libressl400)))]
 mod methods {
     use super::Conf;
     use crate::cvt_p;
@@ -61,5 +61,5 @@ mod methods {
         }
     }
 }
-#[cfg(not(boringssl))]
+#[cfg(not(and(boringssl, libressl400)))]
 pub use methods::*;
diff --git a/vendor/openssl-0.10.71/build.rs b/vendor/openssl-0.10.71/build.rs
index 52f3e89f..ef2645fe 100644
--- a/vendor/openssl-0.10.71/build.rs
+++ b/vendor/openssl-0.10.71/build.rs
@@ -32,6 +32,7 @@ fn main() {
     println!("cargo:rustc-check-cfg=cfg(libressl382)");
     println!("cargo:rustc-check-cfg=cfg(libressl390)");
     println!("cargo:rustc-check-cfg=cfg(libressl400)");
+    println!("cargo:rustc-check-cfg=cfg(libressl410)");
 
     println!("cargo:rustc-check-cfg=cfg(ossl101)");
     println!("cargo:rustc-check-cfg=cfg(ossl102)");
@@ -116,6 +117,9 @@ fn main() {
         if version >= 0x4_00_00_00_0 {
             println!("cargo:rustc-cfg=libressl400");
         }
+        if version >= 0x4_01_00_00_0 {
+            println!("cargo:rustc-cfg=libressl410");
+        }
     }
 
     if let Ok(vars) = env::var("DEP_OPENSSL_CONF") {
diff --git a/vendor/openssl-sys-0.9.102/build/cfgs.rs b/vendor/openssl-sys-0.9.102/build/cfgs.rs
index 91cb734c..c53eabc4 100644
--- a/vendor/openssl-sys-0.9.102/build/cfgs.rs
+++ b/vendor/openssl-sys-0.9.102/build/cfgs.rs
@@ -65,6 +65,12 @@ pub fn get(openssl_version: Option<u64>, libressl_version: Option<u64>) -> Vec<&
         if libressl_version >= 0x3_09_00_00_0 {
             cfgs.push("libressl390");
         }
+        if libressl_version >= 0x4_00_00_00_0 {
+            cfgs.push("libressl400");
+        }
+        if libressl_version >= 0x4_01_00_00_0 {
+            cfgs.push("libressl410");
+        }
     } else {
         let openssl_version = openssl_version.unwrap();
 
diff --git a/vendor/openssl-sys-0.9.102/build/main.rs b/vendor/openssl-sys-0.9.102/build/main.rs
index 606acc35..a30f788b 100644
--- a/vendor/openssl-sys-0.9.102/build/main.rs
+++ b/vendor/openssl-sys-0.9.102/build/main.rs
@@ -304,6 +304,10 @@ See rust-openssl documentation for more information:
             (3, 8, _) => ('3', '8', 'x'),
             (3, 9, 0) => ('3', '9', '0'),
             (3, 9, _) => ('3', '9', 'x'),
+            (4, 0, 0) => ('4', '0', '0'),
+            (4, 0, _) => ('4', '0', 'x'),
+            (4, 1, 0) => ('4', '1', '0'),
+            (4, 1, _) => ('4', '1', 'x'),
             _ => version_error(),
         };
 
@@ -346,7 +350,7 @@ fn version_error() -> ! {
         "
 
 This crate is only compatible with OpenSSL (version 1.0.1 through 1.1.1, or 3), or LibreSSL 2.5
-through 3.9.x, but a different version of OpenSSL was found. The build is now aborting
+through 4.1.x, but a different version of OpenSSL was found. The build is now aborting
 due to this version mismatch.
 
 "
diff --git a/vendor/openssl-sys-0.9.104/build/cfgs.rs b/vendor/openssl-sys-0.9.104/build/cfgs.rs
index cd03888e..ca997074 100644
--- a/vendor/openssl-sys-0.9.104/build/cfgs.rs
+++ b/vendor/openssl-sys-0.9.104/build/cfgs.rs
@@ -68,6 +68,9 @@ pub fn get(openssl_version: Option<u64>, libressl_version: Option<u64>) -> Vec<&
         if libressl_version >= 0x4_00_00_00_0 {
             cfgs.push("libressl400");
         }
+        if libressl_version >= 0x4_01_00_00_0 {
+            cfgs.push("libressl410");
+        }
     } else {
         let openssl_version = openssl_version.unwrap();
 
diff --git a/vendor/openssl-sys-0.9.104/build/main.rs b/vendor/openssl-sys-0.9.104/build/main.rs
index f379e1e6..0429f12a 100644
--- a/vendor/openssl-sys-0.9.104/build/main.rs
+++ b/vendor/openssl-sys-0.9.104/build/main.rs
@@ -103,6 +103,7 @@ fn main() {
     println!("cargo:rustc-check-cfg=cfg(libressl382)");
     println!("cargo:rustc-check-cfg=cfg(libressl390)");
     println!("cargo:rustc-check-cfg=cfg(libressl400)");
+    println!("cargo:rustc-check-cfg=cfg(libressl410)");
 
     println!("cargo:rustc-check-cfg=cfg(ossl101)");
     println!("cargo:rustc-check-cfg=cfg(ossl102)");
@@ -382,6 +383,8 @@ See rust-openssl documentation for more information:
             (3, 9, _) => ('3', '9', 'x'),
             (4, 0, 0) => ('4', '0', '0'),
             (4, 0, _) => ('4', '0', 'x'),
+            (4, 1, 0) => ('4', '1', '0'),
+            (4, 1, _) => ('4', '1', 'x'),
             _ => version_error(),
         };
 
@@ -424,7 +427,7 @@ fn version_error() -> ! {
         "
 
 This crate is only compatible with OpenSSL (version 1.0.1 through 1.1.1, or 3), or LibreSSL 2.5
-through 4.0.x, but a different version of OpenSSL was found. The build is now aborting
+through 4.1.x, but a different version of OpenSSL was found. The build is now aborting
 due to this version mismatch.
 
 "
diff --git a/vendor/openssl-sys-0.9.106/build/main.rs b/vendor/openssl-sys-0.9.106/build/main.rs
index 1a59d6f3..40c722e2 100644
--- a/vendor/openssl-sys-0.9.106/build/main.rs
+++ b/vendor/openssl-sys-0.9.106/build/main.rs
@@ -383,6 +383,8 @@ See rust-openssl documentation for more information:
             (3, 9, _) => ('3', '9', 'x'),
             (4, 0, 0) => ('4', '0', '0'),
             (4, 0, _) => ('4', '0', 'x'),
+            (4, 1, 0) => ('4', '1', '0'),
+            (4, 1, _) => ('4', '1', 'x'),
             _ => version_error(),
         };
 
@@ -425,7 +427,7 @@ fn version_error() -> ! {
         "
 
 This crate is only compatible with OpenSSL (version 1.0.1 through 1.1.1, or 3), or LibreSSL 2.5
-through 4.0.x, but a different version of OpenSSL was found. The build is now aborting
+through 4.1.x, but a different version of OpenSSL was found. The build is now aborting
 due to this version mismatch.
 
 "
diff --git a/vendor/openssl-sys-0.9.72/build/cfgs.rs b/vendor/openssl-sys-0.9.72/build/cfgs.rs
index 18b3e3de..b2b234a9 100644
--- a/vendor/openssl-sys-0.9.72/build/cfgs.rs
+++ b/vendor/openssl-sys-0.9.72/build/cfgs.rs
@@ -34,6 +34,36 @@ pub fn get(openssl_version: Option<u64>, libressl_version: Option<u64>) -> Vec<&
         if libressl_version >= 0x3_03_02_00_0 {
             cfgs.push("libressl332");
         }
+        if libressl_version >= 0x3_04_00_00_0 {
+            cfgs.push("libressl340");
+        }
+        if libressl_version >= 0x3_05_00_00_0 {
+            cfgs.push("libressl350");
+        }
+        if libressl_version >= 0x3_06_00_00_0 {
+            cfgs.push("libressl360");
+        }
+        if libressl_version >= 0x3_07_00_00_0 {
+            cfgs.push("libressl370");
+        }
+        if libressl_version >= 0x3_08_00_00_0 {
+            cfgs.push("libressl380");
+        }
+        if libressl_version >= 0x3_08_01_00_0 {
+            cfgs.push("libressl381");
+        }
+        if libressl_version >= 0x3_08_02_00_0 {
+            cfgs.push("libressl382");
+        }
+        if libressl_version >= 0x3_09_00_00_0 {
+            cfgs.push("libressl390");
+        }
+        if libressl_version >= 0x4_00_00_00_0 {
+            cfgs.push("libressl400");
+        }
+        if libressl_version >= 0x4_01_00_00_0 {
+            cfgs.push("libressl410");
+        }
     } else {
         let openssl_version = openssl_version.unwrap();
 
diff --git a/vendor/openssl-sys-0.9.72/build/main.rs b/vendor/openssl-sys-0.9.72/build/main.rs
index 243f146b..8aacb403 100644
--- a/vendor/openssl-sys-0.9.72/build/main.rs
+++ b/vendor/openssl-sys-0.9.72/build/main.rs
@@ -254,6 +254,21 @@ See rust-openssl README for more information:
             (3, 3, _) => ('3', '3', 'x'),
             (3, 4, 0) => ('3', '4', '0'),
             (3, 4, _) => ('3', '4', 'x'),
+            (3, 5, _) => ('3', '5', 'x'),
+            (3, 6, 0) => ('3', '6', '0'),
+            (3, 6, _) => ('3', '6', 'x'),
+            (3, 7, 0) => ('3', '7', '0'),
+            (3, 7, 1) => ('3', '7', '1'),
+            (3, 7, _) => ('3', '7', 'x'),
+            (3, 8, 0) => ('3', '8', '0'),
+            (3, 8, 1) => ('3', '8', '1'),
+            (3, 8, _) => ('3', '8', 'x'),
+            (3, 9, 0) => ('3', '9', '0'),
+            (3, 9, _) => ('3', '9', 'x'),
+            (4, 0, 0) => ('4', '0', '0'),
+            (4, 0, _) => ('4', '0', 'x'),
+            (4, 1, 0) => ('4', '1', '0'),
+            (4, 1, _) => ('4', '1', 'x'),
             _ => version_error(),
         };
 
@@ -296,7 +311,7 @@ fn version_error() -> ! {
         "
 
 This crate is only compatible with OpenSSL (version 1.0.1 through 1.1.1, or 3.0.0), or LibreSSL 2.5
-through 3.4.1, but a different version of OpenSSL was found. The build is now aborting
+through 4.1.x, but a different version of OpenSSL was found. The build is now aborting
 due to this version mismatch.
 
 "
