diff --git a/src/bootstrap/tool.rs b/src/bootstrap/tool.rs
index 1108a8d8d..ce60db4f2 100644
--- a/src/bootstrap/tool.rs
+++ b/src/bootstrap/tool.rs
@@ -584,10 +584,12 @@ impl Step for Cargo {
                 "src/tools/cargo/credential/cargo-credential-macos-keychain",
             );
         }
+        /*
         build_cred(
             "cargo-credential-1password",
             "src/tools/cargo/credential/cargo-credential-1password",
         );
+        */
         cargo_bin_path
     }
 }
