diff --git a/src/bootstrap/tool.rs b/src/bootstrap/tool.rs
index 1317c3f98..c04ad0408 100644
--- a/src/bootstrap/tool.rs
+++ b/src/bootstrap/tool.rs
@@ -623,6 +623,7 @@ impl Step for Cargo {
             })
             .expect("expected to build -- essential tool");
 
+        /*
         let build_cred = |name, path| {
             // These credential helpers are currently experimental.
             // Any build failures will be ignored.
@@ -654,6 +655,7 @@ impl Step for Cargo {
             "cargo-credential-1password",
             "src/tools/cargo/crates/credential/cargo-credential-1password",
         );
+        */
         cargo_bin_path
     }
 }
