diff --git a/src/bootstrap/src/lib.rs b/src/bootstrap/src/lib.rs
index 0be5ee4a..ae1c0a69 100644
--- a/src/bootstrap/src/lib.rs
+++ b/src/bootstrap/src/lib.rs
@@ -401,7 +401,7 @@ impl Build {
         }
 
         if rust_info.is_from_tarball() && config.description.is_none() {
-            config.description = Some("built from a source tarball".to_owned());
+            config.description = None;
         }
 
         let mut build = Build {
