diff --git a/src/bootstrap/lib.rs b/src/bootstrap/lib.rs
index 49c4c8edd..e5fc45349 100644
--- a/src/bootstrap/lib.rs
+++ b/src/bootstrap/lib.rs
@@ -420,7 +420,7 @@ impl Build {
         }
 
         if rust_info.is_from_tarball() && config.description.is_none() {
-            config.description = Some("built from a source tarball".to_owned());
+            config.description = None;
         }
 
         let mut build = Build {
