diff --git a/src/bootstrap/src/lib.rs b/src/bootstrap/src/lib.rs
index b941b0b1..b11ee14c 100644
--- a/src/bootstrap/src/lib.rs
+++ b/src/bootstrap/src/lib.rs
@@ -536,7 +536,7 @@ impl Build {
         }
 
         if rust_info.is_from_tarball() && config.description.is_none() {
-            config.description = Some("built from a source tarball".to_owned());
+            config.description = None;
         }
 
         let mut build = Build {
