diff --git a/src/bootstrap/src/lib.rs b/src/bootstrap/src/lib.rs
index b670b7dc..1d342272 100644
--- a/src/bootstrap/src/lib.rs
+++ b/src/bootstrap/src/lib.rs
@@ -537,7 +537,7 @@ impl Build {
         }
 
         if rust_info.is_from_tarball() && config.description.is_none() {
-            config.description = Some("built from a source tarball".to_owned());
+            config.description = None;
         }
 
         let mut build = Build {
