diff --git a/src/el.c b/src/el.c
index fda70acb..ae33a7ed 100644
--- a/src/el.c
+++ b/src/el.c
@@ -590,7 +590,7 @@ el_source(EditLine *el, const char *fname)
 			return -1;
 
 		if ((fname = (el->el_getenv)("EDITRC")) == NULL) {
-			static const char elpath[] = "/.editrc";
+			static const char elpath[] = "/.config/libedit/editrc";
 			size_t plen = sizeof(elpath);
 
 			if ((ptr = (el->el_getenv)("HOME")) == NULL)
