diff --git a/src/ncurses_ui.cc b/src/ncurses_ui.cc
index ad09614..44cd627 100644
--- a/src/ncurses_ui.cc
+++ b/src/ncurses_ui.cc
@@ -783,7 +783,8 @@ Optional<Key> NCursesUI::get_next_key()
             return parse_csi().value_or(alt('['));
         if (*next == 'O') // potential SS3
             return parse_ss3().value_or(alt('O'));
-        return alt(parse_key(*next));
+        if (*next != 27)
+            return alt(parse_key(*next));
     }
     return Key{Key::Escape};
 }
