diff --git a/src/window.hh b/src/window.hh
index 65253c8..319a367 100644
--- a/src/window.hh
+++ b/src/window.hh
@@ -70,7 +70,7 @@ private:
 
     Highlighters m_builtin_highlighters;
     bool m_resize_hook_pending = false;
-    DisplaySetup m_last_display_setup;
+    DisplaySetup m_last_display_setup = {};
 
     struct Setup
     {
@@ -82,7 +82,7 @@ private:
         Vector<BasicSelection, MemoryDomain::Display> selections;
     };
     Setup build_setup(const Context& context) const;
-    Setup m_last_setup;
+    Setup m_last_setup = {};
 };
 
 }
