diff --git a/src/main.cc b/src/main.cc
index c584a487..114dea06 100644
--- a/src/main.cc
+++ b/src/main.cc
@@ -44,6 +44,8 @@ extern const char* version;
 namespace
 {
 
+#ifdef SHOW_STARTUP_INFO
+
 struct {
     unsigned int version;
     StringView notes;
@@ -108,6 +110,8 @@ void show_startup_info(Client* local_client, int last_version)
                                  std::move(info), {}, InfoStyle::Prompt);
 }
 
+#endif
+
 inline void write_stdout(StringView str) { try { write(STDOUT_FILENO, str); } catch (runtime_error&) {} }
 inline void write_stderr(StringView str) { try { write(STDERR_FILENO, str); } catch (runtime_error&) {} }
 
@@ -815,8 +819,10 @@ int run_server(StringView session, StringView server_init,
                     local_client->context().faces()["Error"]
                 });
 
+#ifdef SHOW_STARTUP_INFO
             if (flags & ServerFlags::StartupInfo and local_client)
                 show_startup_info(local_client, global_scope.options()["startup_info_version"].get<int>());
+#endif
         }
 
         while (not terminate and
