diff --git a/src/main.cc b/src/main.cc
index 08c2bc5..ae62129 100644
--- a/src/main.cc
+++ b/src/main.cc
@@ -147,6 +147,8 @@ void show_startup_info(Client* local_client, int last_version)
     DisplayLineList info;
     for (auto [version, notes] : version_notes)
     {
+        if (last_version < 0)
+            continue;
         if (version and version <= last_version)
             continue;
 
