diff --git a/rc/filetype/c-family.kak b/rc/filetype/c-family.kak
index 0677b11..6b2486a 100644
--- a/rc/filetype/c-family.kak
+++ b/rc/filetype/c-family.kak
@@ -204,7 +204,7 @@ evaluate-commands %sh{
             maybe_at=''
         fi
 
-        cat <<-EOF
+        cat <<EOF
             add-highlighter shared/$ft regions
             add-highlighter shared/$ft/code default-region group
             add-highlighter shared/$ft/string region %{$maybe_at(?<!')(?<!'\\\\)"} %{(?<!\\\\)(?:\\\\\\\\)*"} fill string
@@ -218,7 +218,7 @@ evaluate-commands %sh{
             add-highlighter shared/$ft/macro/ regex ^\\h*(#\\h*\\S*) 1:meta
             add-highlighter shared/$ft/macro/ regex ^\\h*#\\h*include\\h+(<[^>]*>|"(?:[^"\\\\]|\\\\.)*") 1:module
             add-highlighter shared/$ft/macro/ regex /\\*.*?\\*/ 0:comment
-	EOF
+EOF
     done
 }
 
diff --git a/rc/tools/git.kak b/rc/tools/git.kak
index bbe8d65..af661c5 100644
--- a/rc/tools/git.kak
+++ b/rc/tools/git.kak
@@ -464,16 +464,14 @@ define-command -params 1.. \
 
     commit() {
         # Handle case where message needs not to be edited
-        if grep -E -q -e "-m|-F|-C|--message=.*|--file=.*|--reuse-message=.*|--no-edit|--fixup.*|--squash.*"; then
+        if grep -E -q -e "-m|-F|-C|--message=.*|--file=.*|--reuse-message=.*|--no-edit|--fixup.*|--squash.*" <<< "$@"; then
             if git commit "$@" > /dev/null 2>&1; then
                 echo 'echo -markup "{Information}Commit succeeded"'
             else
                 echo 'fail Commit failed'
             fi
             exit
-        fi <<-EOF
-			$@
-		EOF
+        fi
 
         # fails, and generate COMMIT_EDITMSG
         GIT_EDITOR='' EDITOR='' git commit "$@" > /dev/null 2>&1
diff --git a/rc/tools/patch.kak b/rc/tools/patch.kak
index a481ff4..208a1e6 100644
--- a/rc/tools/patch.kak
+++ b/rc/tools/patch.kak
@@ -41,9 +41,7 @@ define-command patch -params .. -docstring %{
         set-register e nop
         set-register | %{
             # The selected range to apply.
-            IFS=' .,' read min_line _ max_line _ <<-EOF
-            $kak_reg_s
-	EOF
+            IFS=' .,' read min_line _ max_line _ <<< "$kak_reg_s"
             min_line=$((min_line - kak_cursor_line + 1))
             max_line=$((max_line - kak_cursor_line + 1))
 
