diff --git a/rc/windowing/detection.kak b/rc/windowing/detection.kak
index 3dc046e..0b73134 100644
--- a/rc/windowing/detection.kak
+++ b/rc/windowing/detection.kak
@@ -31,11 +31,11 @@ hook -group windowing global KakBegin .* %{
         set -- ${kak_opt_windowing_modules}
         if [ $# -gt 0 ]; then
             echo 'try %{ '
-            while [ $# -gt 1 ]; do
+            while [ $# -gt 0 ]; do
                 echo "require-module ${1} } catch %{ "
                 shift
             done
-            echo "require-module ${1} }"
+            echo "}"
         fi
     }
 }
