diff --git a/rc/windowing/detection.kak b/rc/windowing/detection.kak
index 08867c8..3a634c8 100644
--- a/rc/windowing/detection.kak
+++ b/rc/windowing/detection.kak
@@ -35,7 +35,7 @@ hook -group windowing global KakBegin .* %{
                 echo "require-module ${1} } catch %{ "
                 shift
             done
-            echo "echo -debug 'no windowing module detected' }"
+            echo '}'
         fi
     }
 }
