diff --git a/Makefile b/Makefile
index 9110d177..c9bcfec2 100644
--- a/Makefile
+++ b/Makefile
@@ -3,10 +3,6 @@
 # Device Tree Compiler
 #
 
-$(warning WARNING: Building dtc using make is deprecated, in favour of using Meson (https://mesonbuild.com))
-$(warning )
-$(warning Use `meson setup builddir/ && meson compile -C builddir/` to build, `meson test -C builddir/` to test, or `meson configure` to see build options.)
-
 #
 # Version information will be constructed in this order:
 # DTC_VERSION release version as MAJOR.MINOR.PATCH
