diff --git a/src/exim.c b/src/exim.c
index 8111a448..efcad422 100644
--- a/src/exim.c
+++ b/src/exim.c
@@ -1863,6 +1863,13 @@ extern char **environ;
 store_init();	/* Initialise the memory allocation susbsystem */
 pcre_init();	/* Set up memory handling for pcre */
 
+/* Correct the default exim_path using /proc/self/exe if possible */
+if (lstat("/proc/self/exe", &statbuf) >= 0 && S_ISLNK(statbuf.st_mode)) {
+  uschar *path = calloc(PATH_MAX + 1, 1); /* st_size is broken in /proc */
+  if (path && readlink("/proc/self/exe", CS path, PATH_MAX) >= 0)
+    exim_path = path;
+}
+
 /* If the Exim user and/or group and/or the configuration file owner/group were
 defined by ref:name at build time, we must now find the actual uid/gid values.
 This is a feature to make the lives of binary distributors easier. */
