diff --git a/src/cmd/gofmt/doc.go b/src/cmd/gofmt/doc.go
index e340665..5cd3c6d 100644
--- a/src/cmd/gofmt/doc.go
+++ b/src/cmd/gofmt/doc.go
@@ -22,6 +22,8 @@ The flags are:
 		to standard output.
 	-e
 		Print all (including spurious) errors.
+	-i width
+		Indent with spaces instead of tabs.
 	-l
 		Do not print reformatted sources to standard output.
 		If a file's formatting is different from gofmt's, print its name
diff --git a/src/cmd/gofmt/gofmt.go b/src/cmd/gofmt/gofmt.go
index 4280ed4..7f2a427 100644
--- a/src/cmd/gofmt/gofmt.go
+++ b/src/cmd/gofmt/gofmt.go
@@ -35,6 +35,7 @@ var (
 	simplifyAST = flag.Bool("s", false, "simplify code")
 	doDiff      = flag.Bool("d", false, "display diffs instead of rewriting files")
 	allErrors   = flag.Bool("e", false, "report all errors (not just the first 10 on different lines)")
+	indent      = flag.Int("i", 0, "indent with spaces instead of tabs")
 
 	// debugging
 	cpuprofile = flag.String("cpuprofile", "", "write cpu profile to this file")
@@ -42,9 +43,6 @@ var (
 
 // Keep these in sync with go/format/format.go.
 const (
-	tabWidth    = 8
-	printerMode = printer.UseSpaces | printer.TabIndent | printerNormalizeNumbers
-
 	// printerNormalizeNumbers means to canonicalize number literal prefixes
 	// and exponents while printing. See https://golang.org/doc/go1.13#gofmt.
 	//
@@ -52,6 +50,11 @@ const (
 	printerNormalizeNumbers = 1 << 30
 )
 
+var (
+	tabWidth    = 8
+	printerMode = printer.UseSpaces | printer.TabIndent | printerNormalizeNumbers
+)
+
 // fdSem guards the number of concurrently-open file descriptors.
 //
 // For now, this is arbitrarily set to 200, based on the observation that many
@@ -382,6 +385,11 @@ func gofmtMain(s *sequencer) {
 	flag.Usage = usage
 	flag.Parse()
 
+	if *indent > 0 {
+		tabWidth = *indent
+		printerMode = printer.UseSpaces | printerNormalizeNumbers
+	}
+
 	if *cpuprofile != "" {
 		fdSem <- true
 		f, err := os.Create(*cpuprofile)
