diff --git a/ld/emulparams/elf32_x86_64.sh b/ld/emulparams/elf32_x86_64.sh
index 0a03548..c15328a 100644
--- a/ld/emulparams/elf32_x86_64.sh
+++ b/ld/emulparams/elf32_x86_64.sh
@@ -25,19 +25,3 @@ if [ "x${host}" = "x${target}" ]; then
       NATIVE=yes
   esac
 fi
-
-# Linux modifies the default library search path to first include
-# a 32-bit specific directory.
-case "$target" in
-  x86_64*-linux*|i[3-7]86-*-linux-*)
-    case "$EMULATION_NAME" in
-      *32*)
-        LIBPATH_SUFFIX=x32
-	LIBPATH_SUFFIX_SKIP=64
-	;;
-      *64*)
-        LIBPATH_SUFFIX=64
-	;;
-    esac
-    ;;
-esac
diff --git a/ld/emulparams/elf_x86_64.sh b/ld/emulparams/elf_x86_64.sh
index 5d31da1..75ea293 100644
--- a/ld/emulparams/elf_x86_64.sh
+++ b/ld/emulparams/elf_x86_64.sh
@@ -30,29 +30,3 @@ if [ "x${host}" = "x${target}" ]; then
       NATIVE=yes
   esac
 fi
-
-# Linux/Solaris modify the default library search path to first include
-# a 64-bit specific directory.
-case "$target" in
-  x86_64*-linux*|i[3-7]86-*-linux-*)
-    case "$EMULATION_NAME" in
-      *64*)
-        LIBPATH_SUFFIX=64
-	PARSE_AND_LIST_OPTIONS_BNDPLT='
-  fprintf (file, _("\
-  -z bndplt                   Always generate BND prefix in PLT entries\n"));
-'
-	PARSE_AND_LIST_ARGS_CASE_Z_BNDPLT='
-      else if (strcmp (optarg, "bndplt") == 0)
-	link_info.bndplt = TRUE;
-'
-	PARSE_AND_LIST_OPTIONS="$PARSE_AND_LIST_OPTIONS $PARSE_AND_LIST_OPTIONS_BNDPLT"
-	PARSE_AND_LIST_ARGS_CASE_Z="$PARSE_AND_LIST_ARGS_CASE_Z $PARSE_AND_LIST_ARGS_CASE_Z_BNDPLT"
-        ;;
-    esac
-    ;;
-  *-*-solaris2*)
-    LIBPATH_SUFFIX=/amd64
-    ELF_INTERPRETER_NAME=\"/lib/amd64/ld.so.1\"
-  ;;
-esac
