diff --git a/Lib/compileall.py b/Lib/compileall.py
index 67fe3704..c9ef5e4d 100644
--- a/Lib/compileall.py
+++ b/Lib/compileall.py
@@ -372,9 +372,7 @@ def main():
     parser.add_argument('--invalidation-mode',
                         choices=sorted(invalidation_modes),
                         help=('set .pyc invalidation mode; defaults to '
-                              '"checked-hash" if the SOURCE_DATE_EPOCH '
-                              'environment variable is set, and '
-                              '"timestamp" otherwise.'))
+                              '"checked-hash".'))
     parser.add_argument('-o', action='append', type=int, dest='opt_levels',
                         help=('Optimization levels to run compilation with. '
                               'Default is -1 which uses the optimization level '
diff --git a/Lib/py_compile.py b/Lib/py_compile.py
index 43d8ec90..48b644ea 100644
--- a/Lib/py_compile.py
+++ b/Lib/py_compile.py
@@ -70,10 +70,7 @@ class PycInvalidationMode(enum.Enum):
 
 
 def _get_default_invalidation_mode():
-    if os.environ.get('SOURCE_DATE_EPOCH'):
-        return PycInvalidationMode.CHECKED_HASH
-    else:
-        return PycInvalidationMode.TIMESTAMP
+    return PycInvalidationMode.CHECKED_HASH
 
 
 def compile(file, cfile=None, dfile=None, doraise=False, optimize=-1,
