diff --git a/Modules/getpath.c b/Modules/getpath.c
index d0128b20..253d61df 100644
--- a/Modules/getpath.c
+++ b/Modules/getpath.c
@@ -932,8 +932,8 @@ _PyConfig_InitPathConfig(PyConfig *config, int compute_path_config)
 #else
         !int_to_dict(dict, "WITH_NEXT_FRAMEWORK", 0) ||
 #endif
-        !decode_to_dict(dict, "PREFIX", PREFIX) ||
-        !decode_to_dict(dict, "EXEC_PREFIX", EXEC_PREFIX) ||
+        !decode_to_dict(dict, "PREFIX", PREFIX[0] ? PREFIX : "/") ||
+        !decode_to_dict(dict, "EXEC_PREFIX", EXEC_PREFIX[0] ? EXEC_PREFIX : "/") ||
         !decode_to_dict(dict, "PYTHONPATH", PYTHONPATH) ||
         !decode_to_dict(dict, "VPATH", VPATH) ||
         !decode_to_dict(dict, "PLATLIBDIR", PLATLIBDIR) ||
