diff --git a/Lib/site.py b/Lib/site.py
index de82d33..0d49734 100644
--- a/Lib/site.py
+++ b/Lib/site.py
@@ -445,10 +445,6 @@ def enablerlcompleter():
             # http://bugs.python.org/issue5845#msg198636
             history = os.path.join(os.path.expanduser('~'),
                                    '.python_history')
-            try:
-                readline.read_history_file(history)
-            except OSError:
-                pass
 
             def write_history():
                 try:
@@ -458,7 +454,11 @@ def enablerlcompleter():
                     # or is not writable, or the filesystem is read-only.
                     pass
 
-            atexit.register(write_history)
+            try:
+                readline.read_history_file(history)
+                atexit.register(write_history)
+            except OSError:
+                pass
 
     sys.__interactivehook__ = register_readline
 
