diff --git a/Lib/site.py b/Lib/site.py
index 0a0dc47b..261a7341 100644
--- a/Lib/site.py
+++ b/Lib/site.py
@@ -472,8 +472,7 @@ def gethistoryfile():
         history = os.environ.get("PYTHON_HISTORY")
         if history:
             return history
-    return os.path.join(os.path.expanduser('~'),
-        '.python_history')
+    return None
 
 
 def enablerlcompleter():
@@ -518,8 +517,7 @@ def register_readline():
 
     if readline.get_current_history_length() == 0:
         from _pyrepl.main import CAN_USE_PYREPL
-        # If no history was loaded, default to .python_history,
-        # or PYTHON_HISTORY.
+        # If no history was loaded, default to PYTHON_HISTORY.
         # The guard is necessary to avoid doubling history size at
         # each interpreter exit when readline was already configured
         # through a PYTHONSTARTUP hook, see:
@@ -529,20 +527,22 @@ def register_readline():
             readline_module = readline
         else:
             readline_module = _pyrepl.readline
-        try:
-            readline_module.read_history_file(history)
-        except (OSError,* _pyrepl.unix_console._error):
-            pass
 
-        def write_history():
+        if history:
             try:
-                readline_module.write_history_file(history)
-            except (FileNotFoundError, PermissionError):
-                # home directory does not exist or is not writable
-                # https://bugs.python.org/issue19891
+                readline_module.read_history_file(history)
+            except (OSError,* _pyrepl.unix_console._error):
                 pass
 
-        atexit.register(write_history)
+            def write_history():
+                try:
+                    readline_module.write_history_file(history)
+                except (FileNotFoundError, PermissionError):
+                    # home directory does not exist or is not writable
+                    # https://bugs.python.org/issue19891
+                    pass
+
+            atexit.register(write_history)
 
 
 def venv(known_paths):
