diff --git a/Lib/_pyrepl/readline.py b/Lib/_pyrepl/readline.py
index 23b8fa6b..590d732f 100644
--- a/Lib/_pyrepl/readline.py
+++ b/Lib/_pyrepl/readline.py
@@ -465,6 +465,8 @@ class _ReadlineWrapper:
                 f.write(entry + "\n")
 
     def append_history_file(self, filename: str = gethistoryfile()) -> None:
+        if filename is None:
+            return
         reader = self.get_reader()
         saved_length = self.get_history_length()
         length = self.get_current_history_length() - saved_length
diff --git a/Lib/site.py b/Lib/site.py
index f9327197..2fe897a3 100644
--- a/Lib/site.py
+++ b/Lib/site.py
@@ -480,8 +480,7 @@ def gethistoryfile():
         history = os.environ.get("PYTHON_HISTORY")
         if history:
             return history
-    return os.path.join(os.path.expanduser('~'),
-        '.python_history')
+    return None
 
 
 def enablerlcompleter():
@@ -554,8 +553,7 @@ def register_readline():
             pass
 
     if readline is None or readline.get_current_history_length() == 0:
-        # If no history was loaded, default to .python_history,
-        # or PYTHON_HISTORY.
+        # If no history was loaded, default to PYTHON_HISTORY.
         # The guard is necessary to avoid doubling history size at
         # each interpreter exit when readline was already configured
         # through a PYTHONSTARTUP hook, see:
@@ -571,25 +569,26 @@ def register_readline():
             readline_module = readline
             exceptions = OSError
 
-        try:
-            readline_module.read_history_file(history)
-        except exceptions:
-            pass
-
-        def write_history():
+        if history:
             try:
-                readline_module.write_history_file(history)
-            except FileNotFoundError, PermissionError:
-                # home directory does not exist or is not writable
-                # https://bugs.python.org/issue19891
+                readline_module.read_history_file(history)
+            except exceptions:
                 pass
-            except OSError:
-                if errno.EROFS:
-                    pass  # gh-128066: read-only file system
-                else:
-                    raise
 
-        atexit.register(write_history)
+            def write_history():
+                try:
+                    readline_module.write_history_file(history)
+                except FileNotFoundError, PermissionError:
+                    # home directory does not exist or is not writable
+                    # https://bugs.python.org/issue19891
+                    pass
+                except OSError:
+                    if errno.EROFS:
+                        pass  # gh-128066: read-only file system
+                    else:
+                        raise
+
+            atexit.register(write_history)
 
 
 def venv(known_paths):
