diff --git a/Modules/_ssl.c b/Modules/_ssl.c
index a7a278de..6e6b5775 100644
--- a/Modules/_ssl.c
+++ b/Modules/_ssl.c
@@ -4728,7 +4728,7 @@ _ssl__SSLContext_sni_callback_set_impl(PySSLContext *self, PyObject *value)
     return 0;
 }
 
-#if OPENSSL_VERSION_NUMBER < 0x30300000L
+#if OPENSSL_VERSION_NUMBER < 0x30300000L && !defined(LIBRESSL_VERSION_NUMBER)
 static X509_OBJECT *x509_object_dup(const X509_OBJECT *obj)
 {
     int ok;
