diff --git a/setup.py b/setup.py
index 15d0d45..30d0f91 100644
--- a/setup.py
+++ b/setup.py
@@ -447,6 +447,7 @@ class PyBuildExt(build_ext):
         sysconf_built = set(sysconfig.get_config_var('MODBUILT_NAMES').split())
         sysconf_shared = set(sysconfig.get_config_var('MODSHARED_NAMES').split())
         sysconf_dis = set(sysconfig.get_config_var('MODDISABLED_NAMES').split())
+        sysconf_dis.add('__hashlib')
 
         mods_built = []
         mods_disabled = []
