diff --git a/Modules/_ssl.c b/Modules/_ssl.c
index 417d5ca5..aca63c59 100644
--- a/Modules/_ssl.c
+++ b/Modules/_ssl.c
@@ -910,7 +910,7 @@ newPySSLSocket(PySSLContext *sslctx, PySocketSockObject *sock,
     }
 
     /* bpo43522 and OpenSSL < 1.1.1l: copy hostflags manually */
-#if OPENSSL_VERSION_NUMBER < 0x101010cf
+#if OPENSSL_VERSION_NUMBER < 0x101010cf && !defined(LIBRESSL_VERSION_NUMBER)
     X509_VERIFY_PARAM *ssl_verification_params = SSL_get0_param(self->ssl);
     X509_VERIFY_PARAM *ssl_ctx_verification_params = SSL_CTX_get0_param(ctx);
 
@@ -3870,10 +3870,12 @@ _ssl__SSLContext__host_flags_get_impl(PySSLContext *self)
 /*[clinic end generated code: output=0f9db6654ce32582 input=8e3c49499eefd0e5]*/
 {
     X509_VERIFY_PARAM *ssl_verification_params;
-    unsigned int host_flags;
+    unsigned int host_flags = 0;
 
+#ifndef LIBRESSL_VERSION_NUMBER
     ssl_verification_params = SSL_CTX_get0_param(self->ctx);
     host_flags = X509_VERIFY_PARAM_get_hostflags(ssl_verification_params);
+#endif
     return PyLong_FromUnsignedLong(host_flags);
 }
 
