diff --git a/Lib/ctypes/util.py b/Lib/ctypes/util.py
index 117bf06c..a8f66363 100644
--- a/Lib/ctypes/util.py
+++ b/Lib/ctypes/util.py
@@ -99,7 +99,7 @@ elif sys.platform == "android":
         return fname if os.path.isfile(fname) else None
 
 elif os.name == "posix":
-    # Andreas Degert's find functions, using gcc, /sbin/ldconfig, objdump
+    # Andreas Degert's find functions, using gcc, /bin/ldconfig, objdump
     import re, tempfile
 
     def _is_elf(filename):
@@ -221,7 +221,7 @@ elif os.name == "posix":
             expr = os.fsencode(expr)
 
             try:
-                proc = subprocess.Popen(('/sbin/ldconfig', '-r'),
+                proc = subprocess.Popen(('/bin/ldconfig', '-r'),
                                         stdout=subprocess.PIPE,
                                         stderr=subprocess.DEVNULL)
             except OSError:  # E.g. command not found
diff --git a/Lib/encodings/rot_13.py b/Lib/encodings/rot_13.py
index 5627bfbc..80f3f99a 100755
--- a/Lib/encodings/rot_13.py
+++ b/Lib/encodings/rot_13.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/bin/python
 """ Python Character Mapping Codec for ROT13.
 
 This codec de/encodes from str to str.
diff --git a/Lib/mimetypes.py b/Lib/mimetypes.py
index 8604000e..0f3f85c9 100644
--- a/Lib/mimetypes.py
+++ b/Lib/mimetypes.py
@@ -47,14 +47,6 @@ __all__ = [
 
 knownfiles = [
     "/etc/mime.types",
-    "/etc/httpd/mime.types",                    # Mac OS X
-    "/etc/httpd/conf/mime.types",               # Apache
-    "/etc/apache/mime.types",                   # Apache 1
-    "/etc/apache2/mime.types",                  # Apache 2
-    "/usr/local/etc/httpd/conf/mime.types",
-    "/usr/local/lib/netscape/mime.types",
-    "/usr/local/etc/httpd/conf/mime.types",     # Apache 1.2
-    "/usr/local/etc/mime.types",                # Apache 1.3
     ]
 
 inited = False
diff --git a/Lib/platform.py b/Lib/platform.py
index 59583822..f847a90e 100755
--- a/Lib/platform.py
+++ b/Lib/platform.py
@@ -1388,8 +1388,8 @@ def platform(aliased=False, terse=False):
 ### freedesktop.org os-release standard
 # https://www.freedesktop.org/software/systemd/man/os-release.html
 
-# /etc takes precedence over /usr/lib
-_os_release_candidates = ("/etc/os-release", "/usr/lib/os-release")
+# /etc takes precedence over /lib
+_os_release_candidates = ("/etc/os-release", "/lib/os-release")
 _os_release_cache = None
 
 
diff --git a/Lib/posixpath.py b/Lib/posixpath.py
index f2173032..1c5c71c1 100644
--- a/Lib/posixpath.py
+++ b/Lib/posixpath.py
@@ -18,7 +18,7 @@ pardir = '..'
 extsep = '.'
 sep = '/'
 pathsep = ':'
-defpath = '/bin:/usr/bin'
+defpath = '/bin'
 altsep = None
 devnull = '/dev/null'
 
diff --git a/Lib/site.py b/Lib/site.py
index 095e91e9..fa2f7533 100644
--- a/Lib/site.py
+++ b/Lib/site.py
@@ -321,12 +321,12 @@ def _get_path(userbase):
     implementation_lower = implementation.lower()
     if os.name == 'nt':
         ver_nodot = sys.winver.replace('.', '')
-        return f'{userbase}\\{implementation}{ver_nodot}\\site-packages'
+        return f'{userbase}\\{implementation}\\site-packages'
 
     if sys.platform == 'darwin' and sys._framework:
         return f'{userbase}/lib/{implementation_lower}/site-packages'
 
-    return f'{userbase}/lib/python{version[0]}.{version[1]}{abi_thread}/site-packages'
+    return f'{userbase}/lib/python{abi_thread}/site-packages'
 
 
 def getuserbase():
@@ -405,7 +405,7 @@ def getsitepackages(prefixes=None):
 
             for libdir in libdirs:
                 path = os.path.join(prefix, libdir,
-                                    f"{implementation}{ver[0]}.{ver[1]}{abi_thread}",
+                                    f"{implementation}{abi_thread}",
                                     "site-packages")
                 sitepackages.append(path)
         else:
diff --git a/Lib/sysconfig/__init__.py b/Lib/sysconfig/__init__.py
index ec3b638f..a1ccfe95 100644
--- a/Lib/sysconfig/__init__.py
+++ b/Lib/sysconfig/__init__.py
@@ -27,14 +27,12 @@ _ALWAYS_STR = {
 
 _INSTALL_SCHEMES = {
     'posix_prefix': {
-        'stdlib': '{installed_base}/{platlibdir}/{implementation_lower}{py_version_short}{abi_thread}',
-        'platstdlib': '{platbase}/{platlibdir}/{implementation_lower}{py_version_short}{abi_thread}',
-        'purelib': '{base}/lib/{implementation_lower}{py_version_short}{abi_thread}/site-packages',
-        'platlib': '{platbase}/{platlibdir}/{implementation_lower}{py_version_short}{abi_thread}/site-packages',
-        'include':
-            '{installed_base}/include/{implementation_lower}{py_version_short}{abiflags}',
-        'platinclude':
-            '{installed_platbase}/include/{implementation_lower}{py_version_short}{abiflags}',
+        'stdlib': '{installed_base}/{platlibdir}/{implementation_lower}{abi_thread}',
+        'platstdlib': '{platbase}/{platlibdir}/{implementation_lower}{abi_thread}',
+        'purelib': '{base}/lib/{implementation_lower}{abi_thread}/site-packages',
+        'platlib': '{platbase}/{platlibdir}/{implementation_lower}{abi_thread}/site-packages',
+        'include': '{installed_base}/lib/include/{implementation_lower}{abiflags}',
+        'platinclude': '{installed_platbase}/lib/include/{implementation_lower}{abiflags}',
         'scripts': '{base}/bin',
         'data': '{base}',
         },
@@ -43,8 +41,8 @@ _INSTALL_SCHEMES = {
         'platstdlib': '{base}/lib/{implementation_lower}',
         'purelib': '{base}/lib/{implementation_lower}',
         'platlib': '{base}/lib/{implementation_lower}',
-        'include': '{installed_base}/include/{implementation_lower}',
-        'platinclude': '{installed_base}/include/{implementation_lower}',
+        'include': '{installed_base}/lib/include/{implementation_lower}',
+        'platinclude': '{installed_base}/lib/include/{implementation_lower}',
         'scripts': '{base}/bin',
         'data': '{base}',
         },
@@ -77,14 +75,12 @@ _INSTALL_SCHEMES = {
     # Downstream distributors who patch posix_prefix/nt scheme are encouraged to
     # leave the following schemes unchanged
     'posix_venv': {
-        'stdlib': '{installed_base}/{platlibdir}/{implementation_lower}{py_version_short}{abi_thread}',
-        'platstdlib': '{platbase}/{platlibdir}/{implementation_lower}{py_version_short}{abi_thread}',
-        'purelib': '{base}/lib/{implementation_lower}{py_version_short}{abi_thread}/site-packages',
-        'platlib': '{platbase}/{platlibdir}/{implementation_lower}{py_version_short}{abi_thread}/site-packages',
-        'include':
-            '{installed_base}/include/{implementation_lower}{py_version_short}{abiflags}',
-        'platinclude':
-            '{installed_platbase}/include/{implementation_lower}{py_version_short}{abiflags}',
+        'stdlib': '{installed_base}/{platlibdir}/{implementation_lower}{abi_thread}',
+        'platstdlib': '{platbase}/{platlibdir}/{implementation_lower}{abi_thread}',
+        'purelib': '{base}/lib/{implementation_lower}{abi_thread}/site-packages',
+        'platlib': '{platbase}/{platlibdir}/{implementation_lower}{abi_thread}/site-packages',
+        'include': '{installed_base}/lib/include/{implementation_lower}{abiflags}',
+        'platinclude': '{installed_platbase}/lib/include/{implementation_lower}{abiflags}',
         'scripts': '{base}/bin',
         'data': '{base}',
         },
@@ -139,20 +135,20 @@ if _HAS_USER_BASE:
     _INSTALL_SCHEMES |= {
         # NOTE: When modifying "purelib" scheme, update site._get_path() too.
         'nt_user': {
-            'stdlib': '{userbase}/{implementation}{py_version_nodot_plat}',
-            'platstdlib': '{userbase}/{implementation}{py_version_nodot_plat}',
-            'purelib': '{userbase}/{implementation}{py_version_nodot_plat}/site-packages',
-            'platlib': '{userbase}/{implementation}{py_version_nodot_plat}/site-packages',
-            'include': '{userbase}/{implementation}{py_version_nodot_plat}/Include',
-            'scripts': '{userbase}/{implementation}{py_version_nodot_plat}/Scripts',
+            'stdlib': '{userbase}/{implementation}',
+            'platstdlib': '{userbase}/{implementation}',
+            'purelib': '{userbase}/{implementation}/site-packages',
+            'platlib': '{userbase}/{implementation}/site-packages',
+            'include': '{userbase}/{implementation}/Include',
+            'scripts': '{userbase}/{implementation}/Scripts',
             'data': '{userbase}',
             },
         'posix_user': {
-            'stdlib': '{userbase}/{platlibdir}/{implementation_lower}{py_version_short}{abi_thread}',
-            'platstdlib': '{userbase}/{platlibdir}/{implementation_lower}{py_version_short}{abi_thread}',
-            'purelib': '{userbase}/lib/{implementation_lower}{py_version_short}{abi_thread}/site-packages',
-            'platlib': '{userbase}/lib/{implementation_lower}{py_version_short}{abi_thread}/site-packages',
-            'include': '{userbase}/include/{implementation_lower}{py_version_short}{abi_thread}',
+            'stdlib': '{userbase}/{platlibdir}/{implementation_lower}{abi_thread}',
+            'platstdlib': '{userbase}/{platlibdir}/{implementation_lower}{abi_thread}',
+            'purelib': '{userbase}/lib/{implementation_lower}{abi_thread}/site-packages',
+            'platlib': '{userbase}/lib/{implementation_lower}{abi_thread}/site-packages',
+            'include': '{userbase}/lib/include/{implementation_lower}{abi_thread}',
             'scripts': '{userbase}/bin',
             'data': '{userbase}',
             },
diff --git a/Lib/tempfile.py b/Lib/tempfile.py
index b5a15f7b..97aa7cd4 100644
--- a/Lib/tempfile.py
+++ b/Lib/tempfile.py
@@ -170,7 +170,7 @@ def _candidate_tempdir_list():
                          _os.path.expandvars(r'%SYSTEMROOT%\Temp'),
                          r'c:\temp', r'c:\tmp', r'\temp', r'\tmp' ])
     else:
-        dirlist.extend([ '/tmp', '/var/tmp', '/usr/tmp' ])
+        dirlist.extend([ '/tmp' ])
 
     # As a last resort, the current directory.
     try:
diff --git a/Lib/uuid.py b/Lib/uuid.py
index c286eac3..1e060127 100644
--- a/Lib/uuid.py
+++ b/Lib/uuid.py
@@ -365,7 +365,6 @@ def _get_command_stdout(command, *args):
 
     try:
         path_dirs = os.environ.get('PATH', os.defpath).split(os.pathsep)
-        path_dirs.extend(['/sbin', '/usr/sbin'])
         executable = shutil.which(command, path=os.pathsep.join(path_dirs))
         if executable is None:
             return None
diff --git a/Lib/venv/__init__.py b/Lib/venv/__init__.py
index f7a6d261..abae7c66 100644
--- a/Lib/venv/__init__.py
+++ b/Lib/venv/__init__.py
@@ -182,12 +182,6 @@ class EnvBuilder:
         create_if_needed(incpath)
         context.lib_path = libpath
         create_if_needed(libpath)
-        # Issue 21197: create lib64 as a symlink to lib on 64-bit non-OS X POSIX
-        if ((sys.maxsize > 2**32) and (os.name == 'posix') and
-            (sys.platform != 'darwin')):
-            link_path = os.path.join(env_dir, 'lib64')
-            if not os.path.exists(link_path):   # Issue #21643
-                os.symlink('lib', link_path)
         context.bin_path = binpath
         context.bin_name = os.path.relpath(binpath, env_dir)
         context.env_exe = os.path.join(binpath, exename)
diff --git a/Makefile.pre.in b/Makefile.pre.in
index 03ca4cb6..4921f03a 100644
--- a/Makefile.pre.in
+++ b/Makefile.pre.in
@@ -149,7 +149,7 @@ BINDIR=		@bindir@
 LIBDIR=		@libdir@
 MANDIR=		@mandir@
 INCLUDEDIR=	@includedir@
-CONFINCLUDEDIR=	$(exec_prefix)/include
+CONFINCLUDEDIR=	$(exec_prefix)/lib/include
 PLATLIBDIR=	@PLATLIBDIR@
 SCRIPTDIR=	$(prefix)/$(PLATLIBDIR)
 # executable name for shebangs
@@ -159,9 +159,9 @@ WHEEL_PKG_DIR=	@WHEEL_PKG_DIR@
 
 # Detailed destination directories
 BINLIBDEST=	@BINLIBDEST@
-LIBDEST=	$(SCRIPTDIR)/python$(VERSION)$(ABI_THREAD)
-INCLUDEPY=	$(INCLUDEDIR)/python$(LDVERSION)
-CONFINCLUDEPY=	$(CONFINCLUDEDIR)/python$(LDVERSION)
+LIBDEST=	$(SCRIPTDIR)/python$(ABI_THREAD)
+INCLUDEPY=	$(INCLUDEDIR)/python
+CONFINCLUDEPY=	$(CONFINCLUDEDIR)/python
 
 # Symbols used for using shared libraries
 SHLIB_SUFFIX=	@SHLIB_SUFFIX@
diff --git a/Modules/getpath.py b/Modules/getpath.py
index 1f1bfcb4..787f86e3 100644
--- a/Modules/getpath.py
+++ b/Modules/getpath.py
@@ -179,9 +179,9 @@ if os_name == 'posix' or os_name == 'darwin':
     BUILDDIR_TXT = 'pybuilddir.txt'
     BUILD_LANDMARK = 'Modules/Setup.local'
     DEFAULT_PROGRAM_NAME = f'python{VERSION_MAJOR}'
-    STDLIB_SUBDIR = f'{platlibdir}/python{VERSION_MAJOR}.{VERSION_MINOR}{ABI_THREAD}'
+    STDLIB_SUBDIR = f'{platlibdir}/python{ABI_THREAD}'
     STDLIB_LANDMARKS = [f'{STDLIB_SUBDIR}/os.py', f'{STDLIB_SUBDIR}/os.pyc']
-    PLATSTDLIB_LANDMARK = f'{platlibdir}/python{VERSION_MAJOR}.{VERSION_MINOR}{ABI_THREAD}/lib-dynload'
+    PLATSTDLIB_LANDMARK = f'{platlibdir}/python{ABI_THREAD}/lib-dynload'
     BUILDSTDLIB_LANDMARKS = ['Lib/os.py']
     VENV_LANDMARK = 'pyvenv.cfg'
     ZIP_LANDMARK = f'{platlibdir}/python{VERSION_MAJOR}{VERSION_MINOR}{ABI_THREAD}.zip'
diff --git a/configure b/configure
index ae70f02f..8e39fed7 100755
--- a/configure
+++ b/configure
@@ -24609,7 +24609,7 @@ fi
 
 
 
-BINLIBDEST='$(LIBDIR)/python$(VERSION)$(ABI_THREAD)'
+BINLIBDEST='$(LIBDIR)/python$(ABI_THREAD)'
 
 
 # Check for --with-platlibdir
@@ -24646,9 +24646,9 @@ fi
 
 
 if test x$PLATFORM_TRIPLET = x; then
-  LIBPL='$(prefix)'"/${PLATLIBDIR}/python${VERSION}${ABI_THREAD}/config-${LDVERSION}"
+  LIBPL='$(prefix)'"/${PLATLIBDIR}/python${ABI_THREAD}/config-${LDVERSION}"
 else
-  LIBPL='$(prefix)'"/${PLATLIBDIR}/python${VERSION}${ABI_THREAD}/config-${LDVERSION}-${PLATFORM_TRIPLET}"
+  LIBPL='$(prefix)'"/${PLATLIBDIR}/python${ABI_THREAD}/config-${LDVERSION}-${PLATFORM_TRIPLET}"
 fi
 
 
