diff --git a/libdm/Makefile.in b/libdm/Makefile.in
index 6ae0075f..94b3359e 100644
--- a/libdm/Makefile.in
+++ b/libdm/Makefile.in
@@ -41,9 +41,7 @@ SOURCES =\
 
 INCLUDES = -I$(srcdir)/$(interface)
 
-ifeq ("@STATIC_LINK@", "yes")
 LIB_STATIC = $(interface)/libdevmapper.a
-endif
 
 ifeq ("@SHARED_LINK@", "yes")
 LIB_SHARED = $(interface)/libdevmapper.$(LIB_SUFFIX)
diff --git a/libdm/dm-tools/Makefile.in b/libdm/dm-tools/Makefile.in
index 5da892de..b9c0dfd0 100644
--- a/libdm/dm-tools/Makefile.in
+++ b/libdm/dm-tools/Makefile.in
@@ -61,10 +61,10 @@ LIBDM_LIBS = -L$(interfacebuilddir) -ldevmapper
 LIBDM_SHARED = $(interfacebuilddir)/libdevmapper.so
 LIBDM_STATIC = $(interfacebuilddir)/libdevmapper.a
 
-dmsetup: dmsetup.o $(LIBDM_SHARED)
+dmsetup: dmsetup.o $(LIBDM_STATIC)
 	$(SHOW) "    [CC] $@"
 	$(Q) $(CC) $(CFLAGS) $(LDFLAGS) $(EXTRA_EXEC_LDFLAGS) $(ELDFLAGS) \
-	      -o $@ $< $(LIBDM_LIBS) $(LIBS)
+	      -o $@ $< $(LIBDM_STATIC) $(LIBS)
 
 dmsetup.static: dmsetup.o $(LIBDM_STATIC)
 	$(SHOW) "    [CC] $@"
@@ -82,10 +82,10 @@ install_dmsetup_static: dmsetup.static
 	$(Q) $(LN_S) -f $(<F) $(staticdir)/dmstats.static
 
 
-dmfilemapd: dmfilemapd.o $(LIBDM_SHARED)
+dmfilemapd: dmfilemapd.o $(LIBDM_STATIC)
 	$(SHOW) "    [CC] $@"
 	$(Q) $(CC) $(CFLAGS) $(LDFLAGS) $(EXTRA_EXEC_LDFLAGS) $(ELDFLAGS) \
-	      -o $@ $< $(LIBDM_LIBS) $(LIBS)
+	      -o $@ $< $(LIBDM_STATIC) $(LIBS)
 
 dmfilemapd.static: dmfilemapd.o $(LIBDM_STATIC)
 	$(SHOW) "    [CC] $@"
