diff --git a/src/default.c b/src/default.c
index b4cfdf7..e3a537c 100644
--- a/src/default.c
+++ b/src/default.c
@@ -535,7 +535,7 @@ static const char *default_variables[] =
     "OBJC", "gcc",
 #else
     "CC", "cc",
-    "CXX", "g++",
+    "CXX", "c++",
     "OBJC", "cc",
 #endif
 
diff --git a/src/read.c b/src/read.c
index 0743124..fe866cc 100644
--- a/src/read.c
+++ b/src/read.c
@@ -109,11 +109,6 @@ static const char *default_include_directories[] =
 #endif
 #if defined(INCLUDEDIR)
     INCLUDEDIR,
-#endif
-#ifndef _AMIGA
-    "/usr/gnu/include",
-    "/usr/local/include",
-    "/usr/include",
 #endif
     0
   };
diff --git a/src/remake.c b/src/remake.c
index 4ce3d2a..bc5c5c1 100644
--- a/src/remake.c
+++ b/src/remake.c
@@ -1685,10 +1685,6 @@ library_search (const char *lib, FILE_TIMESTAMP *mtime_ptr)
 {
   static const char *dirs[] =
     {
-#ifndef _AMIGA
-      "/lib",
-      "/usr/lib",
-#endif
 #if defined(WINDOWS32) && !defined(LIBDIR)
 /*
  * This is completely up to the user at product install time. Just define
@@ -1697,6 +1693,8 @@ library_search (const char *lib, FILE_TIMESTAMP *mtime_ptr)
 #define LIBDIR "."
 #endif
       LIBDIR,                   /* Defined by configuration.  */
+      LIBDIR "/shared",
+      LIBDIR "/static",
       0
     };
 
