diff --git a/src/default.c b/src/default.c
index e396269..c238f72 100644
--- a/src/default.c
+++ b/src/default.c
@@ -542,7 +542,7 @@ static const char *default_variables[] =
     "CXX", "gcc",
 #  endif /* __MSDOS__ */
 # else
-    "CXX", "g++",
+    "CXX", "c++",
 # endif
 #endif
     /* This expands to $(CO) $(COFLAGS) $< $@ if $@ does not exist,
diff --git a/src/read.c b/src/read.c
index b0fc1e1..9945a89 100644
--- a/src/read.c
+++ b/src/read.c
@@ -109,11 +109,6 @@ static const char *default_include_directories[] =
 #endif
 #if defined(INCLUDEDIR)
     INCLUDEDIR,
-#endif
-#ifndef _AMIGA
-    "/usr/gnu/include",
-    "/usr/local/include",
-    "/usr/include",
 #endif
     0
   };
@@ -1906,9 +1901,9 @@ check_specials (struct nameseq *files, int set_default)
           /* These default values are based on IEEE Std 1003.1-2008.
              It requires '-O 1' for [CF]FLAGS, but GCC doesn't allow
              space between -O and the number so omit it here.  */
-          define_variable_cname ("CC", "c99", o_default, 0);
+          define_variable_cname ("CC", "cc", o_default, 0);
           define_variable_cname ("CFLAGS", "-O1", o_default, 0);
-          define_variable_cname ("FC", "fort77", o_default, 0);
+          define_variable_cname ("FC", "f77", o_default, 0);
           define_variable_cname ("FFLAGS", "-O1", o_default, 0);
           define_variable_cname ("SCCSGETFLAGS", "-s", o_default, 0);
           define_variable_cname ("ARFLAGS", "-rv", o_default, 0);
diff --git a/src/remake.c b/src/remake.c
index fe67ab2..a6984b9 100644
--- a/src/remake.c
+++ b/src/remake.c
@@ -1689,10 +1689,6 @@ library_search (const char *lib, FILE_TIMESTAMP *mtime_ptr)
 {
   static const char *dirs[] =
     {
-#ifndef _AMIGA
-      "/lib",
-      "/usr/lib",
-#endif
 #if defined(WINDOWS32) && !defined(LIBDIR)
 /*
  * This is completely up to the user at product install time. Just define
@@ -1701,6 +1697,8 @@ library_search (const char *lib, FILE_TIMESTAMP *mtime_ptr)
 #define LIBDIR "."
 #endif
       LIBDIR,                   /* Defined by configuration.  */
+      LIBDIR "/shared",
+      LIBDIR "/static",
       0
     };
 
