diff --git a/Makefile b/Makefile
index 73e7bac7..8892d91c 100644
--- a/Makefile
+++ b/Makefile
@@ -445,6 +445,9 @@ HOST_LFS_CFLAGS := $(shell getconf LFS_CFLAGS 2>/dev/null)
 HOST_LFS_LDFLAGS := $(shell getconf LFS_LDFLAGS 2>/dev/null)
 HOST_LFS_LIBS := $(shell getconf LFS_LIBS 2>/dev/null)
 
+LLVM := $(shell $${CC:-cc} --version | grep -i -q -F clang && echo 1)
+LLVM_IAS := $(LLVM)
+
 ifneq ($(LLVM),)
 ifneq ($(filter %/,$(LLVM)),)
 LLVM_PREFIX := $(LLVM)
