diff --git a/scripts/kconfig/mconf-cfg.sh b/scripts/kconfig/mconf-cfg.sh
index 1e61f50a5..9c095ac2d 100755
--- a/scripts/kconfig/mconf-cfg.sh
+++ b/scripts/kconfig/mconf-cfg.sh
@@ -4,6 +4,12 @@
 cflags=$1
 libs=$2
 
+if echo '#include <ncurses.h>' | ${HOSTCC} -E - >/dev/null 2>&1; then
+	true > ${cflags}
+	echo -lncurses > ${libs}
+	exit 0
+fi
+
 PKG="ncursesw"
 PKG2="ncurses"
 
@@ -36,14 +42,6 @@ if [ -f /usr/include/ncurses/ncurses.h ]; then
 	exit 0
 fi
 
-# As a final fallback before giving up, check if $HOSTCC knows of a default
-# ncurses installation (e.g. from a vendor-specific sysroot).
-if echo '#include <ncurses.h>' | ${HOSTCC} -E - >/dev/null 2>&1; then
-	echo -D_GNU_SOURCE > ${cflags}
-	echo -lncurses > ${libs}
-	exit 0
-fi
-
 echo >&2 "*"
 echo >&2 "* Unable to find the ncurses package."
 echo >&2 "* Install ncurses (ncurses-devel or libncurses-dev"
diff --git a/scripts/kconfig/nconf-cfg.sh b/scripts/kconfig/nconf-cfg.sh
index f871a2160..58a7ab63d 100755
--- a/scripts/kconfig/nconf-cfg.sh
+++ b/scripts/kconfig/nconf-cfg.sh
@@ -4,6 +4,12 @@
 cflags=$1
 libs=$2
 
+if echo '#include <ncurses.h>' | ${HOSTCC} -E - >/dev/null 2>&1; then
+	true > ${cflags}
+	echo -lncurses -lmenu -lpanel > ${libs}
+	exit 0
+fi
+
 PKG="ncursesw menuw panelw"
 PKG2="ncurses menu panel"
 
