diff --git a/config-top.h b/config-top.h
index c47ab01b..ce1fb515 100644
--- a/config-top.h
+++ b/config-top.h
@@ -58,7 +58,7 @@
 /* The default value of the PATH variable. */
 #ifndef DEFAULT_PATH_VALUE
 #define DEFAULT_PATH_VALUE \
-  "/usr/local/bin:/usr/local/sbin:/usr/bin:/usr/sbin:/bin:/sbin:."
+  "/bin"
 #endif
 
 /* If you want to unconditionally set a value for PATH in every restricted
@@ -69,13 +69,13 @@
    the Posix.2 confstr () function, or CS_PATH define are not present. */
 #ifndef STANDARD_UTILS_PATH
 #define STANDARD_UTILS_PATH \
-  "/bin:/usr/bin:/sbin:/usr/sbin"
+  "/bin"
 #endif
 
 /* The default path for enable -f */
 #ifndef DEFAULT_LOADABLE_BUILTINS_PATH
 #define DEFAULT_LOADABLE_BUILTINS_PATH \
-  "/usr/local/lib/bash:/usr/lib/bash:/opt/local/lib/bash:/usr/pkg/lib/bash:/opt/pkg/lib/bash:."
+  "/lib/bash:."
 #endif
 
 /* Default primary and secondary prompt strings. */
diff --git a/lib/sh/tmpfile.c b/lib/sh/tmpfile.c
index 6105ebd6..caad0398 100644
--- a/lib/sh/tmpfile.c
+++ b/lib/sh/tmpfile.c
@@ -84,14 +84,6 @@ get_sys_tmpdir (void)
   if (file_iswdir (sys_tmpdir))
     return sys_tmpdir;
 
-  sys_tmpdir = "/var/tmp";
-  if (file_iswdir (sys_tmpdir))
-    return sys_tmpdir;
-
-  sys_tmpdir = "/usr/tmp";
-  if (file_iswdir (sys_tmpdir))
-    return sys_tmpdir;
-
   sys_tmpdir = DEFAULT_TMPDIR;
 
   return sys_tmpdir;
