diff --git a/config-top.h b/config-top.h
index 735f75e..c038885 100644
--- a/config-top.h
+++ b/config-top.h
@@ -63,7 +63,7 @@
 /* The default value of the PATH variable. */
 #ifndef DEFAULT_PATH_VALUE
 #define DEFAULT_PATH_VALUE \
-  "/usr/local/bin:/usr/local/sbin:/usr/bin:/usr/sbin:/bin:/sbin:."
+  "/bin"
 #endif
 
 /* If you want to unconditionally set a value for PATH in every restricted
@@ -74,7 +74,7 @@
    the Posix.2 confstr () function, or CS_PATH define are not present. */
 #ifndef STANDARD_UTILS_PATH
 #define STANDARD_UTILS_PATH \
-  "/bin:/usr/bin:/sbin:/usr/sbin:/etc:/usr/etc"
+  "/bin"
 #endif
 
 /* Default primary and secondary prompt strings. */
diff --git a/lib/sh/tmpfile.c b/lib/sh/tmpfile.c
index ef8b067..a5db680 100644
--- a/lib/sh/tmpfile.c
+++ b/lib/sh/tmpfile.c
@@ -78,14 +78,6 @@ get_sys_tmpdir ()
   if (file_iswdir (sys_tmpdir))
     return sys_tmpdir;
 
-  sys_tmpdir = "/var/tmp";
-  if (file_iswdir (sys_tmpdir))
-    return sys_tmpdir;
-
-  sys_tmpdir = "/usr/tmp";
-  if (file_iswdir (sys_tmpdir))
-    return sys_tmpdir;
-
   sys_tmpdir = DEFAULT_TMPDIR;
 
   return sys_tmpdir;
