diff --git a/examples/loadables/sync.c b/examples/loadables/sync.c
index 4fbeee1..b256df1 100644
--- a/examples/loadables/sync.c
+++ b/examples/loadables/sync.c
@@ -24,22 +24,60 @@
 #include <unistd.h>
 #endif
 
+#include <errno.h>
+#include <fcntl.h>
+
 #include "builtins.h"
 #include "shell.h"
-#include "bashgetopt.h"
+#include "common.h"
+
+#if !defined (errno)
+extern int errno;
+#endif
 
 int
 sync_builtin (list)
      WORD_LIST *list;
 {
-  sync();
-  return (EXECUTION_SUCCESS);
+  int fd, result = EXECUTION_SUCCESS;
+
+  if (!list)
+    sync();
+  else if (!strcmp (list->word->word, "--"))
+    list = list->next;
+
+  while (list)
+    {
+      fd = open (list->word->word, O_WRONLY);
+      if (fd < 0)
+        fd = open (list->word->word, O_RDONLY);
+      if (fd < 0)
+        {
+          file_error (list->word->word);
+          result = EXECUTION_FAILURE;
+        }
+      else
+        {
+          if (fsync (fd) < 0)
+            {
+              builtin_error ("cannot sync file `%s': %s", list->word->word, strerror (errno));
+              result = EXECUTION_FAILURE;
+            }
+          close (fd);
+        }
+      list = list->next;
+    }
+  return result;
 }
 
 char *sync_doc[] = {
-	"Sync disks.",
-	""
-	"Force completion of pending disk writes",
+	"Sync disks or specified files.",
+	"",
+	"Force completion of pending disk writes.  If one or more files are",
+	"specified, only force completion of pending writes to those files.",
+	"",
+	"Exit Status:",
+	"Returns success unless any specified files could not be synced.",
 	(char *)NULL
 };
 
@@ -48,6 +86,6 @@ struct builtin sync_struct = {
 	sync_builtin,		/* function implementing the builtin */
 	BUILTIN_ENABLED,	/* initial flags for builtin */
 	sync_doc,		/* array of long documentation strings. */
-	"sync",			/* usage synopsis; becomes short_doc */
+	"sync [file ...]",	/* usage synopsis; becomes short_doc */
 	0			/* reserved for internal use */
 };
