diff --git a/pathnames.h b/pathnames.h
index 8713153d..14fe125b 100644
--- a/pathnames.h
+++ b/pathnames.h
@@ -71,7 +71,7 @@
  * Directory spec for ssh-agent sockets and agent sockets forwarded by
  * sshd reside. This directory should not be world-readable.
  */
-#define _PATH_SSH_AGENT_SOCKET_DIR	"user:" _PATH_SSH_USER_DIR "/agent"
+#define _PATH_SSH_AGENT_SOCKET_DIR	"shared:/tmp"
 
 /* Directory spec for ssh-agent sockets in /tmp */
 #define _PATH_SSH_AGENT_SOCKET_TMPDIR	"shared:/tmp"
