diff --git a/auth-rhosts.c b/auth-rhosts.c
index 5672467..633b07e 100644
--- a/auth-rhosts.c
+++ b/auth-rhosts.c
@@ -194,7 +194,11 @@ auth_rhosts2(struct passwd *pw, const char *client_user, const char *hostname,
 {
 	char *path = NULL;
 	struct stat st;
-	static const char * const rhosts_files[] = {".shosts", ".rhosts", NULL};
+	static const char * const rhosts_files[] = {
+		_PATH_SSH_USER_DIR "/shosts",
+		_PATH_SSH_USER_DIR "/rhosts",
+		NULL
+	};
 	u_int rhosts_file_index;
 	int r;
 
@@ -310,7 +314,8 @@ auth_rhosts2(struct passwd *pw, const char *client_user, const char *hostname,
 		 */
 		if (options.ignore_rhosts == IGNORE_RHOSTS_YES ||
 		    (options.ignore_rhosts == IGNORE_RHOSTS_SHOSTS &&
-		    strcmp(rhosts_files[rhosts_file_index], ".shosts") != 0)) {
+		    strcmp(rhosts_files[rhosts_file_index],
+				_PATH_SSH_USER_DIR "/shosts") != 0)) {
 			auth_debug_add("Server has been configured to "
 			    "ignore %.100s.", rhosts_files[rhosts_file_index]);
 			free(path);
diff --git a/hostfile.c b/hostfile.c
index c5669c7..7f6b5ea 100644
--- a/hostfile.c
+++ b/hostfile.c
@@ -475,13 +475,14 @@ write_host_entry(FILE *f, const char *host, const char *ip,
 void
 hostfile_create_user_ssh_dir(const char *filename, int notify)
 {
-	char *dotsshdir = NULL, *p;
+	char *configdir = NULL, *dotsshdir = NULL, *p;
 	size_t len;
 	struct stat st;
 
 	if ((p = strrchr(filename, '/')) == NULL)
 		return;
 	len = p - filename;
+	configdir = tilde_expand_filename("~/" _PATH_CFG_USER_DIR, getuid());
 	dotsshdir = tilde_expand_filename("~/" _PATH_SSH_USER_DIR, getuid());
 	if (strlen(dotsshdir) > len || strncmp(filename, dotsshdir, len) != 0)
 		goto out; /* not ~/.ssh prefixed */
@@ -493,7 +494,11 @@ hostfile_create_user_ssh_dir(const char *filename, int notify)
 #ifdef WITH_SELINUX
 		ssh_selinux_setfscreatecon(dotsshdir);
 #endif
-		if (mkdir(dotsshdir, 0700) == -1)
+		if (stat(configdir, &st) == -1)
+			if (mkdir(configdir, 0777) == -1) /* respect umask */
+				error("Could not create directory '%.200s' (%s).",
+				    configdir, strerror(errno));
+		if (mkdir(dotsshdir, 0777) == -1) /* respect umask */
 			error("Could not create directory '%.200s' (%s).",
 			    dotsshdir, strerror(errno));
 		else if (notify)
@@ -503,6 +508,7 @@ hostfile_create_user_ssh_dir(const char *filename, int notify)
 #endif
 	}
  out:
+	free(configdir);
 	free(dotsshdir);
 }
 
diff --git a/pathnames.h b/pathnames.h
index f7ca5a7..6d02690 100644
--- a/pathnames.h
+++ b/pathnames.h
@@ -57,7 +57,8 @@
  * The directory in user's home directory in which the files reside. The
  * directory should be world-readable (though not all files are).
  */
-#define _PATH_SSH_USER_DIR		".ssh"
+#define _PATH_SSH_USER_DIR		".config/ssh"
+#define _PATH_CFG_USER_DIR		".config"
 
 /*
  * Per-user file containing host keys of known hosts.  This file need not be
