diff --git a/misc.c b/misc.c
index dd0bd032..9b123641 100644
--- a/misc.c
+++ b/misc.c
@@ -1199,6 +1199,7 @@ int
 tilde_expand(const char *filename, uid_t uid, char **retp)
 {
 	char *ocopy = NULL, *copy, *s = NULL;
+	const char *home = getenv("HOME");
 	const char *path = NULL, *user = NULL;
 	struct passwd *pw;
 	size_t len;
@@ -1236,15 +1237,19 @@ tilde_expand(const char *filename, uid_t uid, char **retp)
 			error_f("No such user %s", user);
 			goto out;
 		}
-	} else if ((pw = getpwuid(uid)) == NULL) {
-		error_f("No such uid %ld", (long)uid);
-		goto out;
+		home = pw->pw_dir;
+	} else if (home == NULL || uid != getuid()) {
+		if ((pw = getpwuid(uid)) == NULL) {
+			error_f("No such uid %ld", (long)uid);
+			goto out;
+		}
+		home = pw->pw_dir;
 	}
 
 	/* Make sure directory has a trailing '/' */
-	slash = (len = strlen(pw->pw_dir)) == 0 || pw->pw_dir[len - 1] != '/';
+	slash = (len = strlen(home)) == 0 || home[len - 1] != '/';
 
-	if ((r = xasprintf(&s, "%s%s%s", pw->pw_dir,
+	if ((r = xasprintf(&s, "%s%s%s", home,
 	    slash ? "/" : "", path != NULL ? path : "")) <= 0) {
 		error_f("xasprintf failed");
 		goto out;
diff --git a/ssh-add.c b/ssh-add.c
index 0035cb84..84373e90 100644
--- a/ssh-add.c
+++ b/ssh-add.c
@@ -1027,8 +1027,8 @@ main(int argc, char **argv)
 		}
 
 		for (i = 0; default_files[i]; i++) {
-			snprintf(buf, sizeof(buf), "%s/%s", pw->pw_dir,
-			    default_files[i]);
+			snprintf(buf, sizeof(buf), "%s/%s",
+			    getenv("HOME") ?: pw->pw_dir, default_files[i]);
 			if (stat(buf, &st) == -1)
 				continue;
 			if (do_file(agent_fd, deleting, key_only, cert_only,
diff --git a/ssh-keygen.c b/ssh-keygen.c
index 89c3ed28..139d49cc 100644
--- a/ssh-keygen.c
+++ b/ssh-keygen.c
@@ -299,7 +299,7 @@ ask_filename(struct passwd *pw, const char *prompt)
 		}
 	}
 	snprintf(identity_file, sizeof(identity_file),
-	    "%s/%s", pw->pw_dir, name);
+	    "%s/%s", getenv("HOME") ?: pw->pw_dir, name);
 	printf("%s (%s): ", prompt, identity_file);
 	fflush(stdout);
 	if (fgets(buf, sizeof(buf), stdin) == NULL)
diff --git a/ssh.c b/ssh.c
index dc4886d0..1d9fbc30 100644
--- a/ssh.c
+++ b/ssh.c
@@ -576,8 +576,8 @@ process_config_files(const char *host_name, struct passwd *pw,
 			fatal("Can't open user config file %.100s: "
 			    "%.100s", config, strerror(errno));
 	} else {
-		r = snprintf(buf, sizeof buf, "%s/%s", pw->pw_dir,
-		    _PATH_SSH_USER_CONFFILE);
+		r = snprintf(buf, sizeof buf, "%s/%s",
+		    getenv("HOME") ?: pw->pw_dir, _PATH_SSH_USER_CONFFILE);
 		if (r > 0 && (size_t)r < sizeof(buf))
 			(void)read_config_file(buf, pw, host, host_name, cmd,
 			    &options, SSHCONF_CHECKPERM | SSHCONF_USERCONF |
@@ -1443,7 +1443,7 @@ main(int ac, char **av)
 	    options.host_key_alias : options.host_arg);
 	cinfo->host_arg = xstrdup(options.host_arg);
 	cinfo->remhost = xstrdup(host);
-	cinfo->homedir = xstrdup(pw->pw_dir);
+	cinfo->homedir = xstrdup(getenv("HOME") ?: pw->pw_dir);
 	cinfo->locuser = xstrdup(pw->pw_name);
 	cinfo->jmphost = xstrdup(options.jump_host == NULL ?
 	    "" : options.jump_host);
