diff --git a/sshd.c b/sshd.c
index da2aba93..4fe6dce6 100644
--- a/sshd.c
+++ b/sshd.c
@@ -1477,12 +1477,11 @@ main(int ac, char **av)
 		}
 	}
 
-	if (getexecpath(execpath, sizeof execpath) != 0) {
-		if (strlcpy(execpath, av[0], sizeof execpath) >=
-		    sizeof execpath) {
-			fprintf(stderr, "execution path is too long\n");
-			exit(1);
-		}
+	if ((r = readlink("/proc/self/exe", execpath, sizeof execpath)) < 0)
+		r = strlcpy(execpath, av[0], sizeof execpath);
+	if (r >= sizeof execpath) {
+		fprintf(stderr, "execution path is too long\n");
+		exit(1);
 	}
 	if (!test_flag && !inetd_flag && !do_dump_cfg && !path_absolute(execpath))
 		fatal("sshd requires execution with an absolute path");
