diff --git a/docs/cmdline-opts/Makefile.am b/docs/cmdline-opts/Makefile.am
index 81a4afe..62c776f 100644
--- a/docs/cmdline-opts/Makefile.am
+++ b/docs/cmdline-opts/Makefile.am
@@ -27,8 +27,6 @@ AUTOMAKE_OPTIONS = foreign no-dependencies
 MANPAGE = curl.1
 ASCIIPAGE = curl.txt
 
-man_MANS = $(MANPAGE)
-
 include Makefile.inc
 
 EXTRA_DIST = $(DPAGES) MANPAGE.md $(SUPPORT) CMakeLists.txt mainpage.idx
@@ -42,6 +40,7 @@ MANAGEN=$(abs_top_srcdir)/scripts/managen
 
 if BUILD_DOCS
 CLEANFILES = $(MANPAGE) $(ASCIIPAGE)
+man_MANS = $(MANPAGE)
 
 all: $(MANPAGE) $(ASCIIPAGE)
 
diff --git a/docs/cmdline-opts/Makefile.in b/docs/cmdline-opts/Makefile.in
index e777cfc..8167e77 100644
--- a/docs/cmdline-opts/Makefile.in
+++ b/docs/cmdline-opts/Makefile.in
@@ -453,7 +453,6 @@ top_srcdir = @top_srcdir@
 AUTOMAKE_OPTIONS = foreign no-dependencies
 MANPAGE = curl.1
 ASCIIPAGE = curl.txt
-man_MANS = $(MANPAGE)
 SUPPORT = \
   _AUTHORS.md \
   _BUGS.md \
@@ -742,6 +741,7 @@ GN_1 =
 GN_ = $(GN_0)
 MANAGEN = $(abs_top_srcdir)/scripts/managen
 @BUILD_DOCS_TRUE@CLEANFILES = $(MANPAGE) $(ASCIIPAGE)
+@BUILD_DOCS_TRUE@man_MANS = $(MANPAGE)
 all: all-am
 
 .SUFFIXES:
