diff --git a/configure b/configure
index 61fcd037..667f35d6 100755
--- a/configure
+++ b/configure
@@ -7576,7 +7576,7 @@ fi
 			case "$with_libnl" in
 
 			yes|if_available)
-				incdir=-I/usr/include/libnl3
+				incdir=
 				libnldir=
 				;;
 
@@ -7588,14 +7588,14 @@ fi
 				;;
 			esac
 
-			{ printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking for nl_socket_alloc in -lnl-3" >&5
-printf %s "checking for nl_socket_alloc in -lnl-3... " >&6; }
-if test ${ac_cv_lib_nl_3_nl_socket_alloc+y}
+			{ printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking for nl_socket_alloc in -lnl-tiny" >&5
+printf %s "checking for nl_socket_alloc in -lnl-tiny... " >&6; }
+if test ${ac_cv_lib_nl_tiny_nl_socket_alloc+y}
 then :
   printf %s "(cached) " >&6
 else $as_nop
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lnl-3 ${incdir} ${libnldir} -lnl-genl-3 -lnl-3  $LIBS"
+LIBS="-lnl-tiny ${incdir} ${libnldir} $LIBS"
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
@@ -7613,25 +7613,25 @@ return nl_socket_alloc ();
 _ACEOF
 if ac_fn_c_try_link "$LINENO"
 then :
-  ac_cv_lib_nl_3_nl_socket_alloc=yes
+  ac_cv_lib_nl_tiny_nl_socket_alloc=yes
 else $as_nop
-  ac_cv_lib_nl_3_nl_socket_alloc=no
+  ac_cv_lib_nl_tiny_nl_socket_alloc=no
 fi
 rm -f core conftest.err conftest.$ac_objext conftest.beam \
     conftest$ac_exeext conftest.$ac_ext
 LIBS=$ac_check_lib_save_LIBS
 fi
-{ printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: $ac_cv_lib_nl_3_nl_socket_alloc" >&5
-printf "%s\n" "$ac_cv_lib_nl_3_nl_socket_alloc" >&6; }
-if test "x$ac_cv_lib_nl_3_nl_socket_alloc" = xyes
+{ printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: $ac_cv_lib_nl_tiny_nl_socket_alloc" >&5
+printf "%s\n" "$ac_cv_lib_nl_tiny_nl_socket_alloc" >&6; }
+if test "x$ac_cv_lib_nl_tiny_nl_socket_alloc" = xyes
 then :
 
 				#
-				# Yes, we have libnl 3.x.
+				# Yes, we have libnl-tiny.
 				#
-				ADDITIONAL_LIBS="${libnldir} -lnl-genl-3 -lnl-3 $ADDITIONAL_LIBS"
-				ADDITIONAL_LIBS_STATIC="${libnldir} -lnl-genl-3 -lnl-3 $ADDITIONAL_LIBS_STATIC"
-				LIBS_PRIVATE="${libnldir} -lnl-genl-3 -lnl-3 $LIBS_PRIVATE"
+				ADDITIONAL_LIBS="${libnldir} -lnl-tiny $ADDITIONAL_LIBS"
+				ADDITIONAL_LIBS_STATIC="${libnldir} -lnl-tiny $ADDITIONAL_LIBS_STATIC"
+				LIBS_PRIVATE="${libnldir} -lnl-tiny $LIBS_PRIVATE"
 
 printf "%s\n" "#define HAVE_LIBNL 1" >>confdefs.h
 
diff --git a/configure.ac b/configure.ac
index accb2589..5e9156f9 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1048,7 +1048,7 @@ linux)
 			case "$with_libnl" in
 
 			yes|if_available)
-				incdir=-I/usr/include/libnl3
+				incdir=
 				libnldir=
 				;;
 
@@ -1060,14 +1060,14 @@ linux)
 				;;
 			esac
 
-			AC_CHECK_LIB(nl-3, nl_socket_alloc,
+			AC_CHECK_LIB(nl-tiny, nl_socket_alloc,
 			[
 				#
-				# Yes, we have libnl 3.x.
+				# Yes, we have libnl-tiny.
 				#
-				ADDITIONAL_LIBS="${libnldir} -lnl-genl-3 -lnl-3 $ADDITIONAL_LIBS"
-				ADDITIONAL_LIBS_STATIC="${libnldir} -lnl-genl-3 -lnl-3 $ADDITIONAL_LIBS_STATIC"
-				LIBS_PRIVATE="${libnldir} -lnl-genl-3 -lnl-3 $LIBS_PRIVATE"
+				ADDITIONAL_LIBS="${libnldir} -lnl-tiny $ADDITIONAL_LIBS"
+				ADDITIONAL_LIBS_STATIC="${libnldir} -lnl-tiny $ADDITIONAL_LIBS_STATIC"
+				LIBS_PRIVATE="${libnldir} -lnl-tiny $LIBS_PRIVATE"
 				AC_DEFINE(HAVE_LIBNL,1,[if libnl exists])
 				V_INCLS="$V_INCLS ${incdir}"
 			],[
@@ -1079,7 +1079,7 @@ linux)
 				if test x$with_libnl = xyes ; then
 					AC_MSG_ERROR([libnl support requested but libnl not found])
 				fi
-			], ${incdir} ${libnldir} -lnl-genl-3 -lnl-3 )
+			], ${incdir} ${libnldir} -lnl-tiny )
 		fi
 	fi
 
