diff --git a/configure b/configure
index 4512c34d..987906fb 100755
--- a/configure
+++ b/configure
@@ -7566,7 +7566,7 @@ fi
 			case "$with_libnl" in
 
 			yes|if_available)
-				incdir=-I/usr/include/libnl3
+				incdir=
 				libnldir=
 				;;
 
@@ -7578,14 +7578,14 @@ fi
 				;;
 			esac
 
-			{ printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking for nl_socket_alloc in -lnl-3" >&5
-printf %s "checking for nl_socket_alloc in -lnl-3... " >&6; }
-if test ${ac_cv_lib_nl_3_nl_socket_alloc+y}
+			{ printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking for nl_socket_alloc in -lnl-tiny" >&5
+printf %s "checking for nl_socket_alloc in -lnl-tiny... " >&6; }
+if test ${ac_cv_lib_nl_tiny_nl_socket_alloc+y}
 then :
   printf %s "(cached) " >&6
 else $as_nop
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lnl-3 ${incdir} ${libnldir} -lnl-genl-3 -lnl-3  $LIBS"
+LIBS="-lnl-tiny ${incdir} ${libnldir} $LIBS"
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
@@ -7603,25 +7603,25 @@ return nl_socket_alloc ();
 _ACEOF
 if ac_fn_c_try_link "$LINENO"
 then :
-  ac_cv_lib_nl_3_nl_socket_alloc=yes
+  ac_cv_lib_nl_tiny_nl_socket_alloc=yes
 else $as_nop
-  ac_cv_lib_nl_3_nl_socket_alloc=no
+  ac_cv_lib_nl_tiny_nl_socket_alloc=no
 fi
 rm -f core conftest.err conftest.$ac_objext conftest.beam \
     conftest$ac_exeext conftest.$ac_ext
 LIBS=$ac_check_lib_save_LIBS
 fi
-{ printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: $ac_cv_lib_nl_3_nl_socket_alloc" >&5
-printf "%s\n" "$ac_cv_lib_nl_3_nl_socket_alloc" >&6; }
-if test "x$ac_cv_lib_nl_3_nl_socket_alloc" = xyes
+{ printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: $ac_cv_lib_nl_tiny_nl_socket_alloc" >&5
+printf "%s\n" "$ac_cv_lib_nl_tiny_nl_socket_alloc" >&6; }
+if test "x$ac_cv_lib_nl_tiny_nl_socket_alloc" = xyes
 then :
 
 				#
-				# Yes, we have libnl 3.x.
+				# Yes, we have libnl-tiny.
 				#
-				ADDITIONAL_LIBS="${libnldir} -lnl-genl-3 -lnl-3 $ADDITIONAL_LIBS"
-				ADDITIONAL_LIBS_STATIC="${libnldir} -lnl-genl-3 -lnl-3 $ADDITIONAL_LIBS_STATIC"
-				LIBS_PRIVATE="${libnldir} -lnl-genl-3 -lnl-3 $LIBS_PRIVATE"
+				ADDITIONAL_LIBS="${libnldir} -lnl-tiny $ADDITIONAL_LIBS"
+				ADDITIONAL_LIBS_STATIC="${libnldir} -lnl-tiny $ADDITIONAL_LIBS_STATIC"
+				LIBS_PRIVATE="${libnldir} -lnl-tiny $LIBS_PRIVATE"
 
 printf "%s\n" "#define HAVE_LIBNL 1" >>confdefs.h
 
diff --git a/configure.ac b/configure.ac
index 1fdecc13..b6a72a55 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1038,7 +1038,7 @@ linux)
 			case "$with_libnl" in
 
 			yes|if_available)
-				incdir=-I/usr/include/libnl3
+				incdir=
 				libnldir=
 				;;
 
@@ -1050,14 +1050,14 @@ linux)
 				;;
 			esac
 
-			AC_CHECK_LIB(nl-3, nl_socket_alloc,
+			AC_CHECK_LIB(nl-tiny, nl_socket_alloc,
 			[
 				#
-				# Yes, we have libnl 3.x.
+				# Yes, we have libnl-tiny.
 				#
-				ADDITIONAL_LIBS="${libnldir} -lnl-genl-3 -lnl-3 $ADDITIONAL_LIBS"
-				ADDITIONAL_LIBS_STATIC="${libnldir} -lnl-genl-3 -lnl-3 $ADDITIONAL_LIBS_STATIC"
-				LIBS_PRIVATE="${libnldir} -lnl-genl-3 -lnl-3 $LIBS_PRIVATE"
+				ADDITIONAL_LIBS="${libnldir} -lnl-tiny $ADDITIONAL_LIBS"
+				ADDITIONAL_LIBS_STATIC="${libnldir} -lnl-tiny $ADDITIONAL_LIBS_STATIC"
+				LIBS_PRIVATE="${libnldir} -lnl-tiny $LIBS_PRIVATE"
 				AC_DEFINE(HAVE_LIBNL,1,[if libnl exists])
 				V_INCLS="$V_INCLS ${incdir}"
 			],[
@@ -1069,7 +1069,7 @@ linux)
 				if test x$with_libnl = xyes ; then
 					AC_MSG_ERROR([libnl support requested but libnl not found])
 				fi
-			], ${incdir} ${libnldir} -lnl-genl-3 -lnl-3 )
+			], ${incdir} ${libnldir} -lnl-tiny )
 		fi
 	fi
 
diff --git a/pcap-linux.c b/pcap-linux.c
index 20802e43..f7bf7030 100644
--- a/pcap-linux.c
+++ b/pcap-linux.c
@@ -627,12 +627,11 @@ get_if_type(pcap_t *handle, int sock_fd, struct nl80211_state *state,
 		return PCAP_ERROR;
 	}
 
-	genlmsg_put(msg, NL_AUTO_PORT, NL_AUTO_SEQ,
-		    genl_family_get_id(state->nl80211), 0,
+	genlmsg_put(msg, 0, 0, genl_family_get_id(state->nl80211), 0,
 		    0, NL80211_CMD_GET_INTERFACE, 0);
 	NLA_PUT_U32(msg, NL80211_ATTR_IFINDEX, ifindex);
 
-	err = nl_send_auto(state->nl_sock, msg);
+	err = nl_send_auto_complete(state->nl_sock, msg);
 	nlmsg_free(msg);
 	if (err < 0) {
 		if (err == -NLE_FAILURE) {
@@ -650,7 +649,7 @@ get_if_type(pcap_t *handle, int sock_fd, struct nl80211_state *state,
 			 * available.
 			 */
 			snprintf(handle->errbuf, PCAP_ERRBUF_SIZE,
-			    "%s: nl_send_auto failed getting interface type: %s",
+			    "%s: nl_send_auto_complete failed getting interface type: %s",
 			    device, nl_geterror(-err));
 			return PCAP_ERROR;
 		}
@@ -716,8 +715,7 @@ add_mon_if(pcap_t *handle, int sock_fd, struct nl80211_state *state,
 		return PCAP_ERROR;
 	}
 
-	genlmsg_put(msg, NL_AUTO_PORT, NL_AUTO_SEQ,
-		    genl_family_get_id(state->nl80211), 0,
+	genlmsg_put(msg, 0, 0, genl_family_get_id(state->nl80211), 0,
 		    0, NL80211_CMD_NEW_INTERFACE, 0);
 	NLA_PUT_U32(msg, NL80211_ATTR_IFINDEX, ifindex);
 DIAG_OFF_NARROWING
@@ -725,7 +723,30 @@ DIAG_OFF_NARROWING
 DIAG_ON_NARROWING
 	NLA_PUT_U32(msg, NL80211_ATTR_IFTYPE, NL80211_IFTYPE_MONITOR);
 
-	err = nl_send_sync(state->nl_sock, msg); // calls nlmsg_free()
+	err = nl_send_auto_complete(state->nl_sock, msg);
+	nlmsg_free(msg);
+	if (err < 0) {
+		if (err == -NLE_FAILURE) {
+			/*
+			 * Device not available; our caller should just
+			 * keep trying.  (libnl 2.x maps ENFILE to
+			 * NLE_FAILURE; it can also map other errors
+			 * to that, but there's not much we can do
+			 * about that.)
+			 */
+			return 0;
+		} else {
+			/*
+			 * Real failure, not just "that device is not
+			 * available.
+			 */
+			snprintf(handle->errbuf, PCAP_ERRBUF_SIZE,
+			    "%s: nl_send_auto_complete failed adding %s interface: %s",
+			    device, mondevice, nl_geterror(-err));
+			return PCAP_ERROR;
+		}
+	}
+	err = nl_wait_for_ack(state->nl_sock);
 	if (err < 0) {
 		switch (err) {
 
@@ -755,7 +776,7 @@ DIAG_ON_NARROWING
 			 * error message from libnl.
 			 */
 			snprintf(handle->errbuf, PCAP_ERRBUF_SIZE,
-			    "%s: nl_send_sync failed adding %s interface: %s",
+			    "%s: nl_wait_for_ack failed adding %s interface: %s",
 			    device, mondevice, nl_geterror(-err));
 			return PCAP_ERROR;
 		}
@@ -807,15 +828,22 @@ del_mon_if(pcap_t *handle, int sock_fd, struct nl80211_state *state,
 		return PCAP_ERROR;
 	}
 
-	genlmsg_put(msg, NL_AUTO_PORT, NL_AUTO_SEQ,
-		    genl_family_get_id(state->nl80211), 0,
+	genlmsg_put(msg, 0, 0, genl_family_get_id(state->nl80211), 0,
 		    0, NL80211_CMD_DEL_INTERFACE, 0);
 	NLA_PUT_U32(msg, NL80211_ATTR_IFINDEX, ifindex);
 
-	err = nl_send_sync(state->nl_sock, msg); // calls nlmsg_free()
+	err = nl_send_auto_complete(state->nl_sock, msg);
+	nlmsg_free(msg);
+	if (err < 0) {
+		snprintf(handle->errbuf, PCAP_ERRBUF_SIZE,
+		    "%s: nl_send_auto_complete failed deleting %s interface: %s",
+		    device, mondevice, nl_geterror(-err));
+		return PCAP_ERROR;
+	}
+	err = nl_wait_for_ack(state->nl_sock);
 	if (err < 0) {
 		snprintf(handle->errbuf, PCAP_ERRBUF_SIZE,
-		    "%s: nl_send_sync failed deleting %s interface: %s",
+		    "%s: nl_wait_for_ack failed deleting %s interface: %s",
 		    device, mondevice, nl_geterror(-err));
 		return PCAP_ERROR;
 	}
