diff --git a/lib/commonio.c b/lib/commonio.c
index da530ef2..1f09a981 100644
--- a/lib/commonio.c
+++ b/lib/commonio.c
@@ -231,7 +231,7 @@ static /*@null@*/ /*@dependent@*/FILE *fmkstemp_set_perms (
 	return NULL;
 }
 
-
+#ifdef WITH_BACKUPS
 static int create_backup (const char *name, FILE * fp)
 {
 	char  tmpf[PATH_MAX], target[PATH_MAX];
@@ -285,6 +285,7 @@ static int create_backup (const char *name, FILE * fp)
 	(void) utime(target, &ub);
 	return 0;
 }
+#endif
 
 
 static void free_linked_list (struct commonio_db *db)
@@ -867,6 +868,7 @@ commonio_close(struct commonio_db *db, MAYBE_UNUSED bool process_selinux)
 			goto fail;
 		}
 
+#ifdef WITH_BACKUPS
 		/*
 		 * Create backup file.
 		 */
@@ -879,6 +881,7 @@ commonio_close(struct commonio_db *db, MAYBE_UNUSED bool process_selinux)
 		if (create_backup(db->filename, db->fp) != 0) {
 			errors = true;
 		}
+#endif
 
 		if (fclose (db->fp) != 0) {
 			errors = true;
diff --git a/src/useradd.c b/src/useradd.c
index 5404b46c..db4e93bf 100644
--- a/src/useradd.c
+++ b/src/useradd.c
@@ -715,6 +715,7 @@ set_defaults(void)
 		goto err_free_def;
 	}
 
+#ifdef WITH_BACKUPS
 	/*
 	 * Rename the current default file to its backup name.
 	 */
@@ -725,6 +726,7 @@ set_defaults(void)
 		unlink (new_file);
 		goto err_free_def;
 	}
+#endif
 
 	/*
 	 * Rename the new default file to its correct name.
diff --git a/src/vipw.c b/src/vipw.c
index 783e2ded..a8ac71a9 100644
--- a/src/vipw.c
+++ b/src/vipw.c
@@ -438,8 +438,10 @@ vipwedit (const char *file, int (*file_lock) (void), int (*file_unlock) (bool))
 #ifdef WITH_TCB
 	}
 #endif				/* WITH_TCB */
+#ifdef WITH_BACKUPS
 	unlink (filebackup);
 	link (file, filebackup);
+#endif
 	if (rename (to_rename, file) == -1) {
 		eprinte(_("%s: can't restore %s (your changes are in %s)"),
 		        Prog, file, to_rename);
