diff --git a/lib/obscure.c b/lib/obscure.c
index d508200f..e3a93852 100644
--- a/lib/obscure.c
+++ b/lib/obscure.c
@@ -150,6 +150,9 @@ static /*@observer@*//*@null@*/const char *obscure_msg (
 	}
 
 	result = getdef_str ("ENCRYPT_METHOD");
+	if (NULL == result) {
+		result = "BCRYPT";
+	}
 	if (NULL == result) {
 	/* The traditional crypt() truncates passwords to 8 chars.  It is
 	   possible to circumvent the above checks by choosing an easy
diff --git a/lib/salt.c b/lib/salt.c
index 753e9d45..2936addc 100644
--- a/lib/salt.c
+++ b/lib/salt.c
@@ -369,6 +369,9 @@ static /*@observer@*/const char *gensalt (size_t salt_size)
 		method = meth;
 	else {
 		method = getdef_str ("ENCRYPT_METHOD");
+		if (NULL == method) {
+			method = "BCRYPT";
+		}
 		if (NULL == method) {
 			method = getdef_bool ("MD5_CRYPT_ENAB") ? "MD5" : "DES";
 		}
diff --git a/src/passwd.c b/src/passwd.c
index cc79960a..0685cce6 100644
--- a/src/passwd.c
+++ b/src/passwd.c
@@ -238,6 +238,9 @@ static int new_password (const struct passwd *pw)
 	 * for initial login passwords.
 	 */
 	method = getdef_str ("ENCRYPT_METHOD");
+	if (NULL == method) {
+		method = "BCRYPT";
+	}
 	if (NULL == method) {
 		if (!getdef_bool ("MD5_CRYPT_ENAB")) {
 			pass_max_len = getdef_num ("PASS_MAX_LEN", 8);
