diff --git a/lib/obscure.c b/lib/obscure.c
index c572b96a..a9e1f703 100644
--- a/lib/obscure.c
+++ b/lib/obscure.c
@@ -216,9 +216,7 @@ obscure_get_range(int *minlen, int *maxlen)
 
 	method = getdef_str ("ENCRYPT_METHOD");
 	if (NULL == method) {
-		if (getdef_bool ("MD5_CRYPT_ENAB")) {
-			return;
-		}
+		return; /* BCRYPT is the default */
 	} else {
 		if (   streq(method, "MD5")
 #ifdef USE_SHA_CRYPT
diff --git a/lib/salt.c b/lib/salt.c
index 1efeb2ce..428feff9 100644
--- a/lib/salt.c
+++ b/lib/salt.c
@@ -370,7 +370,7 @@ static /*@observer@*/const char *gensalt (size_t salt_size)
 	else {
 		method = getdef_str ("ENCRYPT_METHOD");
 		if (NULL == method) {
-			method = getdef_bool ("MD5_CRYPT_ENAB") ? "MD5" : "DES";
+			method = "BCRYPT";
 		}
 	}
 
