diff --git a/lib/getdef.c b/lib/getdef.c
index 4616b65e..793ee326 100644
--- a/lib/getdef.c
+++ b/lib/getdef.c
@@ -214,12 +214,17 @@ static void def_load (void);
 
 
 /*
- * getdef_bool - get boolean value from table of definitions.
+ * getdef_bool[_dflt] - get boolean value from table of definitions.
  *
- * Return TRUE if specified item is defined as "yes", else FALSE.
+ * Return TRUE if specified item is defined as "yes", else FALSE or dflt.
  */
 
 bool getdef_bool (const char *item)
+{
+	return getdef_bool_dflt(item, false);
+}
+
+bool getdef_bool_dflt (const char *item, bool dflt)
 {
 	struct itemdef *d;
 
@@ -229,7 +234,7 @@ bool getdef_bool (const char *item)
 
 	d = def_find (item, NULL);
 	if ((NULL == d) || (NULL == d->value)) {
-		return false;
+		return dflt;
 	}
 
 	return strcaseeq(d->value, "yes");
diff --git a/lib/getdef.h b/lib/getdef.h
index e687a2ee..bb8d4a2c 100644
--- a/lib/getdef.h
+++ b/lib/getdef.h
@@ -17,6 +17,7 @@
 
 /* getdef.c */
 extern bool getdef_bool (const char *);
+extern bool getdef_bool_dflt (const char *, bool);
 extern long getdef_long (const char *, long);
 extern int getdef_num (const char *, int);
 extern unsigned long getdef_ulong (const char *, unsigned long);
diff --git a/lib/limits.c b/lib/limits.c
index 3fabe54d..4ebb395a 100644
--- a/lib/limits.c
+++ b/lib/limits.c
@@ -447,7 +447,7 @@ void setup_limits (const struct passwd *info)
 {
 	char *cp;
 
-	if (getdef_bool ("USERGROUPS_ENAB")) {
+	if (getdef_bool_dflt ("USERGROUPS_ENAB", true)) {
 		setup_usergroups (info);
 	}
 
diff --git a/lib/salt.c b/lib/salt.c
index f80a9911..af1acf7e 100644
--- a/lib/salt.c
+++ b/lib/salt.c
@@ -353,7 +353,7 @@ static /*@observer@*/const char *gensalt (size_t salt_size)
 
 	bzero(result, GENSALT_SETTING_SIZE);
 
-	method = meth ?: getdef_str("ENCRYPT_METHOD") ?: "SHA512";
+	method = meth ?: getdef_str("ENCRYPT_METHOD") ?: "BCRYPT";
 
 	if (streq(method, "SHA256")) {
 		MAGNUM(result, '5');
diff --git a/src/useradd.c b/src/useradd.c
index db4e93bf..36c07b1f 100644
--- a/src/useradd.c
+++ b/src/useradd.c
@@ -1434,7 +1434,7 @@ static void process_flags (int argc, char **argv, struct option_flags *flags)
 
 	if (!gflg && !Nflg && !Uflg) {
 		/* Get the settings from login.defs */
-		Uflg = getdef_bool ("USERGROUPS_ENAB");
+		Uflg = getdef_bool_dflt ("USERGROUPS_ENAB", true);
 	}
 
 	/*
diff --git a/src/userdel.c b/src/userdel.c
index 20f28281..d38110db 100644
--- a/src/userdel.c
+++ b/src/userdel.c
@@ -221,7 +221,7 @@ static void update_groups (bool process_selinux)
 			user_name, ngrp->gr_name);
 	}
 
-	if (getdef_bool ("USERGROUPS_ENAB")) {
+	if (getdef_bool_dflt ("USERGROUPS_ENAB", true)) {
 		remove_usergroup (process_selinux);
 	}
 
