diff --git a/src/newgidmap.c b/src/newgidmap.c
index ebf16cd..78a5079 100644
--- a/src/newgidmap.c
+++ b/src/newgidmap.c
@@ -32,6 +32,12 @@ static bool verify_range(struct passwd *pw, struct map_range *range, bool *allow
 	if (range->count == 0)
 		return false;
 
+	/* Root is allowed to map any gid */
+	if (pw->pw_uid == 0) {
+		*allow_setgroups = true;
+		return true;
+	}
+
 	/* Test /etc/subgid. If the mapping is valid then we allow setgroups. */
 	if (have_sub_gids(pw->pw_name, range->lower, range->count)) {
 		*allow_setgroups = true;
@@ -219,7 +225,7 @@ int main(int argc, char **argv)
 		return EXIT_FAILURE;
 	}
 
-	if (!sub_gid_open(O_RDONLY)) {
+	if (pw->pw_uid != 0 && !sub_gid_open(O_RDONLY)) {
 		return EXIT_FAILURE;
 	}
 
@@ -232,7 +238,8 @@ int main(int argc, char **argv)
 
 	write_setgroups(proc_dir_fd, allow_setgroups);
 	write_mapping(proc_dir_fd, ranges, mappings, "gid_map", pw->pw_uid);
-	sub_gid_close();
+	if (pw->pw_uid != 0)
+		sub_gid_close();
 
 	return EXIT_SUCCESS;
 }
diff --git a/src/newuidmap.c b/src/newuidmap.c
index 9aa2ddc..a769767 100644
--- a/src/newuidmap.c
+++ b/src/newuidmap.c
@@ -31,6 +31,10 @@ static bool verify_range(struct passwd *pw, struct map_range *range)
 	if (range->count == 0)
 		return false;
 
+	/* Root is allowed to map any uid */
+	if (pw->pw_uid == 0)
+		return true;
+
 	/* Test /etc/subuid */
 	if (have_sub_uids(pw->pw_name, range->lower, range->count))
 		return true;
@@ -149,7 +153,7 @@ int main(int argc, char **argv)
 		return EXIT_FAILURE;
 	}
 
-	if (!sub_uid_open(O_RDONLY)) {
+	if (pw->pw_uid != 0 && !sub_uid_open(O_RDONLY)) {
 		return EXIT_FAILURE;
 	}
 
@@ -161,7 +165,8 @@ int main(int argc, char **argv)
 	verify_ranges(pw, ranges, mappings);
 
 	write_mapping(proc_dir_fd, ranges, mappings, "uid_map", pw->pw_uid);
-	sub_uid_close();
+	if (pw->pw_uid != 0)
+		sub_uid_close();
 
 	return EXIT_SUCCESS;
 }
