diff --git a/lib/nss.c b/lib/nss.c
index af3e95a..5f9d155 100644
--- a/lib/nss.c
+++ b/lib/nss.c
@@ -10,6 +10,8 @@
 #include "../libsubid/subid.h"
 #include "shadowlog_internal.h"
 
+#ifdef __GLIBC__
+
 #define NSSWITCH "/etc/nsswitch.conf"
 
 // NSS plugin handling for subids
@@ -148,3 +150,12 @@ struct subid_nss_ops *get_subid_nss_handle() {
 	nss_init(NULL);
 	return subid_nss;
 }
+
+#else /* __GLIBC__ */
+
+bool nss_is_initialized(void) { return true; }
+void nss_exit(void) { return; }
+void nss_init(const char *nsswitch_path) { return; }
+struct subid_nss_ops *get_subid_nss_handle(void) { return NULL; };
+
+#endif /* __GLIBC__ */
