diff --git a/lib/btrfs.c b/lib/btrfs.c
index 03915981..eae0877c 100644
--- a/lib/btrfs.c
+++ b/lib/btrfs.c
@@ -14,8 +14,7 @@ static bool path_exists(const char *p)
 
 static const char *btrfs_cmd(void)
 {
-	const char *const btrfs_paths[] = {"/sbin/btrfs",
-		"/bin/btrfs", "/usr/sbin/btrfs", "/usr/bin/btrfs", NULL};
+	const char *const btrfs_paths[] = {"/bin/btrfs", NULL};
 	const char *p;
 	int i;
 
diff --git a/lib/nscd.c b/lib/nscd.c
index e2bb44e3..f74efc87 100644
--- a/lib/nscd.c
+++ b/lib/nscd.c
@@ -20,7 +20,7 @@
 int nscd_flush_cache (const char *service)
 {
 	int status, code;
-	const char *cmd = "/usr/sbin/nscd";
+	const char *cmd = "/bin/nscd";
 	const char *spawnedArgs[] = {"nscd", "-i", service, NULL};
 	const char *spawnedEnv[] = {NULL};
 
diff --git a/lib/setupenv.c b/lib/setupenv.c
index 26322163..9673a6ce 100644
--- a/lib/setupenv.c
+++ b/lib/setupenv.c
@@ -235,7 +235,7 @@ void setup_env (struct passwd *info)
 
 	if (NULL == cp) {
 		/* not specified, use a minimal default */
-		addenv ((info->pw_uid == 0) ? "PATH=/sbin:/bin:/usr/sbin:/usr/bin" : "PATH=/bin:/usr/bin", NULL);
+		addenv ("PATH=/bin", NULL);
 	} else if (strchr(cp, '=')) {
 		/* specified as name=value (PATH=...) */
 		addenv (cp, NULL);
diff --git a/src/Makefile.am b/src/Makefile.am
index 69818150..8ea7a427 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -3,7 +3,7 @@ EXTRA_DIST = \
 	.indent.pro
 
 ubindir = ${prefix}/bin
-usbindir = ${prefix}/sbin
+usbindir = ${prefix}/bin
 suidperms = 4755
 sgidperms = 2755
 
diff --git a/src/Makefile.in b/src/Makefile.in
index 5e64cb74..a6283e0e 100644
--- a/src/Makefile.in
+++ b/src/Makefile.in
@@ -626,7 +626,7 @@ EXTRA_DIST = \
 	.indent.pro
 
 ubindir = ${prefix}/bin
-usbindir = ${prefix}/sbin
+usbindir = ${prefix}/bin
 suidperms = 4755
 sgidperms = 2755
 AM_CPPFLAGS = \
diff --git a/src/su.c b/src/su.c
index 583979c2..a594dbee 100644
--- a/src/su.c
+++ b/src/su.c
@@ -951,7 +951,7 @@ static void set_environment (struct passwd *pw)
 
 	cp = getdef_str ((pw->pw_uid == 0) ? "ENV_SUPATH" : "ENV_PATH");
 	if (NULL == cp) {
-		addenv ((pw->pw_uid == 0) ? "PATH=/sbin:/bin:/usr/sbin:/usr/bin" : "PATH=/bin:/usr/bin", NULL);
+		addenv ("PATH=/bin", NULL);
 	} else if (strchr(cp, '=')) {
 		addenv (cp, NULL);
 	} else {
diff --git a/src/useradd.c b/src/useradd.c
index 899efe3c..f0d3ba84 100644
--- a/src/useradd.c
+++ b/src/useradd.c
@@ -1376,8 +1376,7 @@ static void process_flags (int argc, char **argv, struct option_flags *flags)
 				}
 				if (!streq(optarg, "")
 				     && '*'  != optarg[0]
-				     && !streq(optarg, "/sbin/nologin")
-				     && !streq(optarg, "/usr/sbin/nologin")
+				     && !streq(optarg, "/bin/nologin")
 				     && (   stat(optarg, &st) != 0
 				         || S_ISDIR(st.st_mode)
 				         || access(optarg, X_OK) != 0)) {
@@ -2040,7 +2039,7 @@ static void lastlog_reset (uid_t uid)
 
 static void tallylog_reset (const char *user_name)
 {
-	const char pam_tally2[] = "/sbin/pam_tally2";
+	const char pam_tally2[] = "/bin/pam_tally2";
 	const char *pname;
 	pid_t childpid;
 	int failed;
diff --git a/src/usermod.c b/src/usermod.c
index e8c9da66..6dd12906 100644
--- a/src/usermod.c
+++ b/src/usermod.c
@@ -1185,8 +1185,7 @@ process_flags(int argc, char **argv, struct option_flags *flags)
 				}
 				if (!streq(optarg, "")
 				     && '*'  != optarg[0]
-				     && !streq(optarg, "/sbin/nologin")
-				     && !streq(optarg, "/usr/sbin/nologin")
+				     && !streq(optarg, "/bin/nologin")
 				     && (   stat(optarg, &st) != 0
 				         || S_ISDIR(st.st_mode)
 				         || access(optarg, X_OK) != 0)) {
