diff --git a/lib/btrfs.c b/lib/btrfs.c
index 15b6b315..3c22b35f 100644
--- a/lib/btrfs.c
+++ b/lib/btrfs.c
@@ -10,7 +10,7 @@
 #include "prototypes.h"
 
 
-#define PATH_ENV  "/usr/bin/env"
+#define PATH_ENV  "/bin/env"
 
 
 static int run_btrfs_subvolume_cmd(const char *subcmd, const char *arg1, const char *arg2)
diff --git a/lib/nscd.c b/lib/nscd.c
index 732164ce..4f25cbe0 100644
--- a/lib/nscd.c
+++ b/lib/nscd.c
@@ -20,7 +20,7 @@
 int nscd_flush_cache (const char *service)
 {
 	int status, code;
-	const char *cmd = "/usr/sbin/nscd";
+	const char *cmd = "/bin/nscd";
 	const char *spawnedArgs[] = {"nscd", "-i", service, NULL};
 	const char *spawnedEnv[] = {NULL};
 
diff --git a/lib/setupenv.c b/lib/setupenv.c
index efa0f2d8..620c42a5 100644
--- a/lib/setupenv.c
+++ b/lib/setupenv.c
@@ -157,7 +157,7 @@ void setup_env (struct passwd *info)
 
 	if (NULL == cp) {
 		/* not specified, use a minimal default */
-		addenv ((info->pw_uid == 0) ? "PATH=/sbin:/bin:/usr/sbin:/usr/bin" : "PATH=/bin:/usr/bin", NULL);
+		addenv ("PATH=/bin", NULL);
 	} else if (strchr(cp, '=')) {
 		/* specified as name=value (PATH=...) */
 		addenv (cp, NULL);
diff --git a/lib/sssd.c b/lib/sssd.c
index cc5d4a46..79059593 100644
--- a/lib/sssd.c
+++ b/lib/sssd.c
@@ -26,7 +26,7 @@ sssd_flush_cache(int dbflags)
 	int          status, code, rv;
 	char         *p;
 	char         *sss_cache_args = NULL;
-	const char   *cmd = "/usr/sbin/sss_cache";
+	const char   *cmd = "/bin/sss_cache";
 	const char   *spawnedArgs[] = {"sss_cache", NULL, NULL};
 	const char   *spawnedEnv[] = {NULL};
 	struct stat  sb;
diff --git a/src/Makefile.am b/src/Makefile.am
index c935e05b..793aec44 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -3,7 +3,7 @@ EXTRA_DIST = \
 	.indent.pro
 
 ubindir = ${prefix}/bin
-usbindir = ${prefix}/sbin
+usbindir = ${prefix}/bin
 suidperms = 4755
 sgidperms = 2755
 
diff --git a/src/Makefile.in b/src/Makefile.in
index ee8702fa..c688b456 100644
--- a/src/Makefile.in
+++ b/src/Makefile.in
@@ -606,7 +606,7 @@ EXTRA_DIST = \
 	.indent.pro
 
 ubindir = ${prefix}/bin
-usbindir = ${prefix}/sbin
+usbindir = ${prefix}/bin
 suidperms = 4755
 sgidperms = 2755
 AM_CPPFLAGS = \
diff --git a/src/groupadd.c b/src/groupadd.c
index 483d2822..7fdbe1b3 100644
--- a/src/groupadd.c
+++ b/src/groupadd.c
@@ -564,11 +564,6 @@ int main (int argc, char **argv)
 	 */
 	process_flags (argc, argv, &flags);
 
-	if (run_parts ("/etc/shadow-maint/groupadd-pre.d", group_name,
-			Prog)) {
-		exit(1);
-	}
-
 #ifdef SHADOWGRP
 	is_shadow_grp = sgr_file_present ();
 #endif
@@ -587,11 +582,6 @@ int main (int argc, char **argv)
 
 	grp_update ();
 	close_files (&flags);
-	if (run_parts ("/etc/shadow-maint/groupadd-post.d", group_name,
-			Prog)) {
-		exit(1);
-	}
-
 
 	nscd_flush_cache ("group");
 	sssd_flush_cache (SSSD_DB_GROUP);
diff --git a/src/groupdel.c b/src/groupdel.c
index 19d67fc3..ba5adac8 100644
--- a/src/groupdel.c
+++ b/src/groupdel.c
@@ -395,11 +395,6 @@ int main (int argc, char **argv)
 		group_busy (group_id);
 	}
 
-	if (run_parts ("/etc/shadow-maint/groupdel-pre.d", group_name,
-			Prog)) {
-		exit(1);
-	}
-
 	/*
 	 * Do the hard stuff - open the files, delete the group entries,
 	 * then close and update the files.
@@ -410,11 +405,6 @@ int main (int argc, char **argv)
 
 	close_files (&flags);
 
-	if (run_parts ("/etc/shadow-maint/groupdel-post.d", group_name,
-			Prog)) {
-		exit(1);
-	}
-
 	nscd_flush_cache ("group");
 	sssd_flush_cache (SSSD_DB_GROUP);
 
diff --git a/src/su.c b/src/su.c
index cc74ab3a..6b9b6739 100644
--- a/src/su.c
+++ b/src/su.c
@@ -928,7 +928,7 @@ static void set_environment (struct passwd *pw)
 
 	cp = getdef_str ((pw->pw_uid == 0) ? "ENV_SUPATH" : "ENV_PATH");
 	if (NULL == cp) {
-		addenv ((pw->pw_uid == 0) ? "PATH=/sbin:/bin:/usr/sbin:/usr/bin" : "PATH=/bin:/usr/bin", NULL);
+		addenv ("PATH=/bin", NULL);
 	} else if (strchr(cp, '=')) {
 		addenv (cp, NULL);
 	} else {
diff --git a/src/useradd.c b/src/useradd.c
index 36c07b1f..9eda4712 100644
--- a/src/useradd.c
+++ b/src/useradd.c
@@ -82,9 +82,6 @@
 #ifndef SKEL_DIR
 #define SKEL_DIR "/etc/skel"
 #endif
-#ifndef USRSKELDIR
-#define USRSKELDIR "/usr/etc/skel"
-#endif
 #ifndef USER_DEFAULTS_FILE
 #define USER_DEFAULTS_FILE "/etc/default/useradd"
 #define NEW_USER_FILE "/etc/default/nuaddXXXXXX"
@@ -112,7 +109,9 @@ static const char *def_gname = "other";
 static const char *def_home = "/home";
 static const char *def_shell = "/bin/bash";
 static const char *def_template = SKEL_DIR;
+#ifdef USRSKELDIR
 static const char *def_usrtemplate = USRSKELDIR;
+#endif
 static const char *def_create_mail_spool = "yes";
 static const char *def_btrfs_subvolume_home = "no";
 static const char *def_btrfs_subvolume_system = "no";
@@ -440,6 +439,7 @@ get_defaults(const struct option_flags *flags)
 				def_template = xstrdup(ccp);
 		}
 
+#ifdef USRSKELDIR
 		/*
 		 * Default Usr Skeleton information
 		 */
@@ -453,6 +453,7 @@ get_defaults(const struct option_flags *flags)
 				def_usrtemplate = xstrdup(ccp);
 			}
 		}
+#endif
 
 		/*
 		 * Create by default user mail spool or not ?
@@ -514,7 +515,9 @@ static void show_defaults (void)
 	printf ("EXPIRE=%s\n", def_expire);
 	printf ("SHELL=%s\n", def_shell);
 	printf ("SKEL=%s\n", def_template);
+#ifdef USRSKELDIR
 	printf ("USRSKEL=%s\n", def_usrtemplate);
+#endif
 	printf ("CREATE_MAIL_SPOOL=%s\n", def_create_mail_spool);
 	printf ("BTRFS_SUBVOLUME_HOME=%s\n", def_btrfs_subvolume_home);
 	printf ("BTRFS_SUBVOLUME_SYSTEM=%s\n", def_btrfs_subvolume_system);
@@ -539,7 +542,9 @@ set_defaults(void)
 	bool  out_expire = false;
 	bool  out_shell = false;
 	bool  out_skel = false;
+#ifdef USRSKELDIR
 	bool  out_usrskel = false;
+#endif
 	bool  out_create_mail_spool = false;
 	bool  out_btrfs_subvolume_home = false;
 	bool  out_btrfs_subvolume_system = false;
@@ -642,9 +647,11 @@ set_defaults(void)
 		} else if (!out_skel && streq(buf, DSKEL)) {
 			fprintf(ofp, DSKEL "=%s\n", def_template);
 			out_skel = true;
+#ifdef USRSKELDIR
 		} else if (!out_usrskel && streq(buf, DUSRSKEL)) {
 			fprintf(ofp, DUSRSKEL "=%s\n", def_usrtemplate);
 			out_usrskel = true;
+#endif
 		} else if (!out_create_mail_spool
 			   && streq(buf, DCREATE_MAIL_SPOOL))
 		{
@@ -691,8 +698,10 @@ set_defaults(void)
 		fprintf (ofp, DSHELL "=%s\n", def_shell);
 	if (!out_skel)
 		fprintf (ofp, DSKEL "=%s\n", def_template);
+#ifdef USRSKELDIR
 	if (!out_usrskel)
 		fprintf (ofp, DUSRSKEL "=%s\n", def_usrtemplate);
+#endif
 
 	if (!out_create_mail_spool)
 		fprintf (ofp, DCREATE_MAIL_SPOOL "=%s\n", def_create_mail_spool);
@@ -1378,8 +1387,7 @@ static void process_flags (int argc, char **argv, struct option_flags *flags)
 				}
 				if (!streq(optarg, "")
 				     && '*'  != optarg[0]
-				     && !streq(optarg, "/sbin/nologin")
-				     && !streq(optarg, "/usr/sbin/nologin")
+				     && !streq(optarg, "/bin/nologin")
 				     && (   stat(optarg, &st) != 0
 				         || S_ISDIR(st.st_mode)
 				         || access(optarg, X_OK) != 0)) {
@@ -2011,7 +2019,7 @@ static void lastlog_reset (uid_t uid)
 
 static void tallylog_reset (const char *user_name)
 {
-	const char pam_tally2[] = "/sbin/pam_tally2";
+	const char pam_tally2[] = "/bin/pam_tally2";
 	const char *pname;
 	pid_t childpid;
 	int failed;
@@ -2506,11 +2514,6 @@ int main (int argc, char **argv)
 	process_flags (argc, argv, &flags);
 	process_selinux = !flags.chroot && !flags.prefix;
 
-	if (run_parts ("/etc/shadow-maint/useradd-pre.d", user_name,
-			"useradd")) {
-		exit(1);
-	}
-
 	/*
 	 * See if we are messing with the defaults file, or creating
 	 * a new user.
@@ -2662,8 +2665,10 @@ int main (int argc, char **argv)
 		if (home_added) {
 			copy_tree (def_template, prefix_user_home, false,
 			           (uid_t)-1, user_id, (gid_t)-1, user_gid);
+#ifdef USRSKELDIR
 			copy_tree (def_usrtemplate, prefix_user_home, false,
 			           (uid_t)-1, user_id, (gid_t)-1, user_gid);
+#endif
 		} else {
 			eprintf(_("%s: warning: the home directory %s already exists.\n"
 			          "%s: Not copying any file from skel directory into it.\n"),
@@ -2677,10 +2682,5 @@ int main (int argc, char **argv)
 		create_mail (&flags);
 	}
 
-	if (run_parts ("/etc/shadow-maint/useradd-post.d", user_name,
-			"useradd")) {
-		exit(1);
-	}
-
 	return E_SUCCESS;
 }
diff --git a/src/userdel.c b/src/userdel.c
index d38110db..12be607e 100644
--- a/src/userdel.c
+++ b/src/userdel.c
@@ -965,10 +965,6 @@ int main (int argc, char **argv)
 	{
 		const struct passwd *pwd;
 
-		if (run_parts ("/etc/shadow-maint/userdel-pre.d", user_name,
-				"userdel")) {
-			exit(1);
-		}
 		pw_open(O_RDONLY);
 		pwd = pw_locate (user_name); /* we care only about local users */
 		if (NULL == pwd) {
@@ -1136,10 +1132,6 @@ int main (int argc, char **argv)
 		user_cancel (user_name);
 	close_files (&flags);
 
-	if (run_parts ("/etc/shadow-maint/userdel-post.d", user_name, "userdel")) {
-		exit(1);
-	}
-
 #ifdef WITH_TCB
 	if (remove_tcbdir (user_name, user_id)) {
 		errors = true;
diff --git a/src/usermod.c b/src/usermod.c
index bd9dc82c..27f3690d 100644
--- a/src/usermod.c
+++ b/src/usermod.c
@@ -1189,8 +1189,7 @@ process_flags(int argc, char **argv, struct option_flags *flags)
 				}
 				if (!streq(optarg, "")
 				     && '*'  != optarg[0]
-				     && !streq(optarg, "/sbin/nologin")
-				     && !streq(optarg, "/usr/sbin/nologin")
+				     && !streq(optarg, "/bin/nologin")
 				     && (   stat(optarg, &st) != 0
 				         || S_ISDIR(st.st_mode)
 				         || access(optarg, X_OK) != 0)) {
